<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="unawaz1996" />

<meta name="date" content="2023-03-14" />

<title>Quality Assessment of NMD RNA-seq data</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.27/datatables.js"></script>
<link href="site_libs/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.12.1/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.2.1/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.2.1/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">NMD-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Quality-control.html">Pre-processing</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Differential Expression Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="DEG-analysis.html">DEG analysis</a>
    </li>
    <li>
      <a href="Enichment-analysis-goseq.html">Enrichment analysis using goseq</a>
    </li>
    <li>
      <a href="Enichment-analysis-fgsea.html">Enrichment analysis using fgsea</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Transcript Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="transcript-proprocessing.html">Transcript exploration</a>
    </li>
    <li>
      <a href="Differential-transcript-usage.html">Isoform usage analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/unawaz1996/NMD-analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Quality Assessment of NMD RNA-seq data</h1>
<h4 class="author">unawaz1996</h4>
<h4 class="date">2023-03-14</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-04-24
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>NMD-analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230314code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230314)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230314code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230314)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/home/neuro/Documents/NMD_analysis/Analysis/NMD-analysis/output/Transcript/Thesis_figures/transcript-distribution.svg
</td>
<td style="text-align:left;">
output/Transcript/Thesis_figures/transcript-distribution.svg
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/neuro/Documents/NMD_analysis/Analysis/NMD-analysis/output/Transcript/Thesis_figures/transcript-composition.svg
</td>
<td style="text-align:left;">
output/Transcript/Thesis_figures/transcript-composition.svg
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomunawaz1996NMDanalysistree6110b63d1ba586cb64035203369ea8f1230244cdtargetblank6110b63a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/unawaz1996/NMD-analysis/tree/6110b63d1ba586cb64035203369ea8f1230244cd" target="_blank">6110b63</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomunawaz1996NMDanalysistree6110b63d1ba586cb64035203369ea8f1230244cdtargetblank6110b63a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/unawaz1996/NMD-analysis/tree/6110b63d1ba586cb64035203369ea8f1230244cd" target="_blank">6110b63</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/Differential-transcript-usage.nb.html
    Ignored:    analysis/Enichment-analysis-fgsea.nb.html
    Ignored:    analysis/Enichment-analysis-goseq.nb.html

Untracked files:
    Untracked:  PCA.png
    Untracked:  PCA_plot.pdf
    Untracked:  PCA_transcript.png
    Untracked:  analysis/Differential-transcript-expression.Rmd
    Untracked:  analysis/Differential-transcript-usage.Rmd
    Untracked:  analysis/UPF3B_KD.Rmd
    Untracked:  analysis/figures.Rmd
    Untracked:  analysis/transcript-preprocessing.Rmd
    Untracked:  code/QuasR_log_7219053fb51a5.txt
    Untracked:  code/QuasR_log_74f7f374c3271.txt
    Untracked:  code/QuasR_log_7a13e31c3d08a.txt
    Untracked:  code/QuasR_log_7ac9f1cedf971.txt
    Untracked:  code/QuasR_log_7bbdc44874ffd.txt
    Untracked:  code/eisaR.R
    Untracked:  code/external_code/
    Untracked:  data/LTK_Sample Metafile_V3.txt
    Untracked:  data/Mus_musculus.GRCm39.105__nifs.tsv
    Untracked:  data/fastqc/
    Untracked:  data/nif_output/
    Untracked:  data/samples.txt
    Untracked:  output/DEG-limma-results.Rda
    Untracked:  output/DEG-list.Rda
    Untracked:  output/DEG/
    Untracked:  output/ISAR/
    Untracked:  output/QC/
    Untracked:  output/Transcript/
    Untracked:  output/isoformSwitchAnalyzeR_isoform_AA_complete.fasta
    Untracked:  output/isoformSwitchAnalyzeR_isoform_AA_subset_1_of_3.fasta
    Untracked:  output/isoformSwitchAnalyzeR_isoform_AA_subset_2_of_3.fasta
    Untracked:  output/isoformSwitchAnalyzeR_isoform_AA_subset_3_of_3.fasta
    Untracked:  output/isoformSwitchAnalyzeR_isoform_nt.fasta
    Untracked:  output/limma-matrices.Rda

Unstaged changes:
    Modified:   analysis/DEG-analysis.Rmd
    Modified:   analysis/Enichment-analysis-fgsea.Rmd
    Modified:   analysis/_site.yml
    Modified:   code/functions.R
    Modified:   code/libraries.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Quality-control.Rmd</code>) and
HTML (<code>docs/Quality-control.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/6110b63d1ba586cb64035203369ea8f1230244cd/analysis/Quality-control.Rmd" target="_blank">6110b63</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-24
</td>
<td>
wflow_publish("analysis/Quality-control.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/unawaz1996/NMD-analysis/be057f211fab4959f5a2c0a5db98d864490d6e34/docs/Quality-control.html" target="_blank">be057f2</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/f5fc094a89d3df7eb828b53421fff6ae42fa37d1/analysis/Quality-control.Rmd" target="_blank">f5fc094</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-24
</td>
<td>
wflow_publish("analysis/Quality-control.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/unawaz1996/NMD-analysis/5196f4120527db07aaf6583bc6850638d325cac7/docs/Quality-control.html" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/b9393bc35269a6a75fb81327deeaaa43cb097739/analysis/Quality-control.Rmd" target="_blank">b9393bc</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
<td>
wflow_publish(c("analysis/index.Rmd", "analysis/Quality-control.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/unawaz1996/NMD-analysis/650317eae471a13c56bfde58444267c6ba76b080/docs/Quality-control.html" target="_blank">650317e</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-03-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/f2ed11a9d95ca42cad6af51db48c4fb29b4e94c8/analysis/Quality-control.Rmd" target="_blank">f2ed11a</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-03-22
</td>
<td>
wflow_publish(c("analysis/index.Rmd",
"analysis/Enichment-analysis-fgsea.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level2">
<h2>Set-up</h2>
<p>In order to perform QC, an EnsDB object was obtained using the
<code>AnnotationHub</code> package. This provided the GC content and
length for each of the transcripts contained in the release.</p>
<p>Metadata for each fastq file was also loaded. Reads were provided as
paired-end reads, with n = 3 samples for each genotype.</p>
</div>
<div id="fastqc-summary" class="section level1">
<h1>FastQC summary</h1>
<div id="fastqc-summary-1" class="section level2">
<h2>FastQC summary</h2>
<div class="figure" style="text-align: center">
<img src="figure/Quality-control.Rmd/fastq-summary-1.png" alt="*Basic statistics summary plot. Figure (a) hsows the summary of the PASS/FAIl flags prior to base quality and adapter trimming. Figure (b) shows the summary of PASS/FAIL flags after quality trimming with trimgalore. Green: PASS; Yellow: FAIL; Red: WARN *" width="576" />
<p class="caption">
<em>Basic statistics summary plot. Figure (a) hsows the summary of the
PASS/FAIl flags prior to base quality and adapter trimming. Figure (b)
shows the summary of PASS/FAIL flags after quality trimming with
trimgalore. Green: PASS; Yellow: FAIL; Red: WARN </em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fastq-summary-1">
Past versions of fastq-summary-1.png
</button>
</p>
<div id="fig-fastq-summary-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/fastq-summary-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="figure" style="text-align: center">
<img src="figure/Quality-control.Rmd/unnamed-chunk-7-1.png" alt="*Sequence Length distribution of the RNA seq reads (a) before and (b) after quality trimming. Only reads with the base length of &gt;  150bp were retained after quality trimming.*" width="960" />
<p class="caption">
<em>Sequence Length distribution of the RNA seq reads (a) before and (b)
after quality trimming. Only reads with the base length of &gt; 150bp
were retained after quality trimming.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-7-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="library-sizes" class="section level2">
<h2>Library sizes</h2>
<p>Library sizes for unprocessed data ranged between 86,178,215 and
129,449,694 reads.</p>
<div class="figure" style="text-align: center">
<img src="figure/Quality-control.Rmd/unnamed-chunk-9-1.png" alt="*Total numner 0f reads from each sample (a) before and (b) after quality trimming with trimgalore.*" width="672" />
<p class="caption">
<em>Total numner 0f reads from each sample (a) before and (b) after
quality trimming with trimgalore.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-9-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gc-content" class="section level2">
<h2>GC content</h2>
<p>In poly(A) selected RNA-seq library preparation methods, the
nonuniform coverage of transcripts is a prevalent issue. As poly(A) tail
only occurs at the 3’ end of the mRNA, this can usually result in an
over-representation of the 3’ end. Bias at the 5′ end of RNA can also
happen because of various factors, such as the fragmentation method (the
5′ end of RNA is more stable), reverse transcription from RNA to cDNA
and strand-oriented library construction protocol.</p>
<p>GC content allows the exploration of the sequencing coverage and can
indicate issues in overrepresentation. It has been observed that either
high or low GC content will result in lower depth coverage.</p>
<p><img src="figure/Quality-control.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-10-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="trimmed-data" class="section level2">
<h2>Trimmed data</h2>
<table>
<colgroup>
<col width="5%" />
<col width="14%" />
<col width="12%" />
<col width="12%" />
<col width="9%" />
<col width="11%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">names</th>
<th align="center">Group</th>
<th align="center">Condition_UPF3B</th>
<th align="center">Condition_UPF3A</th>
<th align="center">Label</th>
<th align="center">Filename</th>
<th align="center">Raw</th>
<th align="center">Trimmed</th>
<th align="center">Discarded</th>
<th align="center">Retained</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">212</td>
<td align="center">Control</td>
<td align="center">Control</td>
<td align="center">Control</td>
<td align="center">S2_Vect_A</td>
<td align="center">212_L4_1.fq.gz</td>
<td align="center">129449694</td>
<td align="center">129423518</td>
<td align="center">0.0002022</td>
<td align="center">0.9998</td>
</tr>
<tr class="even">
<td align="center">212</td>
<td align="center">Control</td>
<td align="center">Control</td>
<td align="center">Control</td>
<td align="center">S2_Vect_A</td>
<td align="center">212_L4_2.fq.gz</td>
<td align="center">129449694</td>
<td align="center">129423518</td>
<td align="center">0.0002259</td>
<td align="center">0.9998</td>
</tr>
<tr class="odd">
<td align="center">213</td>
<td align="center">UPF3A_KD</td>
<td align="center">Control</td>
<td align="center">Knockdown</td>
<td align="center">S2_shRNA_A</td>
<td align="center">213_L4_1.fq.gz</td>
<td align="center">124647120</td>
<td align="center">124618959</td>
<td align="center">0.0001968</td>
<td align="center">0.9998</td>
</tr>
<tr class="even">
<td align="center">213</td>
<td align="center">UPF3A_KD</td>
<td align="center">Control</td>
<td align="center">Knockdown</td>
<td align="center">S2_shRNA_A</td>
<td align="center">213_L4_2.fq.gz</td>
<td align="center">124647120</td>
<td align="center">124618959</td>
<td align="center">0.0001568</td>
<td align="center">0.9998</td>
</tr>
<tr class="odd">
<td align="center">214</td>
<td align="center">UPF3A_OE</td>
<td align="center">Control</td>
<td align="center">Overexpression</td>
<td align="center">S2_cDNA_A</td>
<td align="center">214_L4_1.fq.gz</td>
<td align="center">122589413</td>
<td align="center">122565286</td>
<td align="center">0.0001785</td>
<td align="center">0.9998</td>
</tr>
<tr class="even">
<td align="center">214</td>
<td align="center">UPF3A_OE</td>
<td align="center">Control</td>
<td align="center">Overexpression</td>
<td align="center">S2_cDNA_A</td>
<td align="center">214_L4_2.fq.gz</td>
<td align="center">122589413</td>
<td align="center">122565286</td>
<td align="center">0.0002056</td>
<td align="center">0.9998</td>
</tr>
<tr class="odd">
<td align="center">215</td>
<td align="center">UPF3B_KD</td>
<td align="center">Knockdown</td>
<td align="center">Control</td>
<td align="center">T2_Vect_A</td>
<td align="center">215_L4_1.fq.gz</td>
<td align="center">86178215</td>
<td align="center">86164704</td>
<td align="center">0.0002234</td>
<td align="center">0.9998</td>
</tr>
<tr class="even">
<td align="center">215</td>
<td align="center">UPF3B_KD</td>
<td align="center">Knockdown</td>
<td align="center">Control</td>
<td align="center">T2_Vect_A</td>
<td align="center">215_L4_2.fq.gz</td>
<td align="center">86178215</td>
<td align="center">86164704</td>
<td align="center">0.0002671</td>
<td align="center">0.9997</td>
</tr>
<tr class="odd">
<td align="center">216</td>
<td align="center">UPF3A_KD_UPF3B_KD</td>
<td align="center">Knockdown</td>
<td align="center">Knockdown</td>
<td align="center">T2_shRNA_A</td>
<td align="center">216_L4_1.fq.gz</td>
<td align="center">91807145</td>
<td align="center">91790756</td>
<td align="center">0.0001893</td>
<td align="center">0.9998</td>
</tr>
<tr class="even">
<td align="center">216</td>
<td align="center">UPF3A_KD_UPF3B_KD</td>
<td align="center">Knockdown</td>
<td align="center">Knockdown</td>
<td align="center">T2_shRNA_A</td>
<td align="center">216_L4_2.fq.gz</td>
<td align="center">91807145</td>
<td align="center">91790756</td>
<td align="center">0.0001951</td>
<td align="center">0.9998</td>
</tr>
<tr class="odd">
<td align="center">217</td>
<td align="center">UPF3A_OE_UPF3B_KD</td>
<td align="center">Knockdown</td>
<td align="center">Overexpression</td>
<td align="center">T2_cDNA_A</td>
<td align="center">217_L4_1.fq.gz</td>
<td align="center">106731239</td>
<td align="center">106709299</td>
<td align="center">0.0002528</td>
<td align="center">0.9997</td>
</tr>
<tr class="even">
<td align="center">217</td>
<td align="center">UPF3A_OE_UPF3B_KD</td>
<td align="center">Knockdown</td>
<td align="center">Overexpression</td>
<td align="center">T2_cDNA_A</td>
<td align="center">217_L4_2.fq.gz</td>
<td align="center">106731239</td>
<td align="center">106709299</td>
<td align="center">0.001988</td>
<td align="center">0.998</td>
</tr>
<tr class="odd">
<td align="center">218</td>
<td align="center">Control</td>
<td align="center">Control</td>
<td align="center">Control</td>
<td align="center">S2_Vect_B</td>
<td align="center">218_L4_1.fq.gz</td>
<td align="center">109175104</td>
<td align="center">109150719</td>
<td align="center">0.0002647</td>
<td align="center">0.9997</td>
</tr>
<tr class="even">
<td align="center">218</td>
<td align="center">Control</td>
<td align="center">Control</td>
<td align="center">Control</td>
<td align="center">S2_Vect_B</td>
<td align="center">218_L4_2.fq.gz</td>
<td align="center">109175104</td>
<td align="center">109150719</td>
<td align="center">0.0002736</td>
<td align="center">0.9997</td>
</tr>
<tr class="odd">
<td align="center">219</td>
<td align="center">UPF3A_KD</td>
<td align="center">Control</td>
<td align="center">Knockdown</td>
<td align="center">S2_shRNA_B</td>
<td align="center">219_L4_1.fq.gz</td>
<td align="center">99242640</td>
<td align="center">99216132</td>
<td align="center">0.0001691</td>
<td align="center">0.9998</td>
</tr>
<tr class="even">
<td align="center">219</td>
<td align="center">UPF3A_KD</td>
<td align="center">Control</td>
<td align="center">Knockdown</td>
<td align="center">S2_shRNA_B</td>
<td align="center">219_L4_2.fq.gz</td>
<td align="center">99242640</td>
<td align="center">99216132</td>
<td align="center">0.000183</td>
<td align="center">0.9998</td>
</tr>
<tr class="odd">
<td align="center">220</td>
<td align="center">UPF3A_OE</td>
<td align="center">Control</td>
<td align="center">Overexpression</td>
<td align="center">S2_cDNA_B</td>
<td align="center">220_L4_1.fq.gz</td>
<td align="center">101245190</td>
<td align="center">101226023</td>
<td align="center">0.000212</td>
<td align="center">0.9998</td>
</tr>
<tr class="even">
<td align="center">220</td>
<td align="center">UPF3A_OE</td>
<td align="center">Control</td>
<td align="center">Overexpression</td>
<td align="center">S2_cDNA_B</td>
<td align="center">220_L4_2.fq.gz</td>
<td align="center">101245190</td>
<td align="center">101226023</td>
<td align="center">0.0001361</td>
<td align="center">0.9999</td>
</tr>
<tr class="odd">
<td align="center">221</td>
<td align="center">UPF3B_KD</td>
<td align="center">Knockdown</td>
<td align="center">Control</td>
<td align="center">T2_Vect_B</td>
<td align="center">221_L4_1.fq.gz</td>
<td align="center">114129302</td>
<td align="center">114107031</td>
<td align="center">0.0002022</td>
<td align="center">0.9998</td>
</tr>
<tr class="even">
<td align="center">221</td>
<td align="center">UPF3B_KD</td>
<td align="center">Knockdown</td>
<td align="center">Control</td>
<td align="center">T2_Vect_B</td>
<td align="center">221_L4_2.fq.gz</td>
<td align="center">114129302</td>
<td align="center">114107031</td>
<td align="center">0.0002259</td>
<td align="center">0.9998</td>
</tr>
<tr class="odd">
<td align="center">222</td>
<td align="center">UPF3A_KD_UPF3B_KD</td>
<td align="center">Knockdown</td>
<td align="center">Knockdown</td>
<td align="center">T2_shRNA_B</td>
<td align="center">222_L4_1.fq.gz</td>
<td align="center">101073551</td>
<td align="center">1.01e+08</td>
<td align="center">0.0001968</td>
<td align="center">0.9998</td>
</tr>
<tr class="even">
<td align="center">222</td>
<td align="center">UPF3A_KD_UPF3B_KD</td>
<td align="center">Knockdown</td>
<td align="center">Knockdown</td>
<td align="center">T2_shRNA_B</td>
<td align="center">222_L4_2.fq.gz</td>
<td align="center">101073551</td>
<td align="center">1.01e+08</td>
<td align="center">0.0001568</td>
<td align="center">0.9998</td>
</tr>
<tr class="odd">
<td align="center">223</td>
<td align="center">UPF3A_OE_UPF3B_KD</td>
<td align="center">Knockdown</td>
<td align="center">Overexpression</td>
<td align="center">T2_cDNA_B</td>
<td align="center">223_L4_1.fq.gz</td>
<td align="center">94512220</td>
<td align="center">94324353</td>
<td align="center">0.0001785</td>
<td align="center">0.9998</td>
</tr>
<tr class="even">
<td align="center">223</td>
<td align="center">UPF3A_OE_UPF3B_KD</td>
<td align="center">Knockdown</td>
<td align="center">Overexpression</td>
<td align="center">T2_cDNA_B</td>
<td align="center">223_L4_2.fq.gz</td>
<td align="center">94512220</td>
<td align="center">94324353</td>
<td align="center">0.0002056</td>
<td align="center">0.9998</td>
</tr>
<tr class="odd">
<td align="center">224</td>
<td align="center">Control</td>
<td align="center">Control</td>
<td align="center">Control</td>
<td align="center">S2_Vect_C</td>
<td align="center">224_L4_1.fq.gz</td>
<td align="center">104412970</td>
<td align="center">104385336</td>
<td align="center">0.0002234</td>
<td align="center">0.9998</td>
</tr>
<tr class="even">
<td align="center">224</td>
<td align="center">Control</td>
<td align="center">Control</td>
<td align="center">Control</td>
<td align="center">S2_Vect_C</td>
<td align="center">224_L4_2.fq.gz</td>
<td align="center">104412970</td>
<td align="center">104385336</td>
<td align="center">0.0002671</td>
<td align="center">0.9997</td>
</tr>
<tr class="odd">
<td align="center">225</td>
<td align="center">UPF3A_KD</td>
<td align="center">Control</td>
<td align="center">Knockdown</td>
<td align="center">S2_shRNA_C</td>
<td align="center">225_L4_1.fq.gz</td>
<td align="center">93692109</td>
<td align="center">93666471</td>
<td align="center">0.0001893</td>
<td align="center">0.9998</td>
</tr>
<tr class="even">
<td align="center">225</td>
<td align="center">UPF3A_KD</td>
<td align="center">Control</td>
<td align="center">Knockdown</td>
<td align="center">S2_shRNA_C</td>
<td align="center">225_L4_2.fq.gz</td>
<td align="center">93692109</td>
<td align="center">93666471</td>
<td align="center">0.0001951</td>
<td align="center">0.9998</td>
</tr>
<tr class="odd">
<td align="center">226</td>
<td align="center">UPF3A_OE</td>
<td align="center">Control</td>
<td align="center">Overexpression</td>
<td align="center">S2_cDNA_C</td>
<td align="center">226_L4_1.fq.gz</td>
<td align="center">95761339</td>
<td align="center">95745148</td>
<td align="center">0.0002528</td>
<td align="center">0.9997</td>
</tr>
<tr class="even">
<td align="center">226</td>
<td align="center">UPF3A_OE</td>
<td align="center">Control</td>
<td align="center">Overexpression</td>
<td align="center">S2_cDNA_C</td>
<td align="center">226_L4_2.fq.gz</td>
<td align="center">95761339</td>
<td align="center">95745148</td>
<td align="center">0.001988</td>
<td align="center">0.998</td>
</tr>
<tr class="odd">
<td align="center">227</td>
<td align="center">UPF3B_KD</td>
<td align="center">Knockdown</td>
<td align="center">Control</td>
<td align="center">T2_Vect_C</td>
<td align="center">227_L4_1.fq.gz</td>
<td align="center">90236088</td>
<td align="center">90219573</td>
<td align="center">0.0002647</td>
<td align="center">0.9997</td>
</tr>
<tr class="even">
<td align="center">227</td>
<td align="center">UPF3B_KD</td>
<td align="center">Knockdown</td>
<td align="center">Control</td>
<td align="center">T2_Vect_C</td>
<td align="center">227_L4_2.fq.gz</td>
<td align="center">90236088</td>
<td align="center">90219573</td>
<td align="center">0.0002736</td>
<td align="center">0.9997</td>
</tr>
<tr class="odd">
<td align="center">228</td>
<td align="center">UPF3A_KD_UPF3B_KD</td>
<td align="center">Knockdown</td>
<td align="center">Knockdown</td>
<td align="center">T2_shRNA_C</td>
<td align="center">228_L4_1.fq.gz</td>
<td align="center">92761079</td>
<td align="center">92741417</td>
<td align="center">0.0001691</td>
<td align="center">0.9998</td>
</tr>
<tr class="even">
<td align="center">228</td>
<td align="center">UPF3A_KD_UPF3B_KD</td>
<td align="center">Knockdown</td>
<td align="center">Knockdown</td>
<td align="center">T2_shRNA_C</td>
<td align="center">228_L4_2.fq.gz</td>
<td align="center">92761079</td>
<td align="center">92741417</td>
<td align="center">0.000183</td>
<td align="center">0.9998</td>
</tr>
<tr class="odd">
<td align="center">229</td>
<td align="center">UPF3A_OE_UPF3B_KD</td>
<td align="center">Knockdown</td>
<td align="center">Overexpression</td>
<td align="center">T2_cDNA_C</td>
<td align="center">229_L4_1.fq.gz</td>
<td align="center">102071888</td>
<td align="center">102057991</td>
<td align="center">0.000212</td>
<td align="center">0.9998</td>
</tr>
<tr class="even">
<td align="center">229</td>
<td align="center">UPF3A_OE_UPF3B_KD</td>
<td align="center">Knockdown</td>
<td align="center">Overexpression</td>
<td align="center">T2_cDNA_C</td>
<td align="center">229_L4_2.fq.gz</td>
<td align="center">102071888</td>
<td align="center">102057991</td>
<td align="center">0.0001361</td>
<td align="center">0.9999</td>
</tr>
</tbody>
</table>
<p>After adapter trimming, &lt; 1% of reads were discarded.</p>
</div>
</div>
<div id="aligned-data---salmon-quantifications" class="section level1">
<h1>Aligned data - Salmon quantifications</h1>
<p>Counts were generated using Salmon. Briefly, an index was generated
using the GRCm39 build of the mouse transcript with decoys.txt.
Selective alignment mode was used. The reads were also aligned to the
human genome to check for genotyping and to ensure no mislabeling
occurred during any part of the bench work. Counts were imported as
transcript-level and gene-level using <code>tximport</code> and
<code>tximeta</code> respectively.</p>
<p>Annotation data was loaded as an <code>EnsDb</code> object, using
Ensembl resealse 107. Transcript level gene lengths and GC content was
converted to gene level values using:</p>
<ul>
<li>GC Content: The total GC content divided by the total length of
transcripts</li>
<li>Gene Length: The mean transcript length</li>
</ul>
<div id="genotype-checking" class="section level2">
<h2>Genotype checking</h2>
<div class="figure" style="text-align: center">
<img src="figure/Quality-control.Rmd/unnamed-chunk-14-1.png" alt="All samples demonstrated the expected expression patterns, no mislabeling was detected in the dataset." width="960" />
<p class="caption">
All samples demonstrated the expected expression patterns, no
mislabeling was detected in the dataset.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-14-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="filtering" class="section level2">
<h2>Filtering</h2>
<p><img src="figure/Quality-control.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-16-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="library-size" class="section level2">
<h2>Library size</h2>
<p><img src="figure/Quality-control.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-17-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="counts-assingment-rate" class="section level2">
<h2>Counts assingment rate</h2>
<p><img src="figure/Quality-control.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-18-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="total-detected-genes" class="section level2">
<h2>Total detected genes</h2>
<p><img src="figure/Quality-control.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-19-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="transcript-level-exploration-of-data" class="section level2">
<h2>Transcript level exploration of data</h2>
<pre><code># A tibble: 2 × 3
  Filter `median(rowMeans)` `sd(rowMeans)`
  &lt;fct&gt;               &lt;dbl&gt;          &lt;dbl&gt;
1 before             0.0370           115.
2 after             13.3              298.</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/Quality-control.Rmd/unnamed-chunk-21-1.png" alt="*Log10 of the mean TPMs (transcript per million) over all samples before and after filtering out low-expressed transcripts and genes*" width="576" />
<p class="caption">
<em>Log10 of the mean TPMs (transcript per million) over all samples
before and after filtering out low-expressed transcripts and genes</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-21-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="figure" style="text-align: center">
<img src="figure/Quality-control.Rmd/unnamed-chunk-23-1.png" alt="*Violin plots showing the distribution of the number of transcripts per gene (in logarithmic scale). Violin width is scaled by the total number of observations while jittered points represent actual observations.*" width="576" />
<p class="caption">
<em>Violin plots showing the distribution of the number of transcripts
per gene (in logarithmic scale). Violin width is scaled by the total
number of observations while jittered points represent actual
observations.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-23-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pca" class="section level2">
<h2>PCA</h2>
<div class="figure" style="text-align: center">
<img src="figure/Quality-control.Rmd/unnamed-chunk-24-1.png" alt="*Principal component analysis of gene (left) and transcript (right) level data. PCA was performed on log2 transformed TPMs after filtering for each respective datatype (gene and transcript level). Gene level PCA shows that all samples are clustering closely based on condition, with no impact of library size in PC1 and PC2. Transcript level PCA shows that samples are clustering close to their conditions based on PC1, however one of the samples of the UPF3A OE in UPF3B KD cell line (sample 223), seems to deviate from its condition group and the rest of the data, so needs to be further investigated*" width="960" />
<p class="caption">
<em>Principal component analysis of gene (left) and transcript (right)
level data. PCA was performed on log2 transformed TPMs after filtering
for each respective datatype (gene and transcript level). Gene level PCA
shows that all samples are clustering closely based on condition, with
no impact of library size in PC1 and PC2. Transcript level PCA shows
that samples are clustering close to their conditions based on PC1,
however one of the samples of the UPF3A OE in UPF3B KD cell line (sample
223), seems to deviate from its condition group and the rest of the
data, so needs to be further investigated</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/be057f211fab4959f5a2c0a5db98d864490d6e34/docs/figure/Quality-control.Rmd/unnamed-chunk-24-1.png" target="_blank">be057f2</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-24-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>To check if the same variation is observed at PC3 in gene level:</p>
<div class="figure" style="text-align: center">
<img src="figure/Quality-control.Rmd/unnamed-chunk-25-1.png" alt="*Principal component analysis of gene level data showing PC2 and PC3. Sample 223 seems to cluster away from its condition group in PC3*" width="576" />
<p class="caption">
<em>Principal component analysis of gene level data showing PC2 and PC3.
Sample 223 seems to cluster away from its condition group in PC3</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-25-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="investigating-the-differences-in-samples-at-transcript-level"
class="section level3">
<h3>Investigating the differences in samples at transcript level</h3>
<p><img src="figure/Quality-control.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-26-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Quality-control.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-27-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="figure" style="text-align: center">
<img src="figure/Quality-control.Rmd/unnamed-chunk-29-1.png" alt="*Correlations between the first three principal components and measured variables at transcript level. Sample conditions were converted to an ordered categorical variable for the purposes of visualisation*" width="672" />
<p class="caption">
<em>Correlations between the first three principal components and
measured variables at transcript level. Sample conditions were converted
to an ordered categorical variable for the purposes of
visualisation</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-29-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Transcripts were divided in 10 approximately equal sized bins based
on increasing length, and 10 approximately equal sized bins based on
increasing GC content, with the final GC/Length bins being the
combination 100 bins using both sets. The contribution of each gene to
PC1 and PC2 was assessed and a t-test performed on each bin.</p>
<p>If any bin makes a contribution to PC1 the mean will be clearly
non-zero, whilst if there is no contribution the mean will be near zero.
In this way, the impact of gene length and GC content on variance within
the dataset can be assessed.</p>
<div class="figure" style="text-align: center">
<img src="figure/Quality-control.Rmd/unnamed-chunk-30-1.png" alt="*Contribution of each GC/Length Bin to PC1 and PC2. Fill colours indicate the t-statistic, with tranparency denoting significance as -log10(p), using Bonferroni-adjusted p-values.*" width="672" />
<p class="caption">
<em>Contribution of each GC/Length Bin to PC1 and PC2. Fill colours
indicate the t-statistic, with tranparency denoting significance as
-log10(p), using Bonferroni-adjusted p-values.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-30-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="correlation-of-samples" class="section level2">
<h2>Correlation of samples</h2>
<p><img src="figure/Quality-control.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-31-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-e73b99e6611f7d9abfab" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e73b99e6611f7d9abfab">{"x":{"filter":"none","vertical":false,"caption":"<caption>For each sample, a mean of its correlation to the other samples was calculated followed by a z-score transformation. In principle, samples that deviate 3 standard deviations away from the mean would be removed.<\/caption>","data":[["212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229"],[0.954734286747732,0.535319230574479,0.826775360096044,-0.821323088684962,0.0408621209623087,0.104786900612033,0.56241047767458,0.16463033904994,0.170966691558411,0.634588998011526,0.277382192530707,-3.47139484798005,0.692749219205917,0.515706887916418,0.244482446503435,-0.361895743984819,-0.631282137678046,-0.439499333115646],[0.804394359198634,1.85538259776234,1.21446816644299,0.100602146627092,-0.122318734007776,0.64345378346285,-0.0513398699649521,0.028159421919556,0.399166289539115,1.16483054573848,0.541942311912305,-2.27633986472193,-0.535863004478317,-0.62535398074132,-0.725044495679846,-0.311916663388157,-1.54524634057724,-0.558976669043756]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>transcript<\/th>\n      <th>gene<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<div id="correlation-of-samples-with-gc-and-transcript-length"
class="section level3">
<h3>Correlation of samples with GC and transcript length</h3>
<p><img src="figure/Quality-control.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-1">
Past versions of unnamed-chunk-33-1.png
</button>
</p>
<div id="fig-unnamed-chunk-33-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-33-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Quality-control.Rmd/unnamed-chunk-33-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-2">
Past versions of unnamed-chunk-33-2.png
</button>
</p>
<div id="fig-unnamed-chunk-33-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-33-2.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Quality-control.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-34-1">
Past versions of unnamed-chunk-34-1.png
</button>
</p>
<div id="fig-unnamed-chunk-34-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-34-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="figure" style="text-align: center">
<img src="figure/Quality-control.Rmd/unnamed-chunk-35-1.png" alt="PCA on logCPM from counts post normalization with cqn. As a result of the normlization, it seems that whilst the overall variation within the dataset has reduced, sample 223 is no longer cluster with its condition group on PC1 and PC2" width="672" />
<p class="caption">
PCA on logCPM from counts post normalization with cqn. As a result of
the normlization, it seems that whilst the overall variation within the
dataset has reduced, sample 223 is no longer cluster with its condition
group on PC1 and PC2
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-1">
Past versions of unnamed-chunk-35-1.png
</button>
</p>
<div id="fig-unnamed-chunk-35-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/unawaz1996/NMD-analysis/blob/5196f4120527db07aaf6583bc6850638d325cac7/docs/figure/Quality-control.Rmd/unnamed-chunk-35-1.png" target="_blank">5196f41</a>
</td>
<td>
unawaz1996
</td>
<td>
2023-04-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="removing-sample" class="section level2">
<h2>Removing sample</h2>
<p><img src="figure/Quality-control.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /></p>
<!-- # ```{r} -->
<!-- # ## deseq2 -->
<!-- # ddsTxi <- DESeqDataSet(se, design = ~Group) -->
<!-- # vsd <- vst(ddsTxi) -->
<!-- # DESeq2::plotPCA(vsd, intgroup = "Group") -->
<!-- #  -->
<!-- # ``` -->
<!-- # ```{r} -->
<!-- # library(EnsDb.Hsapiens.v86) -->
<!-- # txdf = transcripts(EnsDb.Hsapiens.v86, return.type="DataFrame") -->
<!-- # tx2gene = as.data.frame(txdf[,c("tx_id","gene_id", "tx_biotype")]) -->
<!-- # salmon.files = ("/home/neuro/Documents/neurogenetics/alignments/Illumina/RNAseq/mouse/LJ_LTK/Salmon/Salmon_phoenix") -->
<!-- # salmon = list.files(salmon.files, pattern = "transcripts$", full.names = TRUE) -->
<!-- # all_files = file.path(salmon, "quant.sf") -->
<!-- # names(all_files) <- paste(c(212:229)) -->
<!-- # txi_genes_human  = tximport(all_files, type="salmon", txOut=FALSE, -->
<!-- #                       countsFromAbundance="scaledTPM", tx2gene = tx2gene, ignoreTxVersion = TRUE, ignoreAfterBar = TRUE) -->
<!-- #  -->
<!-- #  -->
<!-- # txi_genes_human$abundance["ENSG00000169062",] %>% melt() %>% -->
<!-- #    rownames_to_column("name") %>% cbind(md) %>% -->
<!-- #    ggplot(aes(x=as.character(names), y= value)) + -->
<!-- #    geom_bar(stat="identity", color = "black", alpha = 0.9) + -->
<!-- #    facet_grid(~Group, labeller = labeller(Group = group.labs), scales = "free_x") + ylab("tpm") + theme_bw() + xlab("") + -->
<!-- #    scale_fill_manual(labels = c("UPF3A_KD_UPF3B_KD" = "Double KD", "UPF3A_KD" = "UPF3A KD", -->
<!-- #                                  "UPF3B_KD" = "UPF3B KD", "UPF3A_OE" = "UPF3A OE", -->
<!-- #                                  "UPF3A_OE_UPF3B_KD" = "UPF3A OE UPF3B KD")) + -->
<!-- #    theme(legend.position = "none") + ggtitle("UPF3A expression") -->
<!-- # ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.2.2 Patched (2022-11-10 r83330)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.2 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] splines   grid      stats4    tools     stats     graphics  grDevices
 [8] utils     datasets  methods   base     

other attached packages:
 [1] cqn_1.44.0                    quantreg_5.94                
 [3] SparseM_1.81                  preprocessCore_1.60.2        
 [5] nor1mix_1.3-0                 mclust_6.0.0                 
 [7] naniar_1.0.0                  glmpca_0.2.0                 
 [9] broom_1.0.4                   glue_1.6.2                   
[11] ggfortify_0.4.16              stargazer_5.2.3              
[13] IsoformSwitchAnalyzeR_2.01.04 pfamAnalyzeR_0.99.0          
[15] sva_3.46.0                    genefilter_1.80.3            
[17] mgcv_1.8-42                   nlme_3.1-162                 
[19] satuRn_1.6.0                  DEXSeq_1.44.0                
[21] BiocParallel_1.32.6           ggrepel_0.9.3                
[23] pander_0.6.5                  msigdbr_7.5.1                
[25] cowplot_1.1.1                 ngsReports_2.0.3             
[27] patchwork_1.1.2               VennDiagram_1.7.3            
[29] futile.logger_1.4.3           UpSetR_1.4.0                 
[31] fgsea_1.24.0                  GOplot_1.0.2                 
[33] RColorBrewer_1.1-3            gridExtra_2.3                
[35] ggdendro_0.1.23               AnnotationHub_3.6.0          
[37] BiocFileCache_2.6.1           dbplyr_2.3.2                 
[39] openxlsx_4.2.5.2              ggiraph_0.8.7                
[41] wasabi_1.0.1                  sleuth_0.30.1                
[43] DT_0.27                       VennDetail_1.14.0            
[45] msigdb_1.6.0                  GSEABase_1.60.0              
[47] graph_1.76.0                  annotate_1.76.0              
[49] XML_3.99-0.14                 pheatmap_1.0.12              
[51] ggvenn_0.1.10                 MetBrewer_0.2.0              
[53] ggpubr_0.6.0                  venn_1.11                    
[55] viridis_0.6.2                 viridisLite_0.4.1            
[57] tximeta_1.16.1                tximport_1.26.1              
[59] goseq_1.50.0                  geneLenDataBase_1.34.0       
[61] BiasedUrn_2.0.9               org.Mm.eg.db_3.16.0          
[63] EnsDb.Mmusculus.v79_2.99.0    ensembldb_2.22.0             
[65] AnnotationFilter_1.22.0       GenomicFeatures_1.50.4       
[67] AnnotationDbi_1.60.2          biomaRt_2.54.1               
[69] edgeR_3.40.2                  limma_3.54.2                 
[71] DESeq2_1.38.3                 SummarizedExperiment_1.28.0  
[73] Biobase_2.58.0                MatrixGenerics_1.10.0        
[75] matrixStats_0.63.0            GenomicRanges_1.50.2         
[77] GenomeInfoDb_1.34.9           IRanges_2.32.0               
[79] S4Vectors_0.36.2              BiocGenerics_0.44.0          
[81] corrplot_0.92                 lubridate_1.9.2              
[83] forcats_1.0.0                 purrr_1.0.1                  
[85] readr_2.1.4                   tidyverse_2.0.0              
[87] stringr_1.5.0                 tidyr_1.3.0                  
[89] scales_1.2.1                  data.table_1.14.8            
[91] readxl_1.4.2                  tibble_3.2.1                 
[93] magrittr_2.0.3                reshape2_1.4.4               
[95] ggplot2_3.4.2                 dplyr_1.1.1                  
[97] workflowr_1.7.0              

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3                rtracklayer_1.58.0           
  [3] visdat_0.6.0                  bit64_4.0.5                  
  [5] knitr_1.42                    DelayedArray_0.24.0          
  [7] hwriter_1.3.2.1               KEGGREST_1.38.0              
  [9] RCurl_1.98-1.12               generics_0.1.3               
 [11] callr_3.7.3                   lambda.r_1.2.4               
 [13] RSQLite_2.3.1                 bit_4.0.5                    
 [15] tzdb_0.3.0                    xml2_1.3.3                   
 [17] httpuv_1.6.9                  xfun_0.38                    
 [19] hms_1.1.3                     jquerylib_0.1.4              
 [21] babelgene_22.9                evaluate_0.20                
 [23] promises_1.2.0.1              fansi_1.0.4                  
 [25] restfulr_0.0.15               progress_1.2.2               
 [27] DBI_1.1.3                     geneplotter_1.76.0           
 [29] htmlwidgets_1.6.2             ellipsis_0.3.2               
 [31] crosstalk_1.2.0               backports_1.4.1              
 [33] locfdr_1.1-8                  vctrs_0.6.1                  
 [35] abind_1.4-5                   cachem_1.0.7                 
 [37] withr_2.5.0                   BSgenome_1.66.3              
 [39] vroom_1.6.1                   GenomicAlignments_1.34.1     
 [41] prettyunits_1.1.1             lazyeval_0.2.2               
 [43] crayon_1.5.2                  labeling_0.4.2               
 [45] pkgconfig_2.0.3               ProtGenerics_1.30.0          
 [47] rlang_1.1.0                   lifecycle_1.0.3              
 [49] MatrixModels_0.5-1            filelock_1.0.2               
 [51] cellranger_1.1.0              rprojroot_2.0.3              
 [53] Matrix_1.5-3                  carData_3.0-5                
 [55] boot_1.3-28.1                 Rhdf5lib_1.20.0              
 [57] zoo_1.8-11                    whisker_0.4.1                
 [59] processx_3.8.0                png_0.1-8                    
 [61] rjson_0.2.21                  bitops_1.0-7                 
 [63] getPass_0.2-2                 rhdf5filters_1.10.1          
 [65] Biostrings_2.66.0             blob_1.2.4                   
 [67] rstatix_0.7.2                 ggsignif_0.6.4               
 [69] memoise_2.0.1                 plyr_1.8.8                   
 [71] gdata_2.18.0.1                zlibbioc_1.44.0              
 [73] compiler_4.2.2                BiocIO_1.8.0                 
 [75] Rsamtools_2.14.0              cli_3.6.1                    
 [77] XVector_0.38.0                pbapply_1.7-0                
 [79] ps_1.7.4                      formatR_1.14                 
 [81] MASS_7.3-58.3                 tidyselect_1.2.0             
 [83] stringi_1.7.12                highr_0.10                   
 [85] yaml_2.3.7                    locfit_1.5-9.7               
 [87] sass_0.4.5                    fastmatch_1.1-3              
 [89] timechange_0.2.0              parallel_4.2.2               
 [91] rstudioapi_0.14               uuid_1.1-0                   
 [93] git2r_0.31.0                  farver_2.1.1                 
 [95] digest_0.6.31                 BiocManager_1.30.20          
 [97] shiny_1.7.4                   Rcpp_1.0.10                  
 [99] car_3.1-2                     BiocVersion_3.16.0           
[101] later_1.3.0                   httr_1.4.5                   
[103] colorspace_2.1-0              fs_1.6.1                     
[105] statmod_1.5.0                 plotly_4.10.1                
[107] systemfonts_1.0.4             xtable_1.8-4                 
[109] jsonlite_1.8.4                futile.options_1.0.1         
[111] R6_2.5.1                      pillar_1.9.0                 
[113] htmltools_0.5.5               mime_0.12                    
[115] fastmap_1.1.1                 interactiveDisplayBase_1.36.0
[117] codetools_0.2-19              utf8_1.2.3                   
[119] lattice_0.20-45               bslib_0.4.2                  
[121] curl_5.0.0                    gtools_3.9.4                 
[123] zip_2.2.2                     GO.db_3.16.0                 
[125] survival_3.5-5                admisc_0.31                  
[127] rmarkdown_2.21                munsell_0.5.0                
[129] rhdf5_2.42.0                  GenomeInfoDbData_1.2.9       
[131] gtable_0.3.3                 </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
