---
title: "DEG-analysis"
author: "unawaz1996"
date: "2023-03-14"
output:
  html_notebook: default
  workflowr::wflow_html:
    code_folding: hide
editor_options:
  
  chunk_output_type: console
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(
    autodep = TRUE,
    eval = TRUE,
	echo = FALSE,
	warning = FALSE,
	message = FALSE
)
```

```{r load}
source("code/plots.R") 
source("code/functions.R")
source("code/libraries.R")
library(biomaRt)
library(nVennR)
library(eulerr)
library(ggsci)
```



# Introduction 

# Count data 

Gene-level count data was retrieved from salmon files by `tximport` into a DGEList object. 
During this process, genes were removed if 

* They were not considered as detectable (TPM < 0.5 in > 3 samples)
* They did not contain any annotation 

These filtering steps returned gene-level counts for 10,957 genes. 

```{r}
# Quality control of genes 
salmon.files = ("/home/neuro/Documents/NMD_analysis/Analysis/Results/Salmon")
md.files = "data/LTK_Sample Metafile_V3.txt"

# Loading data
txdf = transcripts(EnsDb.Mmusculus.v79, return.type="DataFrame")
tx2gene = as.data.frame(txdf[,c("tx_id","gene_id")])

salmon = list.files(salmon.files, pattern = "transcripts$", full.names = TRUE)
all_files = file.path(salmon, "quant.sf")
names(all_files) <- paste(c(212:229))

txi_genes <- tximport(all_files, type="salmon", txOut=FALSE,
                      countsFromAbundance="scaledTPM", tx2gene = tx2gene, ignoreTxVersion = TRUE, ignoreAfterBar = TRUE)


md = read.table(md.files, header= TRUE) %>%  mutate(files = file.path(salmon, "quant.sf"))
md$Group = factor(md$Group)
md$Sample = as.character(md$Sample)
```


```{r}
group_cols <- hcl.colors(
  n = length(unique(md$Group)), 
  palette = "Zissou 1"
  ) %>%
  setNames(unique(md$Group))
```



```{r fig.height = 4, fig.cap="Expression density plots for all samples before and after filtering, showing logCPM values."}
keep = (rowSums(txi_genes$abundance >= .5 ) >= 3)

exp_den = txi_genes$counts %>% cpm(log=TRUE) %>%
    melt() %>%
    dplyr::filter(is.finite(value)) %>% 
    ggplot(aes(x=value, color = as.character(Var2))) +
    geom_density() +
    ggtitle("Before filtering") +
    labs(x = "logCPM", y = "Proportion of Genes") + 
    theme_bw() +
    theme(legend.position = "none")

exp_den_filt = txi_genes$counts %>%
    cpm(log=TRUE) %>%
    magrittr::extract(keep,) %>% 
    melt() %>%
    dplyr::filter(is.finite(value)) %>% 
    ggplot(aes(x=value, color = as.character(Var2))) +
    geom_density() +
    ggtitle("After filtering") +
    labs(x = "logCPM", y = "Proportion of Genes") + theme_bw() +
    theme(legend.position = "none")

grid.arrange(exp_den,exp_den_filt, ncol=2 )

```


```{r fig.height= 4}
txi_genes_filtered = txi_genes$counts[keep,] %>%  
    as.data.frame() %>% 
    rownames_to_column("geneID") %>%
    mutate(geneName = mapIds(org.Mm.eg.db, keys=.$geneID,
                             column="SYMBOL",keytype="ENSEMBL",
                         multiVals="first")) %>%
    drop_na(geneName) %>% 
    column_to_rownames("geneID") %>% dplyr::select(-geneName)


```

# Analysis

## PCA 

```{r fig.height=4.5}
pca = log2(txi_genes$abundance[keep,]  + 10^-6 ) %>%
    t() %>% 
    prcomp(scale = TRUE)

pcaVars <- percent_format(0.1)(summary(pca)$importance["Proportion of Variance",]) 

gene_pca = pca$x[,1:2] %>% 
    as.data.frame()  %>%
    cbind(md) %>%
    as.tibble() %>%
    ggplot(aes(PC1, PC2, shape = Condition_UPF3B, label = Condition_UPF3B, color=Group)) +
    geom_point(size = 11.5) +
    theme_bw() + scale_shape_manual(values = c(0,16)) + labs(shape="UPF3B status", colour= "Samples") +
    theme(axis.title.y = element_text(size = 12), axis.title.x = element_text(size = 12), 
          axis.text = element_text(size = 10)) +
    xlab(paste0("Principal Component 1 (28.1%)")) + ylab("Principal Component 2 (16.6%)") +
    scale_color_manual(values = c("#FCC98A","#6E74B4", "#E12F8F","#EA2A5F", "#FD9675", "#2F124B"),
                      labels = c("UPF3A_KD_UPF3B_KD" = "Double KD", "UPF3A_KD" = "UPF3A KD", 
                                                                   "UPF3B_KD" = "UPF3B KD", "UPF3A_OE" = "UPF3A OE", 
                                                                   "UPF3A_OE_UPF3B_KD" = "UPF3A OE UPF3B KD")) +
  labs(x = paste0("PC1 (", pcaVars[["PC1"]], ")"),
    y = paste0("PC2 (", pcaVars[["PC2"]], ")"))

#ggsave(filename = "/home/neuro/Documents/NMD_analysis/Analysis/NMD-analysis/output/DEG/Thesis_figures/Gene_PCA.svg", plot = gene_pca, 
#       units = c("in"), width = 6.65, height = 5.60)

```

The principal component analysis was performed using the logCPM values from each sample. Each group appears to be clustering together. UPF3A over-expression is clustering with Controls, whereas the UPF3B KD samples are clustering closely with UPF3A overexpression in UPF3B KD files. 

## Boxplots of gene expression and library size 

```{r fig.height =4}
txi_genes$abundance[keep,] %>% 
    as.data.frame() %>% 
    gather(sample,value) %>% 
    ggplot(aes(x=sample, y = log(value), fill = sample)) + geom_boxplot() +
    ylab("Log(TPM)") + xlab("Samples") +
    theme_bw() + theme(legend.position = "none") + scale_fill_manual(values = hydrangea[1:18])
```

```{r fig.height = 3}
# Library size 

countsSum <- colSums(txi_genes$counts) # library size
countsSum %>% melt() %>% rownames_to_column("Sample") %>% 
    ggplot(aes(x=Sample, y = value)) + geom_bar(stat = "identity") + ylab("Library size") +
    theme_bw()

#max(countsSum) / min(countsSum)

```



# Differential gene expression using limma 

## Model description 

In this analysis, we are interested in seeing how the knockdowns behave differently to controls. Given the quality and clustering of the data, we will perform five pairwise comparisons using `limma` where all conditions (KD/OE) will be compared to controls. 

```{r}
y <- DGEList(txi_genes_filtered)


design <- model.matrix(~0+Group, data = md) %>% 
    set_colnames(gsub(pattern = "Group", replacement ="", x = colnames(.)))

contrasts <- makeContrasts(levels = colnames(design), 
                           UPF3B_vs_Control = UPF3B_KD - Control, 
                           UPF3A_vs_Control = UPF3A_KD - Control, 
                           UPF3A_OE_vs_Control = UPF3A_OE - Control, 
                           DoubleKD_vs_Control = UPF3A_KD_UPF3B_KD - Control, 
                           UPF3A_OE_UPF3B_KD_vs_Control = UPF3A_OE_UPF3B_KD - Control)
```


```{r}
y <- calcNormFactors(y)
v <- voom(y, design)

fit = lmFit(v, design) %>% 
    contrasts.fit(contrasts) %>% 
    eBayes()
```

```{r}
v$E %>% 
  as.data.frame() %>% 
  pivot_longer(
    cols = everything(), names_to = "Sample", values_to = "logCPM"
  ) %>% 
  left_join(md) %>% 
  group_by(Sample) %>% 
  mutate(RLE = logCPM - median(logCPM))%>% 
  ggplot(aes(Sample, RLE, fill = Group)) +
  geom_boxplot(alpha = 0.9) +
  geom_hline(yintercept = 0, linetype = 2) +
  facet_wrap(~Group, scales = "free_x") +
  scale_fill_manual(values = group_cols) +
  labs(
    x = "Sample", fill = "Group"
  )
```



```{r}
## QC plots - deciding log2FC thresholds 
results <- decideTests(fit, p.value = 0.05, adjust.method = "fdr", method= "global")
limma_results = write_fit(fit, results= results, adjust = "fdr") %>% 
    rownames_to_column("ensembl_gene_id") %>% 
    mutate(gene = mapIds(org.Mm.eg.db, keys=ensembl_gene_id,  column="SYMBOL",keytype="ENSEMBL", multiVals="first"), 
           entrez = mapIds(org.Mm.eg.db, keys=ensembl_gene_id,  column="ENTREZID",keytype="ENSEMBL", multiVals="first")) %>% drop_na(entrez)


gene_expression_density = data.frame()
comparisons =c("UPF3B_vs_Control", "UPF3A_vs_Control",  "UPF3A_OE_vs_Control", 
               "DoubleKD_vs_Control", "UPF3A_OE_UPF3B_KD_vs_Control") 

for (c in comparisons) {
  sig_genes = limma_results %>% as.data.frame() %>% 
    dplyr::select(contains("Coef"), contains ("p.value.adj"), contains(c)) %>%
  dplyr::filter(p.value.adj.UPF3B_vs_Control < 0.05) %>%
  dplyr::select(contains("Coef")) %>%
  dplyr::select(contains("UPF3B_vs_Control")) %>% melt()
  gene_expression_density = rbind(gene_expression_density, sig_genes )
  
}


  


#ggplot(aes(x=Coef.UPF3B_vs_Control)) + 
 # geom_histogram(aes(y=..density..), colour="black", fill="white") +  
 # geom_density(alpha=.2, fill="#FF6666") + geom_vline(xintercept =c(log2(1.5), -log2(1.5)),color="red", linetype = "dashed") + 
#  theme_bw() + ylab("Density")



```


```{r}
results <- decideTests(fit, p.value = 0.05, adjust.method = "fdr", method= "global", lfc = log2(1.5))
limma_results = write_fit(fit, results= results, adjust = "fdr") %>% 
    rownames_to_column("ensembl_gene_id") %>% 
    mutate(gene = mapIds(org.Mm.eg.db, keys=ensembl_gene_id,  column="SYMBOL",keytype="ENSEMBL", multiVals="first"), 
           entrez = mapIds(org.Mm.eg.db, keys=ensembl_gene_id,  column="ENTREZID",keytype="ENSEMBL", multiVals="first")) %>% drop_na(entrez)


#save(limma_results, file="/home/neuro/Documents/NMD_analysis/Analysis/Results/DEGs/DEG-limma-results.Rda")
#save(results, fit,y,v,contrasts, design, contrasts,  file = "/home/neuro/Documents/NMD_analysis/Analysis/Results/DEGs/limma-matrices.Rda")

```

```{r}
DEGenes <- list()
DEGenes[["UPF3B_vs_Control"]] <- limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control != 0, abs(Coef.UPF3B_vs_Control) > log2(1.5))
DEGenes[["UPF3A_vs_Control"]] <- limma_results %>%
    dplyr::filter(Res.UPF3A_vs_Control != 0, abs(Coef.UPF3A_vs_Control) > log2(1.5))
DEGenes[["UPF3A_OE_vs_Control"]] <- limma_results %>%
    dplyr::filter(Res.UPF3A_OE_vs_Control != 0, abs(Coef.UPF3A_OE_vs_Control) > log2(1.5)) %>%
    arrange(p.value.UPF3A_OE_vs_Control)
DEGenes[["DoubleKD_vs_Control"]] <- limma_results %>%
    dplyr::filter(Res.DoubleKD_vs_Control != 0, abs(Coef.DoubleKD_vs_Control) > log2(1.5)) %>%
    arrange(p.value.DoubleKD_vs_Control)
DEGenes[["UPF3A_OE_UPF3B_KD_vs_Control"]] <- limma_results %>%
    dplyr::filter(Res.UPF3A_OE_UPF3B_KD_vs_Control != 0, abs(Coef.UPF3A_OE_UPF3B_KD_vs_Control) > log2(1.5)) %>%
    arrange(p.value.UPF3A_OE_UPF3B_KD_vs_Control)

#write.xlsx(DEGenes, "NMD-DEG-limma.xlsx")

#save(DEGenes, file="/home/neuro/Documents/NMD_analysis/Analysis/Results/DEGs/DEG-list.Rda")

```

## Total number of DEGs 


```{r}
summary(results)
```

```{r fig.height=4}
summary_DEG = summary(results) %>% melt() %>% 
    dplyr::filter(Var1 != "NotSig") 

summary_DEG$Var1 = factor(summary_DEG$Var1, levels = c("Up", "Down"))
summary_DEG$Var2 = factor(summary_DEG$Var2, levels = c("UPF3B_vs_Control", 
                                                       "UPF3A_vs_Control", 
                                                       "DoubleKD_vs_Control", 
                                                       "UPF3A_OE_vs_Control", 
                                                       "UPF3A_OE_UPF3B_KD_vs_Contro"))
summary_DEG = summary_DEG %>%  
    ggplot(aes(x=Var2, y = value, fill=Var1)) + 
    geom_bar(stat = "identity",   colour = "white", width = .75) + 
    theme_bw() + 
    theme(axis.text.x = element_text(angle = 70,  hjust=1, face = "bold", size=10), 
          legend.position = "bottom", 
          legend.title= element_blank(), axis.title.y = element_text(size = 12), 
          axis.text.y = element_text(size = 12, face = "bold")) + 
    scale_fill_manual(values = c("#EA2A5F", "#2F124B")) +
    ylab("# Differentially expressed genes") + xlab("") + scale_x_discrete(labels=c("DoubleKD_vs_Control" = "Double KD", "UPF3A_vs_Control" = "UPF3A KD", 
                                                            "UPF3B_vs_Control" = "UPF3B KD", "UPF3A_OE_vs_Control" = "UPF3A OE", 
                                                            "UPF3A_OE_UPF3B_KD_vs_Control" = "UPF3A OE UPF3B KD"))


summary_DEG
#ggsave(filename = "/home/neuro/Documents/NMD_analysis/Analysis/NMD-analysis/output/DEG/Thesis_figures/total_DEG.svg", plot = #summary_DEG, 
#       units = c("in"), height = 5.31, width = 4.48)

```


```{r}
ratios = data.frame()
for (c in comparisons){
  temp = summary(results) %>% 
    melt() %>% 
    dplyr::filter(Var2 == c, Var1 != "NotSig") 
  ratio= temp$value[2]/temp$value[1]
  ratio = data.frame(Comparisons = c, 
           ratio = ratio)
  ratios = rbind(ratios, ratio)
  }

ratios$Comparisons = factor(ratios$Comparisons, levels = c("UPF3B_vs_Control", 
                                                           "UPF3A_vs_Control", 
                                                           "DoubleKD_vs_Control", 
                                                           "UPF3A_OE_vs_Control", 
                                                           "UPF3A_OE_UPF3B_KD_vs_Control"))

ratios_fig =ratios  %>%
  ggplot(aes(x =Comparisons, y = ratio, fill = Comparisons)) + 
  geom_bar(stat = "identity",   colour = "white", width = .75) + 
    theme_classic() + 
    theme(axis.text.x = element_text(angle = 70,  hjust=1, face = "bold"), 
          legend.position = "none", 
          legend.title= element_blank(), axis.title.y = element_text(size = 12), 
          axis.text.y = element_text(size = 14, face = "bold")) + 
    scale_fill_manual(values = c("#2F124B","#EA2A5F","#6E74B4","#E12F8F", "#FD9675")) +
    ylab("ratio up/down") + xlab("") + scale_x_discrete(labels=c("DoubleKD_vs_Control" = "Double KD", "UPF3A_vs_Control" = "UPF3A KD", 
                                                            "UPF3B_vs_Control" = "UPF3B KD", "UPF3A_OE_vs_Control" = "UPF3A OE", 
                                                            "UPF3A_OE_UPF3B_KD_vs_Control" = "UPF3A OE UPF3B KD"))

ratios
#ggsave(filename = "/home/neuro/Documents/NMD_analysis/Analysis/NMD-analysis/output/DEG/Thesis_figures/ratio_DEG.svg", plot = #ratios_fig, 
#       units = c("in"), height = 3.30, width = 4.48)

```

After performing pairwise differential gene expression analysis, we can see that majority of all significantly expressed genes in all contrasts are up-regulated. Overall, we find that a total of 395 genes are differentially expressed between UPF3B KD and controls. 388 genes are differentially expressed between UPF3A KD vs Controls.


## Overlapping DEGs 


```{r}
## Venn diagram and table representing each venn diagram with gene name and its results 
ven <- venndetail(list("UPF3B KD" = DEGenes$UPF3B_vs_Control$gene, 
                       "UPF3A KD" = DEGenes$UPF3A_vs_Control$gene,
                    "Double KD" = DEGenes$DoubleKD_vs_Control$gene,
                    "UPF3A OE" = DEGenes$UPF3A_OE_vs_Control$gene, 
                  "UPF3A OE, UPF3B KD" = DEGenes$UPF3A_OE_UPF3B_KD_vs_Control$gene))
```



```{r}
x=list("UPF3B KD" = DEGenes$UPF3B_vs_Control$ensembl_gene_id, 
                       "UPF3A KD" = DEGenes$UPF3A_vs_Control$ensembl_gene_id,
                    "Double KD" = DEGenes$DoubleKD_vs_Control$ensembl_gene_id,
                    "UPF3A OE" = DEGenes$UPF3A_OE_vs_Control$ensembl_gene_id, 
                  "UPF3A OE, UPF3B KD" = DEGenes$UPF3A_OE_UPF3B_KD_vs_Control$ensembl_gene_id)

upset(fromList(x))
```




## UPF3B vs Controls 


```{r fig.height=5}
DEColours <- c("-1" = "#2F124B","1" = "#EA2A5F",  "0" = "#D3D5E3")
UPF3B_volc = limma_results %>%
    ggplot(aes(x = Coef.UPF3B_vs_Control, 
               y = -log10(p.value.adj.UPF3B_vs_Control), 
               colour = as.factor(as.character(Res.UPF3B_vs_Control)))) +
    geom_point(alpha = 0.8, size =1.5) +
    scale_colour_manual(values = DEColours) + theme_classic() + 
    theme(legend.position = "none", 
          axis.title.y = element_text(size = 12)) +
    geom_hline(yintercept = -log10(0.05), color = "grey60", size = 0.5, lty = "dashed") +
    #geom_hline(yintercept = -log10(limma_results %>% 
     #                                  dplyr::filter(Res.UPF3B_vs_Control != 0) %>% 
      #                                 use_series(p.value.adj.UPF3B_vs_Control) %>% max), 
       #        color = "black") +
    labs(x = "log2 Fold Change", y = "-log10 adj p-value") +
    ggtitle("UPF3B KD vs. Control") + 
  geom_vline(xintercept = c(-log2(1.5), log2(1.5)), size = 0.5, lty = "dashed", color = "grey60") + xlim(-8.5, 8.5) 


#my_gg = UPF3B_volc + geom_point_interactive(aes(tooltip = gene, data_id = gene), 
#    size = 1, hover_nearest = TRUE)
#girafe(ggobj = my_gg)
```

```{r}
UPF3B_volc
```

```{r}
## volcano plot
#ggsave(filename = "/home/neuro/Documents/NMD_analysis/Analysis/NMD-analysis/output/DEG/Thesis_figures/UPF3B_KD_volcano.svg", plot = volc, 
#       units = c("in"), width = 5.97, height = 3.63)
```


```{r}
DEGenes$UPF3B_vs_Control %>% 
    dplyr::select(ensembl_gene_id,gene,entrez, contains("UPF3B_vs")) %>% 
    dplyr::select(-contains("Res")) %>% 
    mutate(genename = mapIds(org.Mm.eg.db, keys=ensembl_gene_id,  column="GENENAME",keytype="ENSEMBL", multiVals="first")) %>%     DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'), caption = "Genes significantly expressed in UPF3B KD relative to Controls", style = "auto", width = NULL, height = NULL, elementId = NULL,
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```

```{r}
limma_results %>% 
    dplyr::select(ensembl_gene_id,gene,entrez, contains("UPF3B_vs")) %>% 
    dplyr::select(-contains("Res")) %>% 
    mutate(genename = mapIds(org.Mm.eg.db, keys=ensembl_gene_id,  column="GENENAME",keytype="ENSEMBL", multiVals="first")) %>%    
  DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'), caption = "Genes significantly expressed in UPF3B KD relative to Controls", style = "auto", width = NULL, height = NULL, elementId = NULL,
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


```

Given that previous studies have confirmed UPF3B KD results in a decrease of efficieny of NMD resulting in an increase of upregulated genes, we are interested in seeing how the significantly expressed up-regulated genes behave in other conditions compared to controls.


In order to do this, we will create a heatmap visualisaiton. 


```{r fig.height=30}

rg <- limma_results %>% 
     dplyr::filter(Res.UPF3B_vs_Control == 1) %>%
    dplyr::select(contains("Coef"), gene)  %>%
    column_to_rownames("gene") %>% max(abs(.));

limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control == 1) %>%
    dplyr::select(contains("Coef"), gene)  %>%
    column_to_rownames("gene") %>%
    pheatmap(,
        scale="none", 
        cluster_cols = FALSE,
        breaks = seq(-rg, rg, length.out = 100),
        clustering_distance_rows = "euclidean", 
        border_color = "white", 
        treeheight_row = 0,
        fontsize = 6,
        color = colorRampPalette(c("#2F124B", "#6E74B4","#F9F9F9", "#FD9675", "#EA2A5F"))(100),
        labels_row = row.names(.), 
        cellwidth = 8
    )

```

From the heatmap above, we can notice a few trends that upregulated genes in UPF3B KD have in other conditions. 

1) Inverted expression in UPF3A OE genes

2) Inverted expression in UPF3A KD 


These genes can be explored here: 

```{r}
limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control == 1) %>%
    dplyr::select(gene, contains("Coef"))  %>%
   # column_to_rownames("gene") %>%  
    DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
    
```

#### UPF2 levels 

```{r}
limma_results %>% 
  dplyr::filter(ensembl_gene_id == "ENSMUSG00000038398") %>% 
  column_to_rownames("ensembl_gene_id") %>% 
  dplyr::select(contains("Coef.")) %>%
  melt() %>% 
  ggplot(aes(x=variable, y=value)) + geom_bar(stat="identity")


```

### Overlap with other significant genes 

We are now also interested in seeing how all genes differentially expressed in UPF3B KD overlap with significant genes from other conditions.


#### Comparison with UPF3A KD significant genes 


```{r}
#venn(list("UPF3B KD" = DEGenes$UPF3B_vs_Control$ensembl_gene_id, 
 #                      "UPF3A KD" = DEGenes$UPF3A_vs_Control$ensembl_gene_id))
```

```{r}
x = list( "UPF3B KD" = DEGenes$UPF3B_vs_Control$ensembl_gene_id, 
    "UPF3A KD" = DEGenes$UPF3A_vs_Control$ensembl_gene_id, 
    "Double KD" = DEGenes$DoubleKD_vs_Control$ensembl_gene_id,
    "UPF3A OE" = DEGenes$UPF3A_OE_vs_Control$ensembl_gene_id, 
    "UPF3A OE UPF3B KD" = DEGenes$UPF3A_OE_UPF3B_KD_vs_Control$ensembl_gene_id)

venn_upf3 = ggvenn(x[c(1,2)], fill_color = c("#2F124B", "#EA2A5F"),
       set_name_color = "black", text_color = "white", text_size = 6, 
       stroke_size = 0.5, fill_alpha = 0.6) 

venn_upf3
#ggsave(filename = "/home/neuro/Documents/NMD_analysis/Analysis/NMD-analysis/output/DEG/Thesis_figures/UPF3_venn.svg", plot = #venn_upf3 , 
#       units = c("in"), height = 3.30, width = 4.48)

```


When we compare all significant genes in UPF3B KD with UPF3A KD, we can see that there is an overlap of 144 genes. 

```{r}
cor_upf3KD = limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control != 0 & Res.UPF3A_vs_Control !=0) %>% 
    ggscatter(., x= "Coef.UPF3B_vs_Control", y="Coef.UPF3A_vs_Control", 
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw() + ylab("UPF3A KD (log2FoldChange)") + xlab("UPF3B KD (log2FoldChange)") +
  geom_hline(yintercept = 0, size = 0.5,lty = "dashed", color = "grey60") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60")

ggsave(filename = "/home/neuro/Documents/NMD_analysis/Analysis/NMD-analysis/output/DEG/Thesis_figures/UPF3_corr.svg", plot = cor_upf3KD, 
       units = c("in"), height = 3.80, width = 4.73)


```


When we compare the direction of gene expression in these 144 genes, we can see that there is a positive correlation (r = 0.56). 

We are now interested in seeing which genes are present in each of the quadrants: 

1) Quadrant 1:  Definied as x < 0, y > 0 where is significant genes belogning to UPF3A KD group show upregulation and genes belonging to UPF3B KD group show downregulation. A total of 14 genes belong to this group:

```{r}
limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control == -1 & Res.UPF3A_vs_Control == 1) %>%  
    dplyr::select(ensembl_gene_id,gene,entrez, contains(c("UPF3B_vs", "UPF3A_vs"))) %>% 
    dplyr::arrange(Res.UPF3B_vs_Control) %>%
     DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
                   caption ="Overlapping genes between UPF3B KD and UPF3A KD relative to controls in Quadrant 1 (Upregulated in UPF3A KD, downregulated in UPF3B KD)",
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```

2) Quadrant 2: Defined as x > 0, y > 0 where significant genes from both conditions show upregulation. 114 genes belong to this group. 

```{r}
limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control == 1 & Res.UPF3A_vs_Control ==1) %>% 
    dplyr::select(ensembl_gene_id,gene,entrez, contains(c("UPF3B_vs", "UPF3A_vs"))) %>% 
    dplyr::arrange(Res.UPF3B_vs_Control) %>%
    DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
                  caption ="Overlapping genes between UPF3B KD and UPF3A KD relative to controls in Quadrant 2 (Upregulated in both conditions)",
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```

3)  Quadrant 3 is defined as x < 0, y < 0 where genes in both groups show downregulation. 
12 genes are in this quadrant:


```{r}
limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control == -1 & Res.UPF3A_vs_Control == -1) %>%
    dplyr::select(ensembl_gene_id,gene,entrez, contains(c("UPF3B_vs", "UPF3A_vs"))) %>% 
    dplyr::arrange(Res.UPF3B_vs_Control) %>%
    DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
                  caption ="Overlapping genes between UPF3B KD and UPF3A KD relative to controls in Quadrant 3 (Downregulated in both conditions)",
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```

4) Quadrant 4 is defined as x > 0, y < 0, where significant genes belonging to UPF3B KD group are upregulated and genes belonging to UPF3A KD group are downregulated. Only 4 genes belong in this group. 


```{r}
limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control == 1 & Res.UPF3A_vs_Control ==-1) %>% 
    dplyr::select(ensembl_gene_id,gene,entrez, contains(c("UPF3B_vs", "UPF3A_vs"))) %>% 
    dplyr::arrange(Res.UPF3B_vs_Control) %>%
     DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
                   caption ="Overlapping genes between UPF3B KD and UPF3A KD relative to controls in Quadrant 4 (Upregulated in UPF3B KD, downregulated in UPF3A KD)",
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```


We are also interested in seeing how the 144 overlapping genes between UPF3A and UPF3B KD behave in other conditions. To do this, we will use a heatmap visualisation again.

```{r}
limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control != 0 & Res.UPF3A_vs_Control !=0) %>% 
    dplyr::select(ensembl_gene_id,gene, contains(c("Coef."))) %>%
 DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
               caption ="Overlapping genes between UPF3B KD and UPF3A KD (padj < 0.05; absolute log2FoldChange > log2(1.5)) in other comparisons)",
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))

```




```{r fig.height =20, out.extra='angle=90'}

rg <- limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control != 0 & Res.UPF3A_vs_Control !=0) %>%
    dplyr::select(contains("Coef"), gene)  %>%
    column_to_rownames("gene") %>% max(abs(.));

limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control != 0 & Res.UPF3A_vs_Control !=0) %>%
    dplyr::select(contains("Coef"), gene)  %>%
    column_to_rownames("gene") %>%
    pheatmap(,
       # scale="row", 
        cluster_cols = FALSE,
        breaks = seq(-rg, rg, length.out = 100),
        clustering_distance_rows = "euclidean", 
        border_color = "white", 
        treeheight_row = 0,
        cutree_rows = 2,
        fontsize = 6,
        color = colorRampPalette(c("#2F124B", "#6E74B4","#F9F9F9", "#FD9675", "#EA2A5F"))(100),
        labels_row = row.names(.),
        cellheight = 10, 
        cellwidth = 12
    )


```

```{r}

library(ComplexHeatmap)
library(circlize)

```

```{r}
heatmap = limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control != 0 & Res.UPF3A_vs_Control !=0) %>%
    dplyr::select(contains("Coef"), gene)  %>%
    column_to_rownames("gene")

split = sample(letters[1:5], 144, replace = TRUE)
split = factor(split, levels = letters[1:5])

col_fun1 = colorRamp2(c(-6, 0, 6), c("#2F124B", "#F9F9F9","#EA2A5F"))
color = colorRampPalette(c("#2F124B", "#6E74B4","#F9F9F9", "#FD9675", "#EA2A5F"))(100)

circos.par(gap.after = c(2, 2, 2, 2, 10))
circos.heatmap(heatmap ,col = col_fun1, split = split, rownames.side = "outside")
circos.track(track.index = get.current.track.index(), panel.fun = function(x, y) {
    if(CELL_META$sector.numeric.index == 5) { # the last sector
        cn = colnames(heatmap)
        n = length(cn)
        circos.text(rep(CELL_META$cell.xlim[2], n) + convert_x(1, "mm"), 
            1:n - 0.5, cn, 
            cex = 0.5, adj = c(0, 0.5,0), facing = "inside")
    }
}, bg.border = NA)


```


We also want to see the genes that are unique to UPF3A KD and unique to UPF3B KD and how their expression is

```{r}
upf3b_uniq = limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control != 0 & Res.UPF3A_vs_Control ==0) %>% 
    ggscatter(., x= "Coef.UPF3B_vs_Control", y="Coef.UPF3A_vs_Control", 
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw() + ylab("UPF3A KD (log2FoldChange)") + xlab("UPF3B KD (log2FoldChange)") +
  geom_hline(yintercept = 0, size = 0.5,lty = "dashed", color = "grey60") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60")


ggsave(filename = "/home/neuro/Documents/NMD_analysis/Analysis/NMD-analysis/output/DEG/Thesis_figures/UPF3B_corr.svg", plot = upf3b_uniq, 
       units = c("in"), height = 3.80, width = 4.73)
```

```{r}
upf3a_uniq = limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control == 0 & Res.UPF3A_vs_Control !=0) %>% 
    ggscatter(., x= "Coef.UPF3B_vs_Control", y="Coef.UPF3A_vs_Control", 
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw() + ylab("UPF3A KD (log2FoldChange)") + xlab("UPF3B KD (log2FoldChange)") +
  geom_hline(yintercept = 0, size = 0.5,lty = "dashed", color = "grey60") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60")

ggsave(filename = "/home/neuro/Documents/NMD_analysis/Analysis/NMD-analysis/output/DEG/Thesis_figures/UPF3A_corr.svg", plot = upf3a_uniq, 
       units = c("in"), height = 3.80, width = 4.73)

```

#### UPF3B vs Double KD 


```{r}
ggvenn(x[c(1,3)])
```

A total of 316 genes overlap betweeen the Double KDs and UPF3B KD. When we look at the correlation, we see that there is a strong positive correlation. 

```{r}
limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control != 0 & Res.DoubleKD_vs_Control !=0) %>% 
    ggscatter(., x= "Coef.UPF3B_vs_Control", y="Coef.DoubleKD_vs_Control", 
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw() 
```


1) Quadrant 1:  Defined as x < 0, y > 0 where is significant genes belonging to Double KD group show upregulation and genes belonging to UPF3B KD group show downregulation. A total of 2 genes belong to this group:

```{r}
limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control == -1 & Res.DoubleKD_vs_Control == 1) %>% 
    dplyr::select(ensembl_gene_id,gene,entrez, contains(c("UPF3B_vs", "Double"))) %>% 
    dplyr::arrange(Res.UPF3B_vs_Control) %>%
     DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```
2) Quadrant 2: Defined as x > 0, y > 0 where significant genes from both conditions show upregulation. 255 genes belong to this group. 

```{r}
limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control == 1 & Res.DoubleKD_vs_Control ==1) %>% 
    dplyr::select(ensembl_gene_id,gene,entrez, contains(c("UPF3B_vs", "DoubleKD"))) %>% 
    dplyr::arrange(Res.UPF3B_vs_Control) %>%
   DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```


3)  Quadrant 3 is defined as x < 0, y < 0 where genes in both groups show downregulation. 
58 genes are in this quadrant:

```{r}
limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control == -1 & Res.DoubleKD_vs_Control == -1) %>%
    dplyr::select(ensembl_gene_id,gene,entrez, contains(c("UPF3B_vs", "DoubleKD"))) %>% 
    dplyr::arrange(Res.UPF3B_vs_Control) %>%
     DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))

```

### UPF3B vs UPF3A OE, UPF3B KD 


```{r}
ggvenn(x[c(1,5)])
```

```{r}
limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control != 0 & Res.UPF3A_OE_UPF3B_KD_vs_Control !=0) %>% 
    ggscatter(., x= "Coef.UPF3B_vs_Control", y="Coef.UPF3A_OE_UPF3B_KD_vs_Control", 
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw() 

```


When we compare UPF3B KD with UPF3A OE in UPF3B KD cell lines, we can see there is a strong positive correlation. This correlation is very likely induced by the UPF3B KD cell line itself, and not the impact of the UPF3A OE.  For interest, here are the gene in Quadrant 2 and Quadrant 3


1) Quadrant 2: Defined as x > 0, y > 0 where significant genes from both conditions show upregulation. 183 genes belong to this group. 

```{r}
limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control == 1 & Res.UPF3A_OE_UPF3B_KD_vs_Control ==1) %>% 
    dplyr::select(ensembl_gene_id,gene,entrez, contains(c("UPF3B_vs", "DoubleKD"))) %>% 
    dplyr::arrange(Res.UPF3B_vs_Control) %>%
   DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```


2)  Quadrant 3 is defined as x < 0, y < 0 where genes in both groups show downregulation. 
73 genes are in this quadrant:

```{r}
limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control == -1 & Res.UPF3A_OE_UPF3B_KD_vs_Control== -1) %>% 
    dplyr::select(ensembl_gene_id,gene,entrez, contains(c("UPF3B_vs", "DoubleKD"))) %>% 
    dplyr::arrange(Res.UPF3B_vs_Control) %>%
     DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))

```


Following up from this, it would be good to see how the genes that were specific to the UPF3B KD (162 genes) and UPF3A OE, UPF3B KD (85) behave relative to the other condition.


```{r}
limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control != 0 &  Res.UPF3A_OE_UPF3B_KD_vs_Control == 0 ) %>% 
    ggscatter(., x= "Coef.UPF3B_vs_Control", y="Coef.UPF3A_OE_UPF3B_KD_vs_Control", 
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw() 
```

```{r}
limma_results %>% 
    dplyr::filter(Res.UPF3B_vs_Control ==0 &  Res.UPF3A_OE_UPF3B_KD_vs_Control != 0 ) %>% 
    ggscatter(., x= "Coef.UPF3B_vs_Control", y="Coef.UPF3A_OE_UPF3B_KD_vs_Control", 
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw() 
```

Overall the above plots show that there is a strong positive correlation.


## UPF3A 

```{r UPF3A-volc,fig.height = 5}
UPF3A_volc = limma_results %>%
    ggplot(aes(x = Coef.UPF3A_vs_Control, 
               y = -log10(p.value.adj.UPF3A_vs_Control), 
               colour = as.factor(as.character(Res.UPF3A_vs_Control)))) +
    geom_point(alpha = 0.8, size =0.8) +
    scale_colour_manual(values = DEColours) + theme_classic() + 
    theme(legend.position = "none") +
    geom_hline(yintercept = -log10(0.05), color = "grey60", size = 0.5, lty = "dashed") +
    labs(x = "log2 Fold Change", y = "-log10 adj p-value") +
    ggtitle("UPF3A KD vs. Control") + 
  geom_vline(xintercept = c(-log2(1.5), log2(1.5)), size = 0.5, lty = "dashed", color = "grey60") + xlim(-8.5, 8.5) 


#my_gg = volc + geom_point_interactive(aes(tooltip = gene, data_id = gene), 
#    size = 1, hover_nearest = TRUE)
#girafe(ggobj = my_gg)
```

```{r}
UPF3A_volc
```


```{r}
limma_results %>% 
  dplyr::filter(Res.UPF3A_vs_Control != 0) %>%
    #dplyr::select(ensembl_gene_id,gene,entrez, contains("UPF3A_vs")) %>% 
    mutate(genename = mapIds(org.Mm.eg.db, keys=ensembl_gene_id,  column="GENENAME",keytype="ENSEMBL", multiVals="first")) %>%     DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
                                                                                                                                                 caption = "Genes significantly expression in UPF3A KDs relative to Controls",
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```





### UPF3A vs Double KDs 

```{r}
ggvenn(x[c(2,3)])


```


```{r}
limma_results %>% 
    dplyr::filter(Res.UPF3A_vs_Control != 0 & Res.DoubleKD_vs_Control !=0) %>% 
    ggscatter(., x= "Coef.UPF3B_vs_Control", y="Coef.UPF3A_OE_UPF3B_KD_vs_Control", 
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw() 
```


### UPF3A OE 

```{r}
DEGenes$UPF3A_OE_vs_Control %>% 
    dplyr::select(ensembl_gene_id,gene,entrez, contains("UPF3B_vs")) %>% 
    dplyr::select(-contains("Res")) %>% 
    mutate(genename = mapIds(org.Mm.eg.db, keys=ensembl_gene_id,  column="GENENAME",keytype="ENSEMBL", multiVals="first")) %>%     DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'), caption = "Genes significantly expressed in UPF3B KD relative to Controls", style = "auto", width = NULL, height = NULL, elementId = NULL,
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```

```{r fig.height=30, fig.cap ="Heatmap of log2FCof comparisons based on genes significant in UPF3A OE compared to controls"}

rg <- limma_results %>% 
     dplyr::filter(Res.UPF3A_OE_vs_Control != 0) %>%
    dplyr::select(contains("Coef"), gene)  %>%
    column_to_rownames("gene") %>% max(abs(.));

limma_results %>% 
    dplyr::filter(Res.UPF3A_OE_vs_Control != 0) %>%
    dplyr::select(contains("Coef"), gene)  %>%
    column_to_rownames("gene") %>%
    pheatmap(,
        scale="none", 
        cluster_cols = FALSE,
        breaks = seq(-rg, rg, length.out = 100),
        clustering_distance_rows = "euclidean", 
        border_color = "white", 
        treeheight_row = 0,
        fontsize = 6,
        color = colorRampPalette(c("#2F124B", "#6E74B4","#F9F9F9", "#FD9675", "#EA2A5F"))(100),
        labels_row = row.names(.), 
        cellwidth = 8
    )

```

## UPF3 dKDs 

```{r}
dKds_volc = limma_results %>%
    ggplot(aes(x = Coef.DoubleKD_vs_Control, 
               y = -log10(p.value.adj.DoubleKD_vs_Control), 
               colour = as.factor(as.character(Res.DoubleKD_vs_Control)))) +
    geom_point(alpha = 0.8, size =1.5) +
    scale_colour_manual(values = DEColours) + theme_classic() + 
    theme(legend.position = "none", 
          axis.title.y = element_text(size = 12)) +
    geom_hline(yintercept = -log10(0.05), color = "grey60", size = 0.5, lty = "dashed") +
    #geom_hline(yintercept = -log10(limma_results %>% 
     #                                  dplyr::filter(Res.UPF3B_vs_Control != 0) %>% 
      #                                 use_series(p.value.adj.UPF3B_vs_Control) %>% max), 
       #        color = "black") +
    labs(x = "log2 Fold Change", y = "-log10 adj p-value") +
    ggtitle("UPF3B KD vs. Control") + 
  geom_vline(xintercept = c(-log2(1.5), log2(1.5)), size = 0.5, lty = "dashed", color = "grey60") + xlim(-8.5, 8.5) 

```

```{r}
dKds_volc
```

## UPF3A OE

```{r}
upf3a_oe_volc = limma_results %>%
    ggplot(aes(x = Coef.UPF3A_OE_vs_Control, 
               y = -log10(p.value.adj.UPF3A_OE_vs_Control), 
               colour = as.factor(as.character(Res.UPF3A_OE_vs_Control)))) +
    geom_point(alpha = 0.8, size =1.5) +
    scale_colour_manual(values = DEColours) + theme_classic() + 
    theme(legend.position = "none", 
          axis.title.y = element_text(size = 12)) +
    geom_hline(yintercept = -log10(0.05), color = "grey60", size = 0.5, lty = "dashed") +
    #geom_hline(yintercept = -log10(limma_results %>% 
     #                                  dplyr::filter(Res.UPF3B_vs_Control != 0) %>% 
      #                                 use_series(p.value.adj.UPF3B_vs_Control) %>% max), 
       #        color = "black") +
    labs(x = "log2 Fold Change", y = "-log10 adj p-value") +
    ggtitle("UPF3B KD vs. Control") + 
  geom_vline(xintercept = c(-log2(1.5), log2(1.5)), size = 0.5, lty = "dashed", color = "grey60") + xlim(-8.5, 8.5) 

```

```{r}
upf3a_oe_volc
```


## UPF3A OE in UPF3B KD 

```{r}
upf3a_oe_kd_volc = limma_results %>%
    ggplot(aes(x = Coef.UPF3A_OE_UPF3B_KD_vs_Control, 
               y = -log10(p.value.adj.UPF3A_OE_UPF3B_KD_vs_Control), 
               colour = as.factor(as.character(Res.UPF3A_OE_UPF3B_KD_vs_Control)))) +
    geom_point(alpha = 0.8, size =1.5) +
    scale_colour_manual(values = DEColours) + theme_classic() + 
    theme(legend.position = "none", 
          axis.title.y = element_text(size = 12)) +
    geom_hline(yintercept = -log10(0.05), color = "grey60", size = 0.5, lty = "dashed") +
    #geom_hline(yintercept = -log10(limma_results %>% 
     #                                  dplyr::filter(Res.UPF3B_vs_Control != 0) %>% 
      #                                 use_series(p.value.adj.UPF3B_vs_Control) %>% max), 
       #        color = "black") +
    labs(x = "log2 Fold Change", y = "-log10 adj p-value") +
    ggtitle("UPF3B KD vs. Control") + 
  geom_vline(xintercept = c(-log2(1.5), log2(1.5)), size = 0.5, lty = "dashed", color = "grey60") + xlim(-8.5, 8.5) 


```

```{r}
upf3a_oe_kd_volc
```

# Siginifcant genes across all conditions with respect to other conditions 

```{r fig.height = 10, fig.width=10, fig.cap="*Heatmap of log fold changes across different comparisons for genes that were significant in at least 1 pairwise comparison aganst controls*"}

rg <- limma_results %>% 
     dplyr::filter(Res.UPF3B_vs_Control != 0 | Res.UPF3A_vs_Control != 0 |
                Res.DoubleKD_vs_Control !=0 | Res.UPF3A_OE_vs_Control != 0 |
                  Res.UPF3A_OE_UPF3B_KD_vs_Control != 0) %>%
    dplyr::select(contains("Coef"), gene)  %>%
    column_to_rownames("gene") %>% max(abs(.));

limma_results %>%
  dplyr::filter(Res.UPF3B_vs_Control != 0 | Res.UPF3A_vs_Control != 0 |
                Res.DoubleKD_vs_Control !=0 | Res.UPF3A_OE_vs_Control != 0 |
                  Res.UPF3A_OE_UPF3B_KD_vs_Control != 0 ) %>% 
  dplyr::select(contains("Coef"), gene)  %>%
    column_to_rownames("gene") %>%
    pheatmap(,
       # scale="row", 
        cluster_cols = TRUE,
        breaks = seq(-rg, rg, length.out = 100),
        clustering_distance_rows = "euclidean", 
        border_color = "white", 
        treeheight_row = 0,
        cutree_rows = 2,
        fontsize = 6,
        color = colorRampPalette(c("#2F124B", "#6E74B4","#F9F9F9", "#FD9675", "#EA2A5F"))(100),
        labels_row = row.names(.)
    )
  
```
# Data export
```{r}
save(DEGenes, file=here::here("output/DEG-list.Rda"))
save(limma_results, file=here::here("output/DEG-limma-results.Rda"))
save(results, fit,y,v,contrasts, design, contrasts,  file = here::here("output/limma-matrices.Rda"))
```
