---
title: "Differential-transcript-expression"
author: "unawaz1996"
date: "2023-03-26"
output:
  html_notebook: default
  workflowr::wflow_html:
    code_folding: hide
editor_options:
  
  chunk_output_type: console
---

# Introduction


```{r setup, include=FALSE}
knitr::opts_chunk$set(
    autodep = TRUE,
    eval = TRUE,
	echo = FALSE,
	warning = FALSE,
	message = FALSE
)
```

```{r libraries}
source("code/libraries.R")
source("code/functions.R")
library(fishpond)
library(tximeta)
library(GeneOverlap)
library(ggside)
library(tidyquant)
library(stargazer)
library(ggfortify)
library(glue)
library(cowplot)
library(broom)
library(glmpca)
library(naniar)
library(bigPint)
```

## Set-up 

```{r}
txdf = transcripts(EnsDb.Mmusculus.v79, return.type="DataFrame")
tx2gene = as.data.frame(txdf[,c("tx_id","gene_id")])
```

```{r}
load("output/limma-matrices.Rda")
load("output/DEG-limma-results.Rda")
load("output/DEG-list.Rda")

```

```{r}
md.files = "data/LTK_Sample Metafile_V3.txt"
salmon.files = ("/home/neuro/Documents/NMD_analysis/Analysis/Results/Salmon")
salmon = list.files(salmon.files, pattern = "transcripts$", full.names = TRUE)
names(salmon) <- paste(c(212:229))
md = read.table(md.files, header= TRUE) %>%  mutate(files = file.path(salmon, "quant.sf"))
comparisons = unique(md$Group)[-1]
colnames(md)[1] = c("names")
md$names = as.character(md$names)
md$Group <- factor(md$Group, 
                            levels=c("Control","UPF3B_KD", "UPF3A_KD", "UPF3A_OE", 
                                     "UPF3A_KD_UPF3B_KD", "UPF3A_OE_UPF3B_KD"))
```

```{r}
group_cols <- hcl.colors(
  n = length(unique(md$Group)), 
  palette = "Zissou 1"
  ) %>%
  setNames(unique(md$Group))
```


Transcript-level counts were retrieved using the `catchSalmon()` function from the edgeR package. 
Once retrieved, the counts underwent rigourous QC. 

# Removal of outlier sample 

According to the quality control analysis, it was revealed that one of the samples was not clustering with its condition group. Investigation into GC content, transcript length and library size revealed no contribution of these factors. To ensure we are getting results that are not skewed we will remove the sample from the transcript level analysis. 


```{r include=FALSE}
catch <- catchSalmon(salmon)
colnames(catch$counts) =  paste(c(212:229))

cts.scaled = catch$counts/catch$annotation$Overdispersion

#md_filt = md %>% dplyr::filter(names != "223")


dge <- DGEList(counts = cts.scaled, 
               genes=catch$annotation, 
               samples = md,
               group = md$Group)

keep <- filterByExpr(dge)
dge.scaled.filtr <- dge[keep, , keep.lib.sizes = FALSE]
dge.scaled.filtr <- calcNormFactors(dge.scaled.filtr)


```

```{r fig.height = 7, fig.width=10, fig.cap="*Principal component analysis of transcript level data. PCA was performed on log2 transformed CPM after filtering lowly expressed genes. The PCA shows that samples are clustering close to their conditions based on PC1, however one of the samples of the UPF3A OE in UPF3B KD cell line (sample 223), seems to deviate from its condition group and the rest of the data, so needs to be further investigated*"}
 pca = cpm(dge.scaled.filtr, log = TRUE) %>% 
  t() %>% 
    prcomp(scale = TRUE) %>% 
  tidy() %>% 
  dplyr::rename(names = row) %>% 
  left_join(dge.scaled.filtr$samples, by = "names") %>%
   dplyr::filter(PC %in% 1:3) %>% 
  pivot_wider(names_from = "PC", names_prefix = "PC", values_from = "value") %>% 
  ggplot(
    aes(PC1, PC2, colour = Group, size = lib.size/1e6)
  ) +
  geom_point() +
  geom_text_repel(aes(label = names), show.legend = FALSE) +
  scale_colour_manual(values = group_cols, 
                      labels = c("UPF3A_KD_UPF3B_KD" = "UPF3 dKD", "UPF3A_KD" = "UPF3A KD", 
                                                                   "UPF3B_KD" = "UPF3B KD", "UPF3A_OE" = "UPF3A OE", 
                                                                   "UPF3A_OE_UPF3B_KD" = "UPF3A OE UPF3B KD")) +
  #scale_size_continuous(limits = c(5, 15), breaks = seq(5, 15, by = 5)) +
  theme_bw() + ggtitle("Transcript Level")



pca
  
```


```{r fig.height=5, fig.width=5}
 plotMDS(dge.scaled.filtr,col=as.numeric(dge.scaled.filtr$samples$group))
```



# Analysis

```{r include=FALSE}
catch <- catchSalmon(salmon[-12])
colnames(catch$counts) =  paste(c(212:222, 224:229))

cts.scaled = catch$counts/catch$annotation$Overdispersion

md_filt = md %>% dplyr::filter(names != "223")


dge <- DGEList(counts = cts.scaled, 
               genes=catch$annotation, 
               samples = md_filt,
               group = md_filt$Group)

keep <- filterByExpr(dge)
dge.scaled.filtr <- dge[keep, , keep.lib.sizes = FALSE]
dge.scaled.filtr <- calcNormFactors(dge.scaled.filtr)


cpm(dge.scaled.filtr, log = TRUE) %>% 
  t() %>% 
    prcomp(scale = TRUE) %>% 
  tidy() %>% 
  dplyr::rename(names = row) %>% 
  left_join(dge.scaled.filtr$samples, by = "names") %>%
   dplyr::filter(PC %in% 1:3) %>% 
  pivot_wider(names_from = "PC", names_prefix = "PC", values_from = "value") %>% 
  ggplot(
    aes(PC1, PC2, colour = Group, size = lib.size/1e6)
  ) +
  geom_point() +
  geom_text_repel(aes(label = names), show.legend = FALSE) +
  scale_colour_manual(values = group_cols, 
                      labels = c("UPF3A_KD_UPF3B_KD" = "UPF3 dKD", "UPF3A_KD" = "UPF3A KD", 
                                                                   "UPF3B_KD" = "UPF3B KD", "UPF3A_OE" = "UPF3A OE", 
                                                                   "UPF3A_OE_UPF3B_KD" = "UPF3A OE UPF3B KD")) +
  #scale_size_continuous(limits = c(5, 15), breaks = seq(5, 15, by = 5)) +
  theme_bw() + ggtitle("Transcript Level")
  

```

```{r}
plotMDS(dge.scaled.filtr,col=as.numeric(dge.scaled.filtr$samples$group))
#legend("bottomleft", as.character(unique(dge.scaled.filtr$samples$group)), col=1:3, pch=20)

```


```{r fig.height=6, fig.width=6, fig.cap= "*Biological coeficcient of variation plot against the average abundance of each transcript. The plot shows the square-root estimates of the common, trended and tagwise NB dispersions.*"}
design <-  model.matrix(~0+Group, data = md_filt) %>% 
    set_colnames(gsub(pattern = "Group", replacement ="", x = colnames(.)))

dge.scaled.filtr <- estimateDisp(dge.scaled.filtr,design)

dge.scaled.filtr$common.dispersion

plotBCV(dge.scaled.filtr)
```

```{r fig.height=6, fig.width=6, fig.cap="*Quasi-likelihood dispersion aganist gene abundance. Estimates are shown for raw, tended and squeezed dispersions*"}
fit <- glmQLFit(dge.scaled.filtr,design)
plotQLDisp(fit)
```

```{r}
summary(fit$df.prior)
```


```{r}
contrasts <- makeContrasts(levels = colnames(design), 
                           UPF3B_KD_Control = UPF3B_KD - Control, 
                           UPF3A_KD_Control = UPF3A_KD - Control, 
                           UPF3A_OE_Control = UPF3A_OE - Control, 
                           UPF3A_KD_UPF3B_KD_Control = UPF3A_KD_UPF3B_KD - Control, 
                           UPF3A_OE_UPF3B_KD_Control = UPF3A_OE_UPF3B_KD - Control)


dgeLRT_obj = list()
res_dte = list()
metrics = list()
for (i in 1:ncol(contrasts)){
  temp = glmQLFTest(fit, contrast= contrasts[,i])
  tt = topTags(temp,n = Inf)
  tt %<>% as.data.frame() %>% rownames_to_column("tx_id") %>%
    left_join(decideTestsDGE(temp) %>%
                  as.data.frame() %>%
                  set_colnames(c("Results")) %>%
                  rownames_to_column("tx_id"))
  res_dte[[paste0(colnames(contrasts)[i])]] = tt
  dgeLRT_obj[[paste0(colnames(contrasts)[i])]] = temp
  
  setDT(tt)[]
  tt %<>% dplyr::rename("ID" = "tx_id")
  tt <- as.data.frame(tt)
  metrics[[paste0(colnames(contrasts)[i])]] = tt
}
```

## Checking normalization and QC

```{r}
scaled_matrix <- as.data.frame(t(apply(as.matrix(dge.scaled.filtr$counts), 1, scale)))
#scaled_matrix <- as.character(rownames(dge.scaled.filtr$counts))
colLength = length(scaled_matrix)
scaled_matrix<- scaled_matrix[,c(colLength, 1:(colLength-1))]
colnames(scaled_matrix) <- colnames(dge.scaled.filtr$counts)
nID <- which(is.nan(scaled_matrix[,2]))
scaled_matrix[nID,2:length(scaled_matrix)] <- 0
```

```{r}
scaled_matrix %<>% rownames_to_column("ID")

counts = log(dge.scaled.filtr$counts + 0.05)
counts  %<>% as.data.frame() %>%rownames_to_column("ID")
colnames(counts) =c("ID", "Control.1", "UPF3AKD.1", "UPF3AOE.1", "UPF3BKD.1", "UPF3AKDUPF3BKD.1", "UPF3AOEUPF3BKD.1",
                    "Control.2", "UPF3AKD.2", "UPF3AOE.2", "UPF3BKD.2", "UPF3AKDUPF3BKD.2",
                    "Control.3", "UPF3AKD.3", "UPF3AOE.3", "UPF3BKD.3", "UPF3AKDUPF3BKD.3", "UPF3AOEUPF3BKD.2")

names(metrics) = c("UPF3BKD_Control", "UPF3AKD_Control","UPF3AOE_Control", "UPF3AKDUPF3BKD_Control", "UPF3AOEUPF3BKD_Control")
ret = plotSM(counts[c(1:2, 5,8,11,13,16)], dataMetrics = metrics[1], saveFile = FALSE)
ret 
```

```{r}
upf3a_oe = plotSM(counts[c(1:2, 4,8,10,13,15)], dataMetrics = metrics[3], saveFile = FALSE)
upf3a_oe
```

# Distribution of results 
```{r}
plot_hist = function(df, comparison, legend = FALSE){
  ggplot(data= df, aes(x=PValue)) + geom_histogram(fill = "grey", color="black") + theme_bw() + xlab("Pvalue") + ylab("Frequency")
}
plot_fc = function(df){
  df %>% 
    dplyr::filter(FDR < 0.05) %>%  
    ggplot(aes(x=logFC)) + 
    geom_histogram(aes(y=..density..), colour="black", fill="white") +  
    geom_density(alpha=.2, fill="#FF6666") + geom_vline(xintercept =c(log2(1.5), -log2(1.5)),color="red", linetype = "dashed")
}
```

## Fold change distributions 

```{r}
a_f = plot_fc(res_dte$UPF3B_KD_Control) 
b_f = plot_fc(res_dte$UPF3A_KD_Control)
c_f = plot_fc(res_dte$UPF3A_KD_UPF3B_KD_Control)
d_f = plot_fc(res_dte$UPF3A_OE_Control)
e_f = plot_fc(res_dte$UPF3A_OE_UPF3B_KD_Control)
```


```{r fig.height=10, fig.cap = "Distribution of p-values and log2Fold changes across conditions after running edgeR"}
a = plot_hist(res_dte$UPF3B_KD_Control) + ggtitle("(a) UPF3B KD vs Control")
b = plot_hist(res_dte$UPF3A_KD_Control) + ggtitle("(b) UPF3A KD vs Control")
c = plot_hist(res_dte$UPF3A_KD_UPF3B_KD_Control) + ggtitle("(c) UPF3 dKD vs Control")
d =  plot_hist(res_dte$UPF3A_OE_Control) + ggtitle("(d) UPF3A OE vs Control")
e =  plot_hist(res_dte$UPF3A_OE_UPF3B_KD_Control) + ggtitle("(e) UPF3A OE UPF3B KD vs Control")

#grid.arrange(a,b,c,d,e, ncol = 3, nrow=2)


grid.arrange(a, a_f,
             b,b_f, 
             c, c_f, 
             d, d_f, 
             e,e_f, ncol = 2, nrow=5)
```


```{r fig.width=10, fig.height=10, fig.cap="MA plot of comprisons of KD/OE with Controls. Pink highlights significantly upregulated transcripts whereas blue highlights significantly downregulated transcripts"}
## Ma plot 

ma_1 = ma_plot(res_dte$UPF3B_KD_Control, 0) + ggtitle("UPF3B KD vs Control")
ma_2 = ma_plot(res_dte$UPF3A_KD_Control, 0) + ggtitle("UPF3A KD vs Control")
ma_3 = ma_plot(res_dte$UPF3A_KD_UPF3B_KD_Control, 0) + ggtitle("UPF3 dKD vs Control")
ma_4 = ma_plot(res_dte$UPF3A_OE_Control, 0) + ggtitle("UPF3A OE vs Control")
ma_5 = ma_plot(res_dte$UPF3A_OE_UPF3B_KD_Control, 0) + ggtitle("UPF3A OE in UPF3B KD")

grid.arrange(ma_1, ma_2, ma_3, 
            ma_4, ma_5, ncol=3, nrow=2)
```

```{r fig.height= 5, fig.width=5, fig.cap="*Ratio of up/down differentiallt expressed transcripts*"}
 ratios = data.frame()
 for (c in names(res_dte)){
   temp = table(res_dte[[c]]$Results) %>% 
     melt() %>% 
     dplyr::filter(Var1 != 0) 
   ratio= temp$value[2]/temp$value[1]
   ratio = data.frame(Comparisons = c, 
            ratio = ratio)
   ratios = rbind(ratios, ratio)
   }

ratios  %>%
  ggplot(aes(x =Comparisons, y = ratio, fill = Comparisons)) + 
  geom_bar(stat = "identity",   colour = "white", width = .75) + 
    theme_classic() + 
    theme(axis.text.x = element_text(angle = 70,  hjust=1, face = "bold"), 
          legend.position = "none", 
          legend.title= element_blank(), axis.title.y = element_text(size = 12), 
          axis.text.y = element_text(size = 14, face = "bold")) + 
    scale_fill_manual(values = c("#E12F8F","#6E74B4","#EA2A5F", "#FD9675", "#2F124B")) +
    ylab("ratio up/down") + xlab("") 


```


## DET statistics 

```{r}
NMD_features = read.table("data/nif_output/Mus_musculus__nifs.tsv", sep = "\t", skip =1)

colnames(NMD_features) = c("tx_id",	"NIF_dEJ", "NIF_uORF",	"NIF_long_3utr", "NIF|__|GC(cds[-10:termination_codon])",	"GC_5utr",	"GC_cds",	"GC_3utr",	"GC_exons",	"GC_introns",	"LENGTH_5utr", 	"LENGTH_cds",	"LENGTH_3utr",	"LENGTH_exons", "LENGTH_introns", "EXONS_count",	"dEJs|__|penultimate-exon>=55(nts)",	"uORFs|__|count",	"uORFs|__|gc-content",	"uORFs|__|lengths", "uORFs|__|max-length")

NMD_features$NIF_dEJ = ifelse(NMD_features$NIF_dEJ == "True",gsub("True", 1, NMD_features$NIF_dEJ), 0)
NMD_features$NIF_uORF = ifelse(NMD_features$NIF_uORF == "True",gsub("True", 1, NMD_features$NIF_uORF), 0)
NMD_features$NIF_long_3utr = ifelse(NMD_features$NIF_long_3utr == "True",gsub("True", 1, NMD_features$NIF_long_3utr), 0)
NMD_features %<>% mutate_at(.vars = "tx_id", .funs = gsub, pattern = "\\.[0-9]*$", replacement = "")
```


```{r include=FALSE}
total = data.frame()
for (i in names(res_dte)){
    name = i
    #comparison = rep(name, 4) %>% as.data.frame()
    temp = res_dte[[i]]
    
    message("Merging DE results with NMD table for", paste(name))
    temp %<>%  as.data.frame() %>%
      mutate(tx_version = tx_id) %>%
      mutate_at(.vars = "tx_id", .funs = gsub, pattern = "\\.[0-9]*$", replacement = "") %>% 
      left_join(NMD_features, by = "tx_id")
    
    message("Summing all NMD values to determine number of NIFs per transcript for", paste0(name))
    temp$nif_feature = ifelse(rowSums(sapply(temp[, c(12:14)],
                       function(x) as.numeric(as.character(x)))) == 2 ,"Two",
                       ifelse(rowSums(sapply(temp[, c(12:14)],
                              function(x) as.numeric(as.character(x)))) == 3, "Three", 
                              ifelse(rowSums(sapply(temp[, c(12:14)],
                                                    function(x) as.numeric(as.character(x)))) == 1, "One", "None")))
    
    temp$nif_feature %<>%  replace_na("None")
    res_dte[[paste(name)]] = temp
    
    ## downregulation 
    message("Subsetting downregulated genes for ", paste0(name))
    down = temp %>%  as.data.frame() %>%  
        dplyr::filter(FDR < 0.05 & logFC < 0) 
    down = table(down$nif_feature)
    print(nrow(down))
    repetition = down %>% nrow() %>% as.integer()
    expression = rep(c("down"), repetition) %>% as.data.frame()
    comparison = rep(name, nrow(expression)) %>% as.data.frame()
    down = cbind(down, expression = expression)
    down = cbind(down, comparison = comparison)
    colnames(down) = c("NIF", "Value", "Expression", "Condition")
    
    ## upregulation 
    message("Subsetting upregulated genes for ", paste0(name))
    up = temp %>%  as.data.frame() %>%  
        dplyr::filter(FDR < 0.05 & logFC > 0 ) 
    up = table(up$nif_feature)
    print(dim(up))
    repetition = up %>% nrow() %>% as.integer()
    expression = rep(c("up"), repetition) %>% as.data.frame()
    comparison = rep(name, nrow(expression)) %>% as.data.frame()
    up = cbind(up, expression = expression)
    up = cbind(up, comparison = comparison)
    colnames(up) = c("NIF", "Value", "Expression", "Condition")
    
    ## bind everything together 
    message("Binding everything together")
    #together = rbind(up, down)
   
    #total = rbind(total,as.data.frame(up,down))
    total = rbind(total,up,down, stringsAsFactors = FALSE)
}
```

```{r fig.width=10}
total$NIF <- factor(total$NIF, levels = c("Three", "Two", "One", "None"))
up = total %>% 
  dplyr::filter(Expression == "up") %>% 
  ggplot(aes(x= Condition, y = Value, fill=NIF)) + geom_bar(stat = "identity", width = .5,color = "white") + coord_flip() +
  theme_bw() + ylab("Number of Upregulated Transcripts") + xlab("") + 
  scale_fill_manual(values = c("#FF8DD0","#E0155F", "#C40C71", "#370059")) +
  theme(axis.text.x = element_text(size = 10, face = "bold"),
        legend.position = "bottom") + ylim(0, 4500) + 
  labs(fill = "Number of NMD inducible features (NIF)", labels = c("No NIF", "One NIF feature", "Two NIF features" , "Three NIF features"))

#ggsave(file = "/home/neuro/Documents/NMD_analysis/Analysis/NMD-analysis/output/Transcript/Thesis_figures/total_exp_up.svg", plot = up, 
 #      height = 6.31, width = 5.77, units = "in")


down = total %>% 
  dplyr::filter(Expression == "down") %>% 
  ggplot(aes(x= Condition, y = Value, fill=NIF)) + 
  geom_bar(stat = "identity", width = .5,color = "white") + coord_flip() +
  theme_bw() + ylab("Number of Downregulated Transcripts") + xlab("") + 
  scale_fill_manual(values = c("#FF8DD0","#E0155F", "#C40C71", "#370059")) +
  theme(axis.text.x = element_text(size = 10, face = "bold"), 
        legend.position = "none") + scale_y_reverse(limits=c(4500,0))  + 
  labs(fill = "Number of NMD inducible features (NIF)", labels = c("No NIF", "One NIF feature", "Two NIF features" , "Three NIF features"))



ggarrange(down, up, common.legend = TRUE)
#ggsave(file = "/home/neuro/Documents/NMD_analysis/Analysis/NMD-analysis/output/Transcript/Thesis_figures/total_exp_down.svg", plot = down, 
#       height = 6.31, width = 5.77, units = "in")
```


```{r}
up = total %>% 
  dplyr::filter(Expression == "up") %>% 
  ggplot(aes(x= Condition, y = Value, fill=NIF)) + geom_bar(position ="fill", stat = "identity",color = "white") + coord_flip() +
  theme_bw() + ylab("Proportion of Upregulated Transcripts") + xlab("") + 
  scale_fill_manual(values = c("#FF8DD0","#E0155F", "#C40C71", "#370059")) +
  theme(axis.text.x = element_text(size = 10, face = "bold")) +   
  labs(fill = "Number of NMD inducible features (NIF)", 
       labels = c("No NIF", "One NIF feature", "Two NIF features" , "Three NIF features"))

down = total %>% 
  dplyr::filter(Expression == "down") %>% 
  ggplot(aes(x= Condition, y = Value, fill=NIF)) + 
  geom_bar(position="fill", stat="identity",color = "white") + coord_flip() +
  theme_bw() + ylab("Proportion of Downregulated Transcripts ") + xlab("") + 
  scale_fill_manual(values = c("#FF8DD0","#E0155F", "#C40C71", "#370059")) +
  theme(axis.text.x = element_text(size = 10, face = "bold")) + scale_y_reverse() +   
  labs(fill = "Number of NMD inducible features (NIF)", 
       labels = c("No NIF", "One NIF feature", "Two NIF features" , "Three NIF features"))


ggarrange(down, up,  common.legend = TRUE)
```


```{r fig.width=12, fig.cap= "Number of NIF features in differentially expressed transcripts per comparison"}
nif_features = list()
for (i in names(res_dte)){
  name = i
  temp = res_dte[[i]]
  sig = temp %>%  as.data.frame() %>%  
        dplyr::filter(FDR < 0.05) 
  utr = table(sig$NIF_long_3utr)[2]
  dej = table(sig$NIF_dEJ)[2]
  uorf = table(sig$NIF_uORF)[2]
  total_nif = rbind(utr, dej, uorf) %>% 
    as.data.frame() %>% 
    rownames_to_column("NIF_feature") %>% 
    mutate(Condition = name) %>% 
    dplyr::rename("Count" = "1")
  nif_features= rbind(nif_features, total_nif, stringsAsFactors = FALSE)
  
    
  
}

nif_features %>%
  ggplot(aes(x= NIF_feature, y = Count, fill = NIF_feature)) + 
  geom_bar(stat = "identity", position = "dodge", width = .5) + coord_flip() + 
  facet_grid(~Condition) + theme_bw() + labs(fill = "Type of NIF",
                                              labels = c("Downstream exon junction", "Upstream ORF", 
                                                         "Long 3'UTR"))


```

```{r}
# Calculating enrichment for this work 
### Ask Irina about adjusted pvalues 
enrichment_NIF = list()
for (c in names(res_dte)){
  test.list = list()
  test.list$BG = res_dte[[c]]$tx_id
  test.list$NIF = res_dte[[c]]$tx_id[res_dte[[c]]$nif_feature != "None"]
  test.list$sig = res_dte[[c]]$tx_id[res_dte[[c]]$FDR < 0.05]
  test.list$up = res_dte[[c]]$tx_id[res_dte[[c]]$FDR < 0.05 & res_dte[[c]]$logFC > 0]
  test.list$sig = res_dte[[c]]$tx_id[res_dte[[c]]$FDR < 0.05 & res_dte[[c]]$logFC < 0]
  
  for (det in c("sig", "up", "down")) {
    go.obj <- newGeneOverlap(test.list[[det]],
                         test.list$NIF,
                         genome.size=length(test.list$BG))
    go.obj <- testGeneOverlap(go.obj)
    enrichment_NIF[[paste0(c, "_", det)]] = go.obj@pval
    
  }
}

enrichment_NIF = do.call(rbind, enrichment_NIF) %>% as.data.frame()
colnames(enrichment_NIF)[1] = c("pvalue")
enrichment_NIF$padj = p.adjust(enrichment_NIF$pvalue, method = "BH")


```

```{r}
enrichment_nif_feature = list()
nifs_list = c("NIF_uORF", "NIF_dEJ", "NIF_utr")

for (c in names(res_dte)){
  test.list = list()
  test.list$BG = res_dte[[c]]$tx_id
  test.list$NIF_uORF = res_dte[[c]]$tx_id[res_dte[[c]]$NIF_uORF== 1 ]
  test.list$NIF_dEJ = res_dte[[c]]$tx_id[res_dte[[c]]$NIF_dEJ== 1 ]
  test.list$NIF_utr = res_dte[[c]]$tx_id[res_dte[[c]]$NIF_long_3utr== 1 ]
  test.list$sig = res_dte[[c]]$tx_id[res_dte[[c]]$FDR < 0.05]
  test.list$up = res_dte[[c]]$tx_id[res_dte[[c]]$FDR < 0.05 & res_dte[[c]]$logFC > 0]
  test.list$down = res_dte[[c]]$tx_id[res_dte[[c]]$FDR < 0.05 & res_dte[[c]]$logFC < 0]
  for (nif in nifs_list){
     go.obj <- newGeneOverlap(test.list$sig,
                         test.list[[nif]],
                         genome.size=length(test.list$BG))
     go.obj <- testGeneOverlap(go.obj)
     enrichment_nif_feature[[paste0(c, "_", nif)]] = go.obj@pval
     go.obj <- newGeneOverlap(test.list$up,
                         test.list[[nif]],
                         genome.size=length(test.list$BG))
     go.obj <- testGeneOverlap(go.obj)
     enrichment_nif_feature[[paste0(c,"_",nif,  "_up")]] = go.obj@pval
     go.obj <- newGeneOverlap(test.list$down,
                         test.list[[nif]],
                         genome.size=length(test.list$BG))
     go.obj <- testGeneOverlap(go.obj)
     enrichment_nif_feature[[paste0(c,"_",nif, "_down")]] = go.obj@pval
       
    
  }
}

```

## Distribution of NIFs

```{r}
NIF_dist = function(res_matrix){
  temp = res_matrix
  dEJ_dist = temp %>% as.data.frame() %>%
    ggplot(aes(x=logFC)) +
    geom_density(data=dplyr::filter(temp,NIF_dEJ == 1, FDR < 0.05), fill="#FF2759", alpha = 0.5) +
    geom_density(data=dplyr::filter(temp, NIF_dEJ == 0, FDR < 0.05), fill="lightgray", alpha = 0.2) +
    geom_density(data=dplyr::filter(temp, is.na(NIF_dEJ), FDR < 0.05), fill="#3A7FA4", alpha = 0.2) +  theme_bw() +
    ylab("Filtered Density") + xlab("Log2 Fold Change") + ggtitle("Density distribution of dEJ")

  uORF_dist  = temp %>%  as.data.frame() %>%
    ggplot(aes(x=logFC)) +
    geom_density(data=dplyr::filter(temp,NIF_uORF == 1, FDR < 0.05),
                 fill="#FF2759", alpha = 0.5) +
    geom_density(data=dplyr::filter(temp, NIF_uORF == 0, FDR < 0.05), fill="lightgray", alpha = 0.2) +
    geom_density(data=dplyr::filter(temp, is.na(NIF_uORF), FDR < 0.05), fill="#3A7FA4", alpha = 0.2) +  theme_bw() +
    ylab("Filtered Density") + xlab("Log2 Fold Change") + ggtitle("Density distribution of uORF")

  utr_dist = temp %>%  as.data.frame() %>%
    ggplot(aes(x=logFC)) + geom_density(data=dplyr::filter(temp, NIF_long_3utr == 1, FDR < 0.05), fill="#FF2759", alpha = 0.5) +
    geom_density(data=dplyr::filter(temp, NIF_long_3utr == 0, FDR < 0.05), fill="lightgray", alpha = 0.2) +
    geom_density(data=dplyr::filter(temp, is.na(NIF_long_3utr), FDR < 0.05), fill="#3A7FA4", alpha = 0.2) +  theme_bw() +
    ylab("Filtered Density") + xlab("Log2 Fold Change") + ggtitle("Density distribution Long 3' UTRs")

  plot = grid.arrange(dEJ_dist, uORF_dist, utr_dist, nrow=3)
  return(plot)
}


```

### UPF3B KD vs controls 
```{r fig.height=10, fig.width=8}
upf3b = NIF_dist(res_dte$UPF3B_KD_Control)
```


### UPF3A KD vs controls 
```{r fig.height=10, fig.width=8}
upf3a = NIF_dist(res_dte$UPF3A_KD_Control) 

```

### UPF3 dKD vs controls
```{r fig.height=10, fig.width=8}
dkd = NIF_dist(res_dte$UPF3A_KD_UPF3B_KD_Control) 
```

### UPF3A OE vs controls 
```{r fig.height=10, fig.width=8}
upf3a_oe = NIF_dist(res_dte$UPF3A_OE_Control)

```

### UPF3A OE UPF3B KD vs controls 
```{r fig.height=10, fig.width=8}
upf3a_oe_b_kd =NIF_dist(res_dte$UPF3A_OE_UPF3B_KD_Control) 
```




# UPF3B transcripts 

```{r}
res_dte$UPF3B_KD_Control %>% 
  as.data.frame() %>% 
  dplyr::filter(Results != 0) %>%
  dplyr::select(tx_id, Overdispersion, logFC, 
                logCPM, PValue, FDR, NIF_dEJ, NIF_uORF, LENGTH_3utr, 
                `NIF|__|GC(cds[-10:termination_codon])`) %>% 
  inner_join(tx2gene, by = "tx_id") %>% 
  mutate(gene = mapIds(org.Mm.eg.db, keys=gene_id, 
                       column="SYMBOL",keytype="ENSEMBL", multiVals="first")) %>% 
      DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```

## UPF3B transcripts with NMD features 

```{r fig.cap= "*NIF in upregulated transcripts in UPF3B*"}
sig_res= res_dte$UPF3B_KD_Control  %>% 
  dplyr::filter(FDR < 0.05, abs(logFC) > log2(1))


dEJ = sig_res %>% 
  dplyr::filter(NIF_dEJ == 1) %>% 
  dplyr::select(tx_id)

uORF = sig_res %>% 
  dplyr::filter(NIF_uORF == 1) %>% 
  dplyr::select(tx_id)

UTR = sig_res %>% 
  dplyr::filter(NIF_long_3utr == 1) %>% 
  dplyr::select(tx_id)

x = list( "dEJ" = dEJ$tx_id, 
          "uORF" = uORF$tx_id, 
          "Long 3'UTR" = UTR$tx_id, 
          "All transcript" = sig_res$tx_id)

upset(fromList(x))

```


## Comparison of DEGs with DETs for UPF3B KD 

```{r}
res_dte$UPF3B_KD_Control %>%
  dplyr::filter(FDR < 0.05) %>% 
  inner_join(tx2gene, by = "tx_id") %>% 
  mutate(gene = mapIds(org.Mm.eg.db, keys=gene_id, 
                       column="SYMBOL",keytype="ENSEMBL", multiVals="first")) %>%
  inner_join(dplyr::rename(DEGenes$UPF3B_vs_Control, "gene_id" = "ensembl_gene_id"), by = "gene_id") %>% 
  ggscatter(y="logFC", x="Coef.UPF3B_vs_Control",  cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray"))  + 
  xlab("UPF3B KD (gene level logFC)") + ylab("UPF3B KD (transcript level logFC)")


```

```{r}
res_dte$UPF3B_KD_Control %>%
  dplyr::filter(FDR < 0.05) %>% 
  dplyr::select(tx_id, Overdispersion, logFC, 
                logCPM, PValue, FDR, NIF_dEJ, NIF_uORF, LENGTH_3utr, 
                `NIF|__|GC(cds[-10:termination_codon])`) %>% 
   inner_join(tx2gene, by = "tx_id") %>%
  mutate(gene = mapIds(org.Mm.eg.db, keys=gene_id, 
                       column="SYMBOL",keytype="ENSEMBL", multiVals="first")) %>%
  inner_join(DEGenes$UPF3B_vs_Control %>% 
              dplyr::rename("gene_id" = "ensembl_gene_id") %>% 
              dplyr::select(contains("UPF3B_vs_Control"), gene_id), by = "gene_id") %>% 
     DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))

```

# UPF3A transcripts 

```{r}
res_dte$UPF3A_KD_Control %>% 
  as.data.frame() %>% 
  dplyr::filter(Results != 0) %>%
  dplyr::select(tx_id, Overdispersion, logFC, 
                logCPM, PValue, FDR, NIF_dEJ, NIF_uORF, LENGTH_3utr, 
                `NIF|__|GC(cds[-10:termination_codon])`) %>% 
  inner_join(tx2gene, by = "tx_id") %>% 
  mutate(gene = mapIds(org.Mm.eg.db, keys=gene_id, 
                       column="SYMBOL",keytype="ENSEMBL", multiVals="first")) %>% 
      DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```

## UPF3A transcripts with NMD features 

```{r fig.cap= "*NIF in upregulated transcripts in UPF3B*"}
sig_res= res_dte$UPF3A_KD_Control  %>% 
  dplyr::filter(FDR < 0.05, abs(logFC) > log2(1))


dEJ = sig_res %>% 
  dplyr::filter(NIF_dEJ == 1) %>% 
  dplyr::select(tx_id)

uORF = sig_res %>% 
  dplyr::filter(NIF_uORF == 1) %>% 
  dplyr::select(tx_id)

UTR = sig_res %>% 
  dplyr::filter(NIF_long_3utr == 1) %>% 
  dplyr::select(tx_id)

x = list( "dEJ" = dEJ$tx_id, 
          "uORF" = uORF$tx_id, 
          "Long 3'UTR" = UTR$tx_id, 
          "All transcript" = sig_res$tx_id)

upset(fromList(x))

```


### Comparison of DEGs with DETs

```{r}
res_dte$UPF3A_KD_Control %>%
  dplyr::filter(FDR < 0.05) %>% 
  inner_join(tx2gene, by = "tx_id") %>% 
  mutate(gene = mapIds(org.Mm.eg.db, keys=gene_id, 
                       column="SYMBOL",keytype="ENSEMBL", multiVals="first")) %>%
  inner_join(dplyr::rename(DEGenes$UPF3A_vs_Control, "gene_id" = "ensembl_gene_id"), by = "gene_id") %>% 
  ggscatter(y="logFC", x="Coef.UPF3A_vs_Control",  cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray"))  + 
  xlab("UPF3A KD (gene level logFC)") + ylab("UPF3A KD (transcript level logFC)")


```


```{r}
res_dte$UPF3A_KD_Control %>%
  dplyr::filter(FDR < 0.05) %>% 
  dplyr::select(tx_id, Overdispersion, logFC, 
                logCPM, PValue, FDR, NIF_dEJ, NIF_uORF, LENGTH_3utr, 
                `NIF|__|GC(cds[-10:termination_codon])`) %>% 
   inner_join(tx2gene, by = "tx_id") %>%
  mutate(gene = mapIds(org.Mm.eg.db, keys=gene_id, 
                       column="SYMBOL",keytype="ENSEMBL", multiVals="first")) %>%
  inner_join(DEGenes$UPF3A_vs_Control %>% 
              dplyr::rename("gene_id" = "ensembl_gene_id") %>% 
              dplyr::select(contains("UPF3A_vs_Control"), gene_id), by = "gene_id") %>% 
     DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```



# UPF3 dKD

```{r}
res_dte$UPF3A_KD_UPF3B_KD_Control %>% 
  as.data.frame() %>% 
  dplyr::filter(Results != 0) %>%
  dplyr::select(tx_id, Overdispersion, logFC, 
                logCPM, PValue, FDR, NIF_dEJ, NIF_uORF, LENGTH_3utr, 
                `NIF|__|GC(cds[-10:termination_codon])`) %>% 
  inner_join(tx2gene, by = "tx_id") %>% 
  mutate(gene = mapIds(org.Mm.eg.db, keys=gene_id, 
                       column="SYMBOL",keytype="ENSEMBL", multiVals="first")) %>% 
      DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```


### Comparison of DEGs with DETs

```{r}
res_dte$UPF3A_KD_UPF3B_KD_Control %>%
  dplyr::filter(FDR < 0.05) %>% 
  inner_join(tx2gene, by = "tx_id") %>% 
  mutate(gene = mapIds(org.Mm.eg.db, keys=gene_id, 
                       column="SYMBOL",keytype="ENSEMBL", multiVals="first")) %>%
  inner_join(dplyr::rename(DEGenes$DoubleKD_vs_Control, "gene_id" = "ensembl_gene_id"), by = "gene_id") %>% 
  ggscatter(y="logFC", x="Coef.DoubleKD_vs_Control",  cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
  xlab("UPF3 dKD (gene level logFC)") + ylab("UPF3 dKD (transcript level logFC)")

```


```{r}
res_dte$UPF3A_KD_UPF3B_KD_Control %>%
  dplyr::filter(FDR < 0.05) %>% 
  dplyr::select(tx_id, Overdispersion, logFC, 
                logCPM, PValue, FDR, NIF_dEJ, NIF_uORF, LENGTH_3utr, 
                `NIF|__|GC(cds[-10:termination_codon])`) %>% 
   inner_join(tx2gene, by = "tx_id") %>%
  mutate(gene = mapIds(org.Mm.eg.db, keys=gene_id, 
                       column="SYMBOL",keytype="ENSEMBL", multiVals="first")) %>%
  inner_join(DEGenes$DoubleKD_vs_Control %>% 
              dplyr::rename("gene_id" = "ensembl_gene_id") %>% 
              dplyr::select(contains("DoubleKD_vs_Control"), gene_id), by = "gene_id") %>% 
     DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```




## Transcripts with NIFs

```{r fig.cap= "*NIF in upregulated transcripts in UPF3B*"}
sig_res= res_dte$UPF3A_KD_UPF3B_KD_Control  %>% 
  dplyr::filter(FDR < 0.05, abs(logFC) > log2(1))


dEJ = sig_res %>% 
  dplyr::filter(NIF_dEJ == 1) %>% 
  dplyr::select(tx_id)

uORF = sig_res %>% 
  dplyr::filter(NIF_uORF == 1) %>% 
  dplyr::select(tx_id)

UTR = sig_res %>% 
  dplyr::filter(NIF_long_3utr == 1) %>% 
  dplyr::select(tx_id)

x = list( "dEJ" = dEJ$tx_id, 
          "uORF" = uORF$tx_id, 
          "Long 3'UTR" = UTR$tx_id, 
          "All transcript" = sig_res$tx_id)

upset(fromList(x))

```

# UPF3A OE 


```{r}
res_dte$UPF3A_OE_Control %>% 
  as.data.frame() %>% 
  dplyr::filter(Results != 0) %>%
  dplyr::select(tx_id, Overdispersion, logFC, 
                logCPM, PValue, FDR, NIF_dEJ, NIF_uORF, LENGTH_3utr, 
                `NIF|__|GC(cds[-10:termination_codon])`) %>% 
  inner_join(tx2gene, by = "tx_id") %>% 
  mutate(gene = mapIds(org.Mm.eg.db, keys=gene_id, 
                       column="SYMBOL",keytype="ENSEMBL", multiVals="first")) %>% 
      DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))
```


```{r}
res_dte$UPF3A_OE_Control %>%
  dplyr::filter(FDR < 0.05) %>% 
  dplyr::select(tx_id, Overdispersion, logFC, 
                logCPM, PValue, FDR, NIF_dEJ, NIF_uORF, LENGTH_3utr, 
                `NIF|__|GC(cds[-10:termination_codon])`) %>% 
   inner_join(tx2gene, by = "tx_id") %>%
  mutate(gene = mapIds(org.Mm.eg.db, keys=gene_id, 
                       column="SYMBOL",keytype="ENSEMBL", multiVals="first")) %>%
  inner_join(DEGenes$UPF3A_OE_vs_Control %>% 
              dplyr::rename("gene_id" = "ensembl_gene_id") %>% 
              dplyr::select(contains("UPF3A_OE_vs_Control"), gene_id), by = "gene_id") %>% 
     DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


```

# UPF3A OE in UPF3B KD 

```{r}
res_dte$UPF3A_OE_UPF3B_KD_Control %>% 
  as.data.frame() %>% 
  dplyr::filter(Results != 0) %>%
  dplyr::select(tx_id, Overdispersion, logFC, 
                logCPM, PValue, FDR, NIF_dEJ, NIF_uORF, LENGTH_3utr, 
                `NIF|__|GC(cds[-10:termination_codon])`) %>% 
  inner_join(tx2gene, by = "tx_id") %>% 
  mutate(gene = mapIds(org.Mm.eg.db, keys=gene_id, 
                       column="SYMBOL",keytype="ENSEMBL", multiVals="first")) %>% 
      DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))

```


```{r}
res_dte$UPF3A_OE_UPF3B_KD_Control %>%
  dplyr::filter(FDR < 0.05) %>% 
  inner_join(tx2gene, by = "tx_id") %>% 
  mutate(gene = mapIds(org.Mm.eg.db, keys=gene_id, 
                       column="SYMBOL",keytype="ENSEMBL", multiVals="first")) %>%
  inner_join(dplyr::rename(DEGenes$UPF3A_OE_UPF3B_KD_vs_Control, "gene_id" = "ensembl_gene_id"), by = "gene_id") %>% 
  ggscatter(y="logFC", x="Coef.UPF3A_OE_UPF3B_KD_vs_Control",  cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
  xlab("UPF3A OE in UPF3B KD (gene level logFC)") + ylab("UPF3A OE in UPF3B KD (transcript level logFC)")

```

```{r}
res_dte$UPF3A_OE_UPF3B_KD_Control %>%
  dplyr::filter(FDR < 0.05) %>% 
  dplyr::select(tx_id, Overdispersion, logFC, 
                logCPM, PValue, FDR, NIF_dEJ, NIF_uORF, LENGTH_3utr, 
                `NIF|__|GC(cds[-10:termination_codon])`) %>% 
   inner_join(tx2gene, by = "tx_id") %>%
  mutate(gene = mapIds(org.Mm.eg.db, keys=gene_id, 
                       column="SYMBOL",keytype="ENSEMBL", multiVals="first")) %>%
  inner_join(DEGenes$UPF3A_OE_UPF3B_KD_vs_Control %>% 
              dplyr::rename("gene_id" = "ensembl_gene_id") %>% 
              dplyr::select(contains("UPF3A_OE_UPF3B_KD_vs_Control"), gene_id), by = "gene_id") %>% 
     DT::datatable(filter = 'top', extensions = c('Buttons', 'FixedColumns'),
            options = list(scrollX = TRUE, 
                           scrollCollapse = TRUE, 
                           pageLength = 5, autoWidth = TRUE, 
                           dom = 'Blfrtip',
                           buttons = c('copy', 'csv', 'excel', 'pdf', 'colvis'),
                           lengthMenu = list(c(10,25,50,-1),
                                             c(10,25,50,"All"))))


```


# Are DET shared between UPF3B KD and UPF3A KD? 

```{r}
degs_b= res_dte$UPF3B_KD_Control %>% as.data.frame %>%
  dplyr::filter(FDR < .05) %>% 
  with(., 
       table(sign.lfc=sign(logFC))) %>% as.data.frame()
```

In UPF3B KD, there are 1860 upregulated genes aand 1328 downregulated genes 

```{r}

res_dte$UPF3A_KD_Control %>% as.data.frame %>%
  dplyr::filter(FDR < .05) %>% 
  with(., 
       table(sign.lfc=sign(logFC)))

```

In UPF3B KD, there are 1504 upregulated genes and 816 downregulated genes 


```{r}
x = list( "UPF3B KD" = res_dte$UPF3B_KD_Control$tx_id[res_dte$UPF3B_KD_Control$FDR < 0.05],  
    "UPF3A KD" = res_dte$UPF3A_KD_Control$tx_id[res_dte$UPF3A_KD_Control$FDR < 0.05], 
    "Double KD" = res_dte$UPF3A_KD_UPF3B_KD_Control$tx_id[res_dte$UPF3A_KD_UPF3B_KD_Contro$FDR < 0.05],
    "UPF3A OE" = res_dte$UPF3A_OE_Control$tx_id[res_dte$UPF3A_OE_Control$FDR < 0.05], 
    "UPF3A OE UPF3B KD" = res_dte$UPF3A_OE_UPF3B_KD_Control$tx_id[res_dte$UPF3A_OE_UPF3B_KD_Control$FDR < 0.05])

ggvenn(x[c(1,2)])
```

```{r fig.height=8, fig.width=15}
grid.arrange((res_dte$UPF3B_KD_Control %>% as.data.frame() %>% 
  dplyr::filter(FDR < 0.05) %>% 
  dplyr::rename("logFC.UPF3B" = "logFC") %>% 
  dplyr::select(tx_id, logFC.UPF3B) %>% 
  inner_join(dplyr::filter(as.data.frame(res_dte$UPF3A_KD_Control), FDR < 0.05), by = "tx_id")  %>% 
   ggscatter(., x = "logFC.UPF3B", y = "logFC", color = "NIF_uORF",
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw()  + 
  geom_xsidedensity(aes(у = after_stat(density), fill = NIF_uORF), alpha = 0.5, 
                    size = 1,position = "stack") + geom_ysidedensity(aes(x=after_stat(density), 
                                                                         fill = NIF_uORF), 
                                                                     alpha=0.5, size = 1) +
    xlab("UPF3B KD (log2FC)") + ylab("UPF3A KD (log2FC)")),

(res_dte$UPF3B_KD_Control %>% as.data.frame() %>% 
  dplyr::filter(FDR < 0.05) %>% 
  dplyr::rename("logFC.UPF3B" = "logFC") %>% 
  dplyr::select(tx_id, logFC.UPF3B) %>% 
  inner_join(dplyr::filter(as.data.frame(res_dte$UPF3A_KD_Control), FDR < 0.05), by = "tx_id")  %>% 
   ggscatter(., x = "logFC.UPF3B", y = "logFC", color = "NIF_dEJ",
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw()  + 
  geom_xsidedensity(aes(у = after_stat(density), fill = NIF_dEJ), alpha = 0.5, 
                    size = 1,position = "stack") + geom_ysidedensity(aes(x=after_stat(density), 
                                                                         fill = NIF_dEJ), 
                                                                     alpha=0.5, size = 1) 
 + xlab("UPF3B KD (log2FC)")+ ylab("UPF3A KD (log2FC)")), 


(res_dte$UPF3B_KD_Control %>% as.data.frame() %>% 
  dplyr::filter(FDR < 0.05) %>% 
  dplyr::rename("logFC.UPF3B" = "logFC") %>% 
  dplyr::select(tx_id, logFC.UPF3B) %>% 
  inner_join(dplyr::filter(as.data.frame(res_dte$UPF3A_KD_Control), FDR < 0.05), by = "tx_id")  %>% 
   ggscatter(., x = "logFC.UPF3B", y = "logFC", color = "NIF_long_3utr",
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw()  + 
  geom_xsidedensity(aes(у = after_stat(density), fill = NIF_long_3utr), alpha = 0.5, 
                    size = 1,position = "stack") + geom_ysidedensity(aes(x=after_stat(density), 
                                                                         fill = NIF_long_3utr), 
                                                                     alpha=0.5, size = 1) +
   xlab("UPF3B KD (log2FC)") + ylab("UPF3A KD (log2FC)") ), 
ncol=3

)
```


```{r}
res_dte$UPF3B_KD_Control %>% as.data.frame() %>% 
  dplyr::filter(Results != 0) %>% 
  dplyr::rename("logFC.UPF3B" = "logFC") %>% 
  dplyr::select(tx_id, logFC.UPF3B) %>% 
  left_join(dplyr::filter(as.data.frame(res_dte$UPF3A_KD_Control), Results == 0), by = "tx_id")  %>% 
   ggscatter(., x = "logFC.UPF3B", y = "logFC", color= "nif_feature", 
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw()   
```


```{r}
res_dte$UPF3A_KD_Control %>% as.data.frame() %>% 
  dplyr::filter(Results != 0) %>% 
  dplyr::rename("logFC.UPF3A" = "logFC") %>% 
  dplyr::select(tx_id, logFC.UPF3A) %>% 
  left_join(dplyr::filter(as.data.frame(res_dte$UPF3B_KD_Control), Results == 0), by = "tx_id")  %>% 
   ggscatter(., x = "logFC.UPF3A", y = "logFC",
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw()  

```

### What is the distribution between transcripts in UPF3B KD and UPF3A OE? 

```{r}
ggvenn(x[c(1, 4)])
```

```{r}
res_dte$UPF3B_KD_Control %>% as.data.frame() %>% 
  dplyr::filter(Results != 0) %>% 
  dplyr::rename("logFC.UPF3B" = "logFC") %>% 
  dplyr::select(tx_id, logFC.UPF3B) %>% 
  left_join(as.data.frame(res_dte$UPF3A_OE_Control, by = "tx_id")) %>% 
   ggscatter(., x = "logFC.UPF3B", y = "logFC", color = "nif_feature",
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw()  + xlab("UPF3B KD") + ylab("UPF3A OE") + 
  geom_xsidedensity(aes(у = after_stat(density), fill = nif_feature), alpha = 0.5, 
                    size = 1,position = "stack") + geom_ysidedensity(aes(x=after_stat(density), 
                                                                         fill = nif_feature), 
                                                                     alpha=0.5, size = 1) +
  scale_color_tq()  + scale_fill_tq() + geom_hline(yintercept = 0 )
```



** All conditions - transcript and gene level comparison 

** Capacity to do genes with isoforms with different NIFs - try in UPF3B 

** Is NMD a cell-type specific thing? --? Maybe UPF1 work with brain coexp networks 


Any transcript that is expression - 40k 
transcripts that have multiple transcripts and have a look at the CPM - Just do it in UPF3B - average of the three samples
