---
title: "Differential-transcript-expression"
author: "unawaz1996"
date: "2023-03-26"
output:
  html_notebook: default
  workflowr::wflow_html:
    code_folding: hide
editor_options:
  
  chunk_output_type: console
---

Things to do:

* QC plots and normalization
  * Transcript level QC - DONE
  * PCA - DONE
  * BCV and other edgeR associated plots - DONE 
  * bigPint plot interpretations 

* DET analysis - all data
  * MA plots for all analyses - DONE 
  * Log fold change distribution - DONE
  * Ratio of up/down
  * Pvalue histograms - DONE
  * NIF enrichment 
    * How many transcripts that are DE have multiple NIF features - DONE
    * Is NIF enrichment in a given analysis significant  - DONE
    * Number of individual NIF in DET analysis - DONE
    * Distribution of NIFs in DETs - DONE 
  * How many genes have multiple DETs, and how many of those have NIFs?
  
* Overlap of transcripts between UPF3B and UPPF3A and NIF distribution of those alignments? - DONE 
* Overlap of UPF3B sig only transcripts with UPF3B and NIF distribution - DONE
* Overlap of UPF3A sig only transcripts with UPF3A and NIF distribution - DONE
* Overlap of UPF3B KD and UPF3A OE in UPF3B KD and NIF distribution? --> What is the logFC of overlapping transcripts?
* Overlap of UPF3B KD sig only transcripts in UPF3A OE in UPF3B KD and vice versa
  
  

# Introduction


```{r setup, include=FALSE}
knitr::opts_chunk$set(
    autodep = TRUE,
    eval = TRUE,
	echo = FALSE,
	warning = FALSE,
	message = FALSE
)
```

```{r libraries}
source("code/libraries.R")
source("code/functions.R")
library(fishpond)
library(tximeta)
library(GeneOverlap)
library(ggside)
library(tidyquant)
library(stargazer)
library(ggfortify)
library(glue)
library(cowplot)
library(broom)
library(glmpca)
library(naniar)
library(bigPint)
```

## Set-up 


```{r}
ah <- AnnotationHub() %>%
    subset(species == "Mus musculus") %>%
    subset(rdataclass == "EnsDb") %>% 
  subset(genome == "GRCm39")
ensDb <- ah[["AH104895"]]
grTrans <- transcripts(ensDb)
trLengths <- exonsBy(ensDb, "tx") %>%
    width() %>%
    vapply(sum, integer(1))
mcols(grTrans)$length <- trLengths[names(grTrans)]

genesGR = genes(ensDb)
transGR = transcripts(ensDb)

mcols(transGR) = mcols(transGR) %>%
  cbind(
    transcriptLengths(ensDb)[rownames(.), c("nexon", "tx_len")]
  )
```

```{r}
mcols(genesGR) <- mcols(genesGR) %>%
  as.data.frame() %>%
  dplyr::select(
    gene_id, gene_name, gene_biotype, entrezid
  ) %>%
  left_join(
    mcols(transGR) %>%
      as.data.frame() %>%
      mutate(
        tx_support_level = case_when(
          is.na(tx_support_level) ~ 1L, 
          TRUE ~ tx_support_level
        )
      ) %>%
      group_by(gene_id) %>%
      summarise(
        n_tx = n(),
        longest_tx = max(tx_len),
        ave_tx_len = mean(tx_len),
        gc_content = sum(tx_len*gc_content) / sum(tx_len)
      ) %>%
      mutate(
        bin_length = cut(
          x = ave_tx_len,
          labels = seq_len(10),
          breaks = quantile(ave_tx_len, probs = seq(0, 1, length.out = 11)),
          include.lowest = TRUE
        ),
        bin_gc = cut(
          x = gc_content,
          labels = seq_len(10),
          breaks = quantile(gc_content, probs = seq(0, 1, length.out = 11)),
          include.lowest = TRUE
        ),
        bin = paste(bin_gc, bin_length, sep = "_")
      ),
    by = "gene_id"
  ) %>%
  set_rownames(.$gene_id) %>%
  as("DataFrame")

trans2Gene <- mcols(transGR) %>%
    as.data.frame() %>%
    dplyr::select(tx_id, gene_id) %>%
    dplyr::filter(!is.na(tx_id), !is.na(gene_id)) %>%
    as_tibble()

```

```{r}
md.files = "data/LTK_Sample Metafile_V3.txt"
salmon.files = ("/home/neuro/Documents/neurogenetics/alignments/Illumina/RNAseq/mouse/LJ_LTK/Salmon/GR39m_decoy_aware")
salmon = list.files(salmon.files, pattern = "transcripts$", full.names = TRUE)
names(salmon) <- paste(c(212:229))
md = read.table(md.files, header= TRUE) %>%  mutate(files = file.path(salmon, "quant.sf"))
comparisons = unique(md$Group)[-1]
colnames(md)[1] = c("names")
md$names = as.character(md$names)
md$Group <- factor(md$Group, 
                            levels=c("Control","UPF3B_KD", "UPF3A_KD", "UPF3A_OE", 
                                     "UPF3A_KD_UPF3B_KD", "UPF3A_OE_UPF3B_KD"))
```

```{r}
group_cols <- hcl.colors(
  n = length(unique(md$Group)), 
  palette = "Zissou 1"
  ) %>%
  setNames(unique(md$Group))
```


Transcript-level counts were retrieved using the `catchSalmon()` function from the edgeR package. 
Once retrieved, the counts underwent rigourous QC. 

# Removal of outlier sample 

According to the quality control analysis, it was revealed that one of the samples was not clustering with its condition group. Investigation into GC content, transcript length and library size revealed no contribution of these factors. To ensure we are getting results that are not skewed we will remove the sample from the transcript level analysis. 


```{r include=FALSE}
catch <- catchSalmon(salmon)
colnames(catch$counts) =  paste(c(212:229))

cts.scaled = catch$counts/catch$annotation$Overdispersion

#md_filt = md %>% dplyr::filter(names != "223")


dge <- DGEList(counts = cts.scaled, 
               genes=catch$annotation, 
               samples = md,
               group = md$Group)

keep <- filterByExpr(dge)
dge.scaled.filtr <- dge[keep, , keep.lib.sizes = FALSE]
dge.scaled.filtr <- calcNormFactors(dge.scaled.filtr)


```

```{r fig.height = 7, fig.width=10, fig.cap="*Principal component analysis of transcript level data. PCA was performed on log2 transformed CPM after filtering lowly expressed genes. The PCA shows that samples are clustering close to their conditions based on PC1, however one of the samples of the UPF3A OE in UPF3B KD cell line (sample 223), seems to deviate from its condition group and the rest of the data, so needs to be further investigated*"}
 pca = cpm(dge.scaled.filtr, log = TRUE) %>% 
  t() %>% 
    prcomp(scale = TRUE) %>% 
  tidy() %>% 
  dplyr::rename(names = row) %>% 
  left_join(dge.scaled.filtr$samples, by = "names") %>%
   dplyr::filter(PC %in% 1:3) %>% 
  pivot_wider(names_from = "PC", names_prefix = "PC", values_from = "value") %>% 
  ggplot(
    aes(PC1, PC2, colour = Group, size = lib.size/1e6)
  ) +
  geom_point() +
  geom_text_repel(aes(label = names), show.legend = FALSE) +
  scale_colour_manual(values = group_cols, 
                      labels = c("UPF3A_KD_UPF3B_KD" = "UPF3 dKD", "UPF3A_KD" = "UPF3A KD", 
                                                                   "UPF3B_KD" = "UPF3B KD", "UPF3A_OE" = "UPF3A OE", 
                                                                   "UPF3A_OE_UPF3B_KD" = "UPF3A OE UPF3B KD")) +
  #scale_size_continuous(limits = c(5, 15), breaks = seq(5, 15, by = 5)) +
  theme_bw() + ggtitle("Transcript Level")



pca
  
```


```{r fig.height=5, fig.width=5}
 plotMDS(dge.scaled.filtr,col=as.numeric(dge.scaled.filtr$samples$group))
```



# Analysis

```{r include=FALSE}
catch <- catchSalmon(salmon[-12])
colnames(catch$counts) =  paste(c(212:222, 224:229))

cts.scaled = catch$counts/catch$annotation$Overdispersion

md_filt = md %>% dplyr::filter(names != "223")


dge <- DGEList(counts = cts.scaled, 
               genes=catch$annotation, 
               samples = md_filt,
               group = md_filt$Group)

keep <- filterByExpr(dge)
dge.scaled.filtr <- dge[keep, , keep.lib.sizes = FALSE]
dge.scaled.filtr <- calcNormFactors(dge.scaled.filtr)


cpm(dge.scaled.filtr, log = TRUE) %>% 
  t() %>% 
    prcomp(scale = TRUE) %>% 
  tidy() %>% 
  dplyr::rename(names = row) %>% 
  left_join(dge.scaled.filtr$samples, by = "names") %>%
   dplyr::filter(PC %in% 1:3) %>% 
  pivot_wider(names_from = "PC", names_prefix = "PC", values_from = "value") %>% 
  ggplot(
    aes(PC1, PC2, colour = Group, size = lib.size/1e6)
  ) +
  geom_point() +
  geom_text_repel(aes(label = names), show.legend = FALSE) +
  scale_colour_manual(values = group_cols, 
                      labels = c("UPF3A_KD_UPF3B_KD" = "UPF3 dKD", "UPF3A_KD" = "UPF3A KD", 
                                                                   "UPF3B_KD" = "UPF3B KD", "UPF3A_OE" = "UPF3A OE", 
                                                                   "UPF3A_OE_UPF3B_KD" = "UPF3A OE UPF3B KD")) +
  #scale_size_continuous(limits = c(5, 15), breaks = seq(5, 15, by = 5)) +
  theme_bw() + ggtitle("Transcript Level")
  

```

```{r}
plotMDS(dge.scaled.filtr,col=as.numeric(dge.scaled.filtr$samples$group))
#legend("bottomleft", as.character(unique(dge.scaled.filtr$samples$group)), col=1:3, pch=20)

```


```{r fig.height=6, fig.width=6, fig.cap= "*Biological coeficcient of variation plot against the average abundance of each transcript. The plot shows the square-root estimates of the common, trended and tagwise NB dispersions.*"}
design <-  model.matrix(~0+Group, data = md_filt) %>% 
    set_colnames(gsub(pattern = "Group", replacement ="", x = colnames(.)))

dge.scaled.filtr <- estimateDisp(dge.scaled.filtr,design)

dge.scaled.filtr$common.dispersion

plotBCV(dge.scaled.filtr)
```

```{r fig.height=6, fig.width=6, fig.cap="*Quasi-likelihood dispersion aganist gene abundance. Estimates are shown for raw, tended and squeezed dispersions*"}
fit <- glmQLFit(dge.scaled.filtr,design)
plotQLDisp(fit)
```

```{r}
summary(fit$df.prior)
```


```{r}
contrasts <- makeContrasts(levels = colnames(design), 
                           UPF3B_KD_Control = UPF3B_KD - Control, 
                           UPF3A_KD_Control = UPF3A_KD - Control, 
                           UPF3A_OE_Control = UPF3A_OE - Control, 
                           UPF3A_KD_UPF3B_KD_Control = UPF3A_KD_UPF3B_KD - Control, 
                           UPF3A_OE_UPF3B_KD_Control = UPF3A_OE_UPF3B_KD - Control)


dgeLRT_obj = list()
res_dte = list()
metrics = list()
for (i in 1:ncol(contrasts)){
  temp = glmQLFTest(fit, contrast= contrasts[,i])
  tt = topTags(temp,n = Inf)
  tt %<>% as.data.frame() %>% rownames_to_column("tx_id") %>%
    left_join(decideTestsDGE(temp) %>%
                  as.data.frame() %>%
                  rownames_to_column("tx_id"))
  res_dte[[paste0(colnames(contrasts)[i])]] = tt
  dgeLRT_obj[[paste0(colnames(contrasts)[i])]] = temp
  
  setDT(tt)[]
  tt %<>% dplyr::rename("ID" = "tx_id")
  tt <- as.data.frame(tt)
  metrics[[paste0(colnames(contrasts)[i])]] = tt
}
```

## Checking normalization and QC


```{r}
scaled_matrix <- as.data.frame(t(apply(as.matrix(dge.scaled.filtr$counts), 1, scale)))
#scaled_matrix <- as.character(rownames(dge.scaled.filtr$counts))
colLength = length(scaled_matrix)
scaled_matrix<- scaled_matrix[,c(colLength, 1:(colLength-1))]
colnames(scaled_matrix) <- colnames(dge.scaled.filtr$counts)
nID <- which(is.nan(scaled_matrix[,2]))
scaled_matrix[nID,2:length(scaled_matrix)] <- 0
```

```{r}
scaled_matrix %<>% rownames_to_column("ID")

counts = log(dge.scaled.filtr$counts + 0.05)
counts  %<>% as.data.frame() %>%rownames_to_column("ID")
colnames(counts) =c("ID", "Control.1", "UPF3AKD.1", "UPF3AOE.1", "UPF3BKD.1", "UPF3AKDUPF3BKD.1", "UPF3AOEUPF3BKD.1",
                    "Control.2", "UPF3AKD.2", "UPF3AOE.2", "UPF3BKD.2", "UPF3AKDUPF3BKD.2",
                    "Control.3", "UPF3AKD.3", "UPF3AOE.3", "UPF3BKD.3", "UPF3AKDUPF3BKD.3", "UPF3AOEUPF3BKD.2")

names(metrics) = c("UPF3BKD_Control", "UPF3AKD_Control","UPF3AOE_Control", "UPF3AKDUPF3BKD_Control", "UPF3AOEUPF3BKD_Control")
ret = plotSM(counts[c(1:2, 5,8,11,13,16)], dataMetrics = metrics[1], saveFile = FALSE)
ret 
```

```{r}
upf3a_oe = plotSM(counts[c(1:2, 4,8,10,13,15)], dataMetrics = metrics[3], saveFile = FALSE)
upf3a_oe
```

# Distribution of results 
```{r}
plot_hist = function(df, comparison, legend = FALSE){
  ggplot(data= df, aes(x=PValue)) + geom_histogram(fill = "grey", color="black") + theme_bw() + xlab("Pvalue") + ylab("Frequency")
}
plot_fc = function(df){
  df %>% 
    dplyr::filter(FDR < 0.05) %>%  
    ggplot(aes(x=logFC)) + 
    geom_histogram(aes(y=..density..), colour="black", fill="white") +  
    geom_density(alpha=.2, fill="#FF6666") + geom_vline(xintercept =c(log2(1.5), -log2(1.5)),color="red", linetype = "dashed")
}
```

## Fold change distributions 

```{r}
a_f = plot_fc(res_dte$UPF3B_KD_Control)
b_f = plot_fc(res_dte$UPF3A_KD_Control)
c_f = plot_fc(res_dte$UPF3A_KD_UPF3B_KD_Control)
d_f = plot_fc(res_dte$UPF3A_OE_Control)
e_f = plot_fc(res_dte$UPF3A_OE_UPF3B_KD_Control)
```


```{r fig.height=10}
a = plot_hist(res_dte$UPF3B_KD_Control) + ggtitle("UPF3B KD vs Control")
b = plot_hist(res_dte$UPF3A_KD_Control) + ggtitle("UPF3A KD vs Control")
c = plot_hist(res_dte$UPF3A_KD_UPF3B_KD_Control) + ggtitle("UPF3 dKD vs Control")
d =  plot_hist(res_dte$UPF3A_OE_Control) + ggtitle("UPF3A OE vs Control")
e =  plot_hist(res_dte$UPF3A_OE_UPF3B_KD_Control) + ggtitle("UPF3A OE UPF3B KD vs Control")

#grid.arrange(a,b,c,d,e, ncol = 3, nrow=2)


grid.arrange(a, a_f,
             b,b_f, 
             c, c_f, 
             d, d_f, 
             e,e_f, ncol = 2, nrow=5)
```


```{r}
## Ma plot 

ma_1 = ma_plot(res_dte$UPF3B_KD_Control, 0) + ggtitle("UPF3B KD vs Control")
ma_2 = ma_plot(res_dte$UPF3A_KD_Control, 0) + ggtitle("UPF3A KD vs Control")
ma_3 = ma_plot(res_dte$UPF3A_KD_UPF3B_KD_Control, 0) + ggtitle("UPF3 dKD vs Control")
ma_4 = ma_plot(res_dte$UPF3A_OE_Control, 0) + ggtitle("UPF3A OE vs Control")
ma_5 = ma_plot(res_dte$UPF3A_OE_UPF3B_KD_Control, 0) + ggtitle("UPF3A OE in UPF3B KD")

grid.arrange(ma_1, ma_2, ma_3, 
            ma_4, ma_5, ncol=3, nrow=2)
```

```{r}
# ratios = data.frame()
# for (c in names(res_dte)){
#   temp = table(res_dte[[c]]$) %>% 
#     melt() %>% 
#     dplyr::filter(Var2 == c, Var1 != "NotSig") 
#   ratio= temp$value[2]/temp$value[1]
#   ratio = data.frame(Comparisons = c, 
#            ratio = ratio)
#   ratios = rbind(ratios, ratio)
#   }
# ratios

```

## DET statistics 

```{r}
NMD_features = read.table("data/nif_output/Mus_musculus__nifs.tsv", sep = "\t", skip =1)

colnames(NMD_features) = c("tx_id",	"NIF_dEJ", "NIF_uORF",	"NIF_long_3utr", "NIF|__|GC(cds[-10:termination_codon])",	"GC_5utr",	"GC_cds",	"GC_3utr",	"GC_exons",	"GC_introns",	"LENGTH_5utr", 	"LENGTH_cds",	"LENGTH_3utr",	"LENGTH_exons", "LENGTH_introns", "EXONS_count",	"dEJs|__|penultimate-exon>=55(nts)",	"uORFs|__|count",	"uORFs|__|gc-content",	"uORFs|__|lengths", "uORFs|__|max-length")

NMD_features$NIF_dEJ = ifelse(NMD_features$NIF_dEJ == "True",gsub("True", 1, NMD_features$NIF_dEJ), 0)
NMD_features$NIF_uORF = ifelse(NMD_features$NIF_uORF == "True",gsub("True", 1, NMD_features$NIF_uORF), 0)
NMD_features$NIF_long_3utr = ifelse(NMD_features$NIF_long_3utr == "True",gsub("True", 1, NMD_features$NIF_long_3utr), 0)
NMD_features %<>% mutate_at(.vars = "tx_id", .funs = gsub, pattern = "\\.[0-9]*$", replacement = "")
```


```{r include=FALSE}
total = data.frame()
for (i in names(res_dte)){
    name = i
    #comparison = rep(name, 4) %>% as.data.frame()
    temp = res_dte[[i]]
    
    message("Merging DE results with NMD table for", paste(name))
    temp %<>%  as.data.frame() %>%
      mutate(tx_version = tx_id) %>%
      mutate_at(.vars = "tx_id", .funs = gsub, pattern = "\\.[0-9]*$", replacement = "") %>% 
      left_join(NMD_features, by = "tx_id")
    
    message("Summing all NMD values to determine number of NIFs per transcript for", paste0(name))
    temp$nif_feature = ifelse(rowSums(sapply(temp[, c(12:14)],
                       function(x) as.numeric(as.character(x)))) == 2 ,"Two",
                       ifelse(rowSums(sapply(temp[, c(12:14)],
                              function(x) as.numeric(as.character(x)))) == 3, "Three", 
                              ifelse(rowSums(sapply(temp[, c(12:14)],
                                                    function(x) as.numeric(as.character(x)))) == 1, "One", "None")))
    
    temp$nif_feature %<>%  replace_na("None")
    res_dte[[paste(name)]] = temp
    
    ## downregulation 
    message("Subsetting downregulated genes for ", paste0(name))
    down = temp %>%  as.data.frame() %>%  
        dplyr::filter(FDR < 0.05 & logFC < 0) 
    down = table(down$nif_feature)
    print(nrow(down))
    repetition = down %>% nrow() %>% as.integer()
    expression = rep(c("down"), repetition) %>% as.data.frame()
    comparison = rep(name, nrow(expression)) %>% as.data.frame()
    down = cbind(down, expression = expression)
    down = cbind(down, comparison = comparison)
    colnames(down) = c("NIF", "Value", "Expression", "Condition")
    
    ## upregulation 
    message("Subsetting upregulated genes for ", paste0(name))
    up = temp %>%  as.data.frame() %>%  
        dplyr::filter(FDR < 0.05 & logFC > 0 ) 
    up = table(up$nif_feature)
    print(dim(up))
    repetition = up %>% nrow() %>% as.integer()
    expression = rep(c("up"), repetition) %>% as.data.frame()
    comparison = rep(name, nrow(expression)) %>% as.data.frame()
    up = cbind(up, expression = expression)
    up = cbind(up, comparison = comparison)
    colnames(up) = c("NIF", "Value", "Expression", "Condition")
    
    ## bind everything together 
    message("Binding everything together")
    #together = rbind(up, down)
   
    #total = rbind(total,as.data.frame(up,down))
    total = rbind(total,up,down, stringsAsFactors = FALSE)
}
```

```{r fig.width=10}
total$NIF <- factor(total$NIF, levels = c("Three", "Two", "One", "None"))
up = total %>% 
  dplyr::filter(Expression == "up") %>% 
  ggplot(aes(x= Condition, y = Value, fill=NIF)) + geom_bar(stat = "identity", width = .5,color = "white") + coord_flip() +
  theme_bw() + ylab("Number of Transcripts") + xlab("") + 
  scale_fill_manual(values = c("#FF8DD0","#E0155F", "#C40C71", "#370059")) +
  theme(axis.text.x = element_text(size = 10, face = "bold"),
        legend.position = "none") + ylim(0, 4500)

#ggsave(file = "/home/neuro/Documents/NMD_analysis/Analysis/NMD-analysis/output/Transcript/Thesis_figures/total_exp_up.svg", plot = up, 
 #      height = 6.31, width = 5.77, units = "in")


down = total %>% 
  dplyr::filter(Expression == "down") %>% 
  ggplot(aes(x= Condition, y = Value, fill=NIF)) + 
  geom_bar(stat = "identity", width = .5,color = "white") + coord_flip() +
  theme_bw() + ylab("Number of Transcripts") + xlab("") + 
  scale_fill_manual(values = c("#FF8DD0","#E0155F", "#C40C71", "#370059")) +
  theme(axis.text.x = element_text(size = 10, face = "bold"), 
        legend.position = "none") + scale_y_reverse(limits=c(4500,0)) 



ggarrange(down, up)
#ggsave(file = "/home/neuro/Documents/NMD_analysis/Analysis/NMD-analysis/output/Transcript/Thesis_figures/total_exp_down.svg", plot = down, 
#       height = 6.31, width = 5.77, units = "in")
```


```{r}
up = total %>% 
  dplyr::filter(Expression == "up") %>% 
  ggplot(aes(x= Condition, y = Value, fill=NIF)) + geom_bar(position ="fill", stat = "identity",color = "white") + coord_flip() +
  theme_bw() + ylab("Proportion of Upregulated Transcripts") + xlab("") + 
  scale_fill_manual(values = c("#FF8DD0","#E0155F", "#C40C71", "#370059")) +
  theme(axis.text.x = element_text(size = 10, face = "bold"),
        legend.position = "none") 
down = total %>% 
  dplyr::filter(Expression == "down") %>% 
  ggplot(aes(x= Condition, y = Value, fill=NIF)) + 
  geom_bar(position="fill", stat="identity",color = "white") + coord_flip() +
  theme_bw() + ylab("Proportion of Downregulated Transcripts ") + xlab("") + 
  scale_fill_manual(values = c("#FF8DD0","#E0155F", "#C40C71", "#370059")) +
  theme(axis.text.x = element_text(size = 10, face = "bold"), 
        legend.position = "none") + scale_y_reverse()

ggarrange(down, up)
```


```{r}
nif_features = list()
for (i in names(res_dte)){
  name = i
  temp = res_dte[[i]]
  sig = temp %>%  as.data.frame() %>%  
        dplyr::filter(FDR < 0.05) 
  utr = table(sig$NIF_long_3utr)[2]
  dej = table(sig$NIF_dEJ)[2]
  uorf = table(sig$NIF_uORF)[2]
  total_nif = rbind(utr, dej, uorf) %>% 
    as.data.frame() %>% 
    rownames_to_column("NIF_feature") %>% 
    mutate(Condition = name) %>% 
    dplyr::rename("Count" = "1")
  nif_features= rbind(nif_features, total_nif, stringsAsFactors = FALSE)
  
    
  
}

nif_features %>%
  ggplot(aes(x= NIF_feature, y = Count, fill = NIF_feature)) + 
  geom_bar(stat = "identity", position = "dodge", width = .5) + coord_flip() + 
  facet_grid(~Condition) + theme_bw()


```

```{r}
# Calculating enrichment for this work 
### Ask Irina about adjusted pvalues 
enrichment_NIF = list()
for (c in names(res_dte)){
  test.list = list()
  test.list$BG = res_dte[[c]]$tx_id
  test.list$NIF = res_dte[[c]]$tx_id[res_dte[[c]]$nif_feature != "None"]
  test.list$sig = res_dte[[c]]$tx_id[res_dte[[c]]$FDR < 0.05]
  test.list$up = res_dte[[c]]$tx_id[res_dte[[c]]$FDR < 0.05 & res_dte[[c]]$logFC > 0]
  test.list$sig = res_dte[[c]]$tx_id[res_dte[[c]]$FDR < 0.05 & res_dte[[c]]$logFC < 0]
  
  for (det in c("sig", "up", "down")) {
    go.obj <- newGeneOverlap(test.list[[det]],
                         test.list$NIF,
                         genome.size=length(test.list$BG))
    go.obj <- testGeneOverlap(go.obj)
    enrichment_NIF[[paste0(c, "_", det)]] = go.obj@pval
    
  }
}

enrichment_NIF = do.call(rbind, enrichment_NIF) %>% as.data.frame()
colnames(enrichment_NIF)[1] = c("pvalue")
enrichment_NIF$padj = p.adjust(enrichment_NIF$pvalue, method = "BH")

```

```{r}
enrichment_nif_feature = list()
nifs_list = c("NIF_uORF", "NIF_dEJ", "NIF_utr")

for (c in names(res_dte)){
  test.list = list()
  test.list$BG = res_dte[[c]]$tx_id
  test.list$NIF_uORF = res_dte[[c]]$tx_id[res_dte[[c]]$NIF_uORF== 1 ]
  test.list$NIF_dEJ = res_dte[[c]]$tx_id[res_dte[[c]]$NIF_dEJ== 1 ]
  test.list$NIF_utr = res_dte[[c]]$tx_id[res_dte[[c]]$NIF_long_3utr== 1 ]
  test.list$sig = res_dte[[c]]$tx_id[res_dte[[c]]$FDR < 0.05]
  test.list$up = res_dte[[c]]$tx_id[res_dte[[c]]$FDR < 0.05 & res_dte[[c]]$logFC > 0]
  test.list$down = res_dte[[c]]$tx_id[res_dte[[c]]$FDR < 0.05 & res_dte[[c]]$logFC < 0]
  for (nif in nifs_list){
     go.obj <- newGeneOverlap(test.list$sig,
                         test.list[[nif]],
                         genome.size=length(test.list$BG))
     go.obj <- testGeneOverlap(go.obj)
     enrichment_nif_feature[[paste0(c, "_", nif)]] = go.obj@pval
     go.obj <- newGeneOverlap(test.list$up,
                         test.list[[nif]],
                         genome.size=length(test.list$BG))
     go.obj <- testGeneOverlap(go.obj)
     enrichment_nif_feature[[paste0(c,"_",nif,  "_up")]] = go.obj@pval
     go.obj <- newGeneOverlap(test.list$down,
                         test.list[[nif]],
                         genome.size=length(test.list$BG))
     go.obj <- testGeneOverlap(go.obj)
     enrichment_nif_feature[[paste0(c,"_",nif, "_down")]] = go.obj@pval
       
    
  }
}

```

## Distribution of NIFs
```{r include=FALSE}
upf3b = NIF_dist(res_dte$UPF3B_KD_Control) 
upf3a = NIF_dist(res_dte$UPF3A_KD_Control) 
dkd = NIF_dist(res_dte$UPF3A_KD_UPF3B_KD_Control) 
upf3a_oe = NIF_dist(res_dte$UPF3A_OE_Control) 
upf3a_oe_b_kd =NIF_dist(res_dte$UPF3A_OE_UPF3B_KD_Control) 
```

```{r}
grid.arrange(upf3b, upf3a, 
             dkd, upf3a_oe, 
             upf3a_oe_b_kd, ncol = 5)

```

```{r}
## Add gene ensembl IDs 
## See how many DETs belong to > 1 gene
## How many of those have NIFs 
```





## Are DET shared between UPF3B KD and UPF3A KD? 

```{r}
res_dte$UPF3B_KD_Control %>% as.data.frame %>%
  dplyr::filter(FDR < .05) %>% 
  with(., 
       table(sign.lfc=sign(logFC)))
```

```{r}

res_dte$UPF3A_KD_Control %>% as.data.frame %>%
  dplyr::filter(FDR < .05) %>% 
  with(., 
       table(sign.lfc=sign(logFC)))

```

```{r}
x = list( "UPF3B KD" = res_dte$UPF3B_KD_Control$tx_id[res_dte$UPF3B_KD_Control$FDR < 0.05],  
    "UPF3A KD" = res_dte$UPF3A_KD_Control$tx_id[res_dte$UPF3A_KD_Control$FDR < 0.05], 
    "Double KD" = res_dte$UPF3A_KD_UPF3B_KD_Control$tx_id[res_dte$UPF3A_KD_UPF3B_KD_Contro$FDR < 0.05],
    "UPF3A OE" = res_dte$UPF3A_OE_Control$tx_id[res_dte$UPF3A_OE_Control$FDR < 0.05], 
    "UPF3A OE UPF3B KD" = res_dte$UPF3A_OE_UPF3B_KD_Control$tx_id[res_dte$UPF3A_OE_UPF3B_KD_Control$FDR < 0.05])

ggvenn(x[c(1,2)])
```

```{r fig.height=5, fig.width=12}
grid.arrange((res_dte$UPF3B_KD_Control %>% as.data.frame() %>% 
  dplyr::filter(FDR < 0.05) %>% 
  dplyr::rename("logFC.UPF3B" = "logFC") %>% 
  dplyr::select(tx_id, logFC.UPF3B) %>% 
  inner_join(dplyr::filter(as.data.frame(res_dte$UPF3A_KD_Control), FDR < 0.05), by = "tx_id")  %>% 
   ggscatter(., x = "logFC.UPF3B", y = "logFC", color = "NIF_uORF",
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw()  + 
  geom_xsidedensity(aes(у = after_stat(density), fill = NIF_uORF), alpha = 0.5, 
                    size = 1,position = "stack") + geom_ysidedensity(aes(x=after_stat(density), 
                                                                         fill = NIF_uORF), 
                                                                     alpha=0.5, size = 1) +
  scale_color_tq()  + scale_fill_tq() +  xlab("UPF3B KD (log2FC)") + ylab("UPF3A KD (log2FC)")),

(res_dte$UPF3B_KD_Control %>% as.data.frame() %>% 
  dplyr::filter(FDR < 0.05) %>% 
  dplyr::rename("logFC.UPF3B" = "logFC") %>% 
  dplyr::select(tx_id, logFC.UPF3B) %>% 
  inner_join(dplyr::filter(as.data.frame(res_dte$UPF3A_KD_Control), FDR < 0.05), by = "tx_id")  %>% 
   ggscatter(., x = "logFC.UPF3B", y = "logFC", color = "NIF_dEJ",
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw()  + 
  geom_xsidedensity(aes(у = after_stat(density), fill = NIF_dEJ), alpha = 0.5, 
                    size = 1,position = "stack") + geom_ysidedensity(aes(x=after_stat(density), 
                                                                         fill = NIF_dEJ), 
                                                                     alpha=0.5, size = 1) +
  scale_color_tq()  + scale_fill_tq() + xlab("UPF3B KD (log2FC)")+ ylab("UPF3A KD (log2FC)")), 


(res_dte$UPF3B_KD_Control %>% as.data.frame() %>% 
  dplyr::filter(FDR < 0.05) %>% 
  dplyr::rename("logFC.UPF3B" = "logFC") %>% 
  dplyr::select(tx_id, logFC.UPF3B) %>% 
  inner_join(dplyr::filter(as.data.frame(res_dte$UPF3A_KD_Control), FDR < 0.05), by = "tx_id")  %>% 
   ggscatter(., x = "logFC.UPF3B", y = "logFC", color = "NIF_long_3utr",
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw()  + 
  geom_xsidedensity(aes(у = after_stat(density), fill = NIF_long_3utr), alpha = 0.5, 
                    size = 1,position = "stack") + geom_ysidedensity(aes(x=after_stat(density), 
                                                                         fill = NIF_long_3utr), 
                                                                     alpha=0.5, size = 1) +
  scale_color_tq()  + scale_fill_tq() + xlab("UPF3B KD (log2FC)") + ylab("UPF3A KD (log2FC)") ), 
ncol=3

)
```


```{r}
res_dte$UPF3B_KD_Control %>% as.data.frame() %>% 
  dplyr::filter(`-1*Control 1*UPF3B_KD` != 0) %>% 
  dplyr::rename("logFC.UPF3B" = "logFC") %>% 
  dplyr::select(tx_id, logFC.UPF3B) %>% 
  left_join(dplyr::filter(as.data.frame(res_dte$UPF3A_KD_Control), `-1*Control 1*UPF3A_KD` == 0), by = "tx_id")  %>% 
   ggscatter(., x = "logFC.UPF3B", y = "logFC", color= "nif_feature", 
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw()   
```


```{r}
res_dte$UPF3A_KD_Control %>% as.data.frame() %>% 
  dplyr::filter(`-1*Control 1*UPF3A_KD` != 0) %>% 
  dplyr::rename("logFC.UPF3A" = "logFC") %>% 
  dplyr::select(tx_id, logFC.UPF3A) %>% 
  left_join(dplyr::filter(as.data.frame(res_dte$UPF3B_KD_Control), `-1*Control 1*UPF3B_KD` == 0), by = "tx_id")  %>% 
   ggscatter(., x = "logFC.UPF3A", y = "logFC",
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw()  

```

### What is the distribution between transcripts in UPF3B KD and UPF3A OE? 

```{r}
ggvenn(x[c(1, 4)])
```

```{r}
res_dte$UPF3B_KD_Control %>% as.data.frame() %>% 
  dplyr::filter(`-1*Control 1*UPF3B_KD` != 0) %>% 
  dplyr::rename("logFC.UPF3B" = "logFC") %>% 
  dplyr::select(tx_id, logFC.UPF3B) %>% 
  left_join(as.data.frame(res_dte$UPF3A_OE_Control, by = "tx_id")) %>% 
   ggscatter(., x = "logFC.UPF3B", y = "logFC", color = "nif_feature",
              cor.coef = TRUE, add = "reg.line", 
              conf.int = TRUE, add.params = list(color = "blue",
                                                 fill = "lightgray")) + 
    theme_bw()  + xlab("UPF3B KD") + ylab("UPF3A OE") + 
  geom_xsidedensity(aes(у = after_stat(density), fill = nif_feature), alpha = 0.5, 
                    size = 1,position = "stack") + geom_ysidedensity(aes(x=after_stat(density), 
                                                                         fill = nif_feature), 
                                                                     alpha=0.5, size = 1) +
  scale_color_tq()  + scale_fill_tq() + geom_hline(yintercept = 0 )
```

<!-- # ```{r} -->
<!-- #  ## Create manual transcriptome annotation -->
<!-- # indexDir = file.path("/home/neuro/Documents/NMD_analysis/Analysis/Genome/for_salmon/April_2023/decoy_awaresalmon_index/") -->
<!-- # fasta = file.path("/home/neuro/Documents/NMD_analysis/Analysis/Genome/for_salmon/Mus_musculus.GRCm39.cdna.all.fa") -->
<!-- # gtf = file.path("/home/neuro/Documents/NMD_analysis/Analysis/Genome/Mus_musculus.GRCm39.107.gtf.gz") -->
<!-- # tmp <- tempdir() -->
<!-- # jsonFile <- file.path(tmp, paste0(basename(indexDir), ".json")) -->
<!-- #  -->
<!-- # makeLinkedTxome(indexDir=indexDir, -->
<!-- #                 source="Ensembl", organism="Mus musculus", -->
<!-- #                 release="107", genome="GRCm39.107", -->
<!-- #                 fasta=fasta, gtf=gtf, -->
<!-- #                 jsonFile=jsonFile) -->
<!-- #  -->
<!-- # ``` -->

<!-- ```{r} -->
<!-- se <- tximeta(md) -->
<!-- keep = (rowSums(se@assays@data$abundance >= .5) >= 3) -->
<!-- dte = se@assays@data$counts[keep,] %>% -->
<!--   as.data.frame %>% -->
<!--   rownames_to_column("tx_id") %>% -->
<!--   set_colnames(basename(colnames(.))) %>% -->
<!--   dplyr::filter(tx_id %in% trans2Gene$tx_id) %>% melt() -->
<!--   #pivot_longer(cols = all_of(md$names), names_to = "Mm", values_to = "count") %>% -->
<!--     left_join(trans2Gene) %>% -->
<!--     group_by(variable, gene_id) %>% -->
<!--     summarise(count = sum(value)) %>% -->
<!--   pivot_wider(names_from = "Mm", values_from = "count") %>% -->
<!--     dplyr::filter(grepl("ENSMUS", gene_id)) %>% -->
<!--     as.data.frame() %>% -->
<!--     column_to_rownames("gene_id") %>% -->
<!--     DGEList() -->
<!-- dge$samples %<>% -->
<!--     mutate(names = rownames(.)) %>% -->
<!--   dplyr::select(-group) %>% -->
<!--     left_join(md, by = "names") %>% -->
<!--     set_rownames(.$names) -->
<!-- dge$genes <- genesGR[rownames(dge)] %>% -->
<!--   mcols() -->

<!-- ``` -->




<!-- # ```{r transcript-pca} -->
<!-- #  -->
<!-- # pca = log(assays(se)$abundance[keep,] + 0.05) %>% -->
<!-- #   t() %>%  -->
<!-- #   prcomp(scale = FALSE) -->
<!-- #  -->
<!-- # showLabel = nrow(md) <= 18 -->
<!-- # trans_pca =  pca %>%  -->
<!-- #   tidy() %>%  -->
<!-- #   dplyr::rename(names = row) %>%  -->
<!-- #   left_join(dte$samples, by = "names") %>% -->
<!-- #    dplyr::filter(PC %in% 1:3) %>%  -->
<!-- #   pivot_wider(names_from = "PC", names_prefix = "PC", values_from = "value") %>%  -->
<!-- #   ggplot( -->
<!-- #     aes(PC1, PC2, colour = Group, size = lib.size/1e6) -->
<!-- #   ) + -->
<!-- #   geom_point() + -->
<!-- #   geom_text_repel(aes(label = names), show.legend = FALSE) + -->
<!-- #   scale_colour_manual(values = group_cols,  -->
<!-- #                       labels = c("UPF3A_KD_UPF3B_KD" = "Condition 4", "UPF3A_KD" = "Condition 2",  -->
<!-- #                                                                    "UPF3B_KD" = "Condition 1", "UPF3A_OE" = "Condition 3",  -->
<!-- #                                                                    "UPF3A_OE_UPF3B_KD" = "Condition 5")) + -->
<!-- #   #scale_size_continuous(limits = c(5, 15), breaks = seq(5, 15, by = 5)) + -->
<!-- #   labs( -->
<!-- #     x = glue("PC1 ({percent(pca$sdev[[1]]^2 / sum(pca$sdev^2), 0.1)})"), -->
<!-- #     y = glue("PC2 ({percent(pca$sdev[[2]]^2 / sum(pca$sdev^2), 0.1)})"), -->
<!-- #     colour = "Group", -->
<!-- #     size = "Library Size\n(millions)" -->
<!-- #   ) + theme_bw() + ggtitle("Transcript Level") -->
<!-- # ``` -->


<!-- # ```{r} -->
<!-- # se <- tximeta(md) -->
<!-- #  -->
<!-- # ## do manual txi metadata load with Salmon index with decoy aware  -->
<!-- #  -->
<!-- # y <- scaleInfReps(se) -->
<!-- # keep = (rowSums(se@assays@data$abundance >= .5) >= 3) -->
<!-- # y = y[keep,] -->
<!-- #  -->
<!-- # set.seed(1) -->
<!-- #  -->
<!-- # results = list() -->
<!-- # results_obj = list() -->
<!-- #  -->
<!-- # for (c in comparisons){ -->
<!-- #   comp_mtx = y[, y$Group %in% c("Control", c)] -->
<!-- #   comp_mtx$Group = droplevels(comp_mtx$Group) -->
<!-- #   comp_mtx = swish(comp_mtx, x="Group") -->
<!-- #   results[[paste0(c)]] = mcols(comp_mtx) -->
<!-- #   results_obj[[paste0(c)]] = comp_mtx -->
<!-- # } -->
<!-- #  -->
<!-- #  -->
<!-- # hist(results$UPF3A_OE$pvalue) -->
<!-- # ``` -->


<!-- ```{r} -->
<!-- #res_trans = results$UPF3B_KD %>% as.data.frame %>% -->
<!--  # dplyr::filter(qvalue < .05 & abs(log2FC) > log2(1.5)) -->
<!-- #res_DTU = ASwitchListAnalyzed$isoformFeatures %>%  -->
<!-- #  dplyr::filter(condition_2 == "UPF3B_KD" & isoform_switch_q_value < 0.05 & abs(dIF) > 0.1 ) %>%  -->
<!-- #  mutate(ens_gene = mapIds(org.Mm.eg.db, keys=.$gene_name, -->
<!-- #                             column="ENSEMBL",keytype="SYMBOL", -->
<!-- #                         multiVals="first")) -->
<!-- #x = list(DEG = DEGenes$UPF3B_vs_Control$ensembl_gene_id,  -->
<!-- #         DET = res_trans$gene_id, -->
<!-- #         DTU = unname(res_DTU$ens_gene)) -->
<!-- #ggvenn(x[c(1:3)]) -->
<!-- ``` -->

<!-- ```{r} -->

<!-- #left_join(tx2gene, by = "tx_id") -->
<!-- #ntxInfo  <- df %>% -->
<!-- #    dplyr::distinct(gene_id,.keep_all = TRUE)   -->
<!-- #biotypeCount <- df %>% dplyr::group_by(Filter) %>% -->
<!-- #    dplyr::count(tx_biotype) %>% -->
<!-- #    mutate(sum=sum(n),frac = n / sum(n))  -->

<!-- ``` -->

<!-- ```{r} -->
<!-- #DEGenes$UPF3B_vs_Control %<>%  -->
<!-- #  dplyr::rename("gene_id" = "ensembl_gene_id") -->

<!-- #in_all = calculate.overlap(x = list(DEG = DEGenes$UPF3B_vs_Control$ensembl_gene_id,  -->
<!-- #         DET = res_trans$gene_id, -->
<!-- #         DTU = unname(res_DTU$ens_gene))) -->
<!-- #length(in_all$a5) -->

<!-- #res_trans %>%  -->
<!-- #  dplyr::filter(gene_id %in% in_all$a5)  -->

<!-- #DEGenes$UPF3B_vs_Control %>%  -->
<!-- #  dplyr::filter(gene_id %in% in_all$a5) -->

<!-- #res_DTU %>%  -->
<!-- #  dplyr::filter(ens_gene %in% in_all$a5)  -->
<!-- ``` -->


<!-- ```{r} -->
<!-- #results$UPF3B_KD %>% as.data.frame() %>% -->
<!-- #rownames_to_column("tx_id") %>%  -->
<!-- #   mutate_at(.vars = "tx_id", .funs = gsub, pattern = "\\.[0-9]*$", replacement = "") %>% -->
<!-- #  mutate(ens_gene = mapIds(org.Mm.eg.db, keys=.$tx_id, -->
<!-- #                             column="ENSEMBL",keytype="ENSEMBLTRANS", -->
<!-- #                         multiVals="first"))  -->

<!-- ``` -->

<!-- ```{r} -->
<!-- # Overlap between UPF3B genes and transcripts  -->


<!-- #res_NMD %>%  -->
<!-- #  mutate_at(.vars = "tx_id", .funs = gsub, pattern = "\\.[0-9]*$", replacement = "")  -->
<!-- ``` -->



<!-- ```{r fig.height=6, fig.width=10} -->
<!-- sig_res_UPF3B = results_UPF3B %>% -->
<!--   dplyr::filter(qvalue < 0.05, abs(log2FC) > log2(1)) -->

<!-- dEJ = sig_res_UPF3B %>% -->
<!--   dplyr::filter(NIF_dEJ == 1) %>% -->
<!--   dplyr::select(tx_id) -->

<!-- uORF = sig_res_UPF3B %>% -->
<!--   dplyr::filter(NIF_uORF == 1) %>% -->
<!--   dplyr::select(tx_id) -->

<!-- UTR =  sig_res_UPF3B%>% -->
<!--   dplyr::filter(NIF_long_3utr == 1) %>% -->
<!--   dplyr::select(tx_id) -->

<!-- x = list( "dEJ" = dEJ$tx_id, -->
<!--           "uORF" = uORF$tx_id, -->
<!--           "Long 3'UTR" = UTR$tx_id, -->
<!--           "All transcript" = sig_res_UPF3B$tx_id) -->
<!-- upset(fromList(x)) -->
<!-- ``` -->


<!-- # ```{r} -->
<!-- # results_UPF3B %>%  -->
<!-- #   dplyr::filter(qvalue < 0.05) %>%  -->
<!-- #   dplyr::select(tx_id,log2FC,LENGTH_3utr,diffexpressed) %>%  -->
<!-- #   ggplot(aes(x=diffexpressed, y=LENGTH_3utr, fill = diffexpressed)) + geom_boxplot() -->
<!-- # ``` -->
