---
title: "Quality Assessment of NMD RNA-seq data"
author: "unawaz1996"
date: "2023-03-14"
output:
  pdf_document: default
  workflowr::wflow_html:
    code_folding: hide
  html_notebook: default
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
    autodep = TRUE,
    eval = TRUE,
	echo = FALSE,
	warning = FALSE,
	message = FALSE
)
```




## Set-up

```{r}
source("code/libraries.R")
library(stargazer)
library(ggfortify)
library(glue)
library(cowplot)
library(broom)
library(glmpca)
library(naniar)
```

```{r}
twoCols <- c(rgb(0.8, 0.1, 0.1), rgb(0.2, 0.2, 0.8))
```

In order to perform QC, an EnsDB object was obtained using the `AnnotationHub` package. This provided the GC content and length for each of the transcripts contained in the release. 

```{r}
ah <- AnnotationHub() %>%
    subset(species == "Mus musculus") %>%
    subset(rdataclass == "EnsDb") %>% 
  subset(genome == "GRCm39")
ensDb <- ah[["AH104895"]]
grTrans <- transcripts(ensDb)
trLengths <- exonsBy(ensDb, "tx") %>%
    width() %>%
    vapply(sum, integer(1))
mcols(grTrans)$length <- trLengths[names(grTrans)]

genesGR = genes(ensDb)
transGR = transcripts(ensDb)

mcols(transGR) = mcols(transGR) %>%
  cbind(
    transcriptLengths(ensDb)[rownames(.), c("nexon", "tx_len")]
  )
```

```{r}
mcols(genesGR) <- mcols(genesGR) %>%
  as.data.frame() %>%
  dplyr::select(
    gene_id, gene_name, gene_biotype, entrezid
  ) %>%
  left_join(
    mcols(transGR) %>%
      as.data.frame() %>%
      mutate(
        tx_support_level = case_when(
          is.na(tx_support_level) ~ 1L, 
          TRUE ~ tx_support_level
        )
      ) %>%
      group_by(gene_id) %>%
      summarise(
        n_tx = n(),
        longest_tx = max(tx_len),
        ave_tx_len = mean(tx_len),
        gc_content = sum(tx_len*gc_content) / sum(tx_len)
      ) %>%
      mutate(
        bin_length = cut(
          x = ave_tx_len,
          labels = seq_len(10),
          breaks = quantile(ave_tx_len, probs = seq(0, 1, length.out = 11)),
          include.lowest = TRUE
        ),
        bin_gc = cut(
          x = gc_content,
          labels = seq_len(10),
          breaks = quantile(gc_content, probs = seq(0, 1, length.out = 11)),
          include.lowest = TRUE
        ),
        bin = paste(bin_gc, bin_length, sep = "_")
      ),
    by = "gene_id"
  ) %>%
  set_rownames(.$gene_id) %>%
  as("DataFrame")

trans2Gene <- mcols(transGR) %>%
    as.data.frame() %>%
    dplyr::select(tx_id, gene_id) %>%
    dplyr::filter(!is.na(tx_id), !is.na(gene_id)) %>%
    as_tibble()

```

```{r}
txdf = transcripts(EnsDb.Mmusculus.v79, return.type="DataFrame")
tx2gene = as.data.frame(txdf[,c("tx_id","gene_id", "tx_biotype")])
```


Metadata for each fastq file was also loaded. Reads were provided as paired-end reads, with n = 3 samples for each genotype. 

```{r load-md}
salmon.files = ("/home/neuro/Documents/neurogenetics/alignments/Illumina/RNAseq/mouse/LJ_LTK/Salmon/GR39m_decoy_aware")
salmon = list.files(salmon.files, pattern = "transcripts$", full.names = TRUE)
all_files = file.path(salmon, "quant.sf")
names(all_files) <- paste(c(212:229))
md = read.table("data/LTK_Sample Metafile_V3.txt", header= TRUE) %>%  
  mutate(files = file.path(salmon, "quant.sf")) %>% 
  dplyr::rename("names" = "Sample")
comparisons = unique(md$Group)[-1]
md$names = as.character(md$names)
```


```{r}
group_cols <- hcl.colors(
  n = length(unique(md$Group)), 
  palette = "Zissou 1"
  ) %>%
  setNames(unique(md$Group))
```


# FastQC summary 

```{r load-fastqc}
rawFqc <- list.files(
    path = "data/fastqc/Raw",
    pattern = "zip",
    full.names = TRUE
    ) %>%
    FastqcDataList()
trmdFqc <- list.files(
    path = "data/fastqc/Trimmed",
    pattern = "zip",
    full.names = TRUE) %>%
    FastqcDataList()
```




## FastQC summary 

```{r fastq-summary, fig.height=10, fig.width=6, fig.cap="*Basic statistics summary plot. Figure (a) hsows the summary of the PASS/FAIl flags prior to base quality and adapter trimming. Figure (b) shows the summary of PASS/FAIL flags after quality trimming with trimgalore. Green: PASS; Yellow: FAIL; Red: WARN *"}
sum_raw = plotSummary(rawFqc)
sum_trmd = plotSummary(trmdFqc)

grid.arrange(sum_raw, sum_trmd, nrow=2)
```

```{r fig.width = 10, fig.cap="*Sequence Length distribution of the RNA seq reads (a) before and (b) after quality trimming. Only reads with the base length of >  150bp were retained after quality trimming.*"}
dist_raw = ngsReports::plotSeqLengthDistn(rawFqc[[1]])
dist_trmd = ngsReports::plotSeqLengthDistn(trmdFqc[[1]])
grid.arrange(dist_raw, dist_trmd, ncol=2)
```

```{r}
reads <- readTotals(rawFqc)
reads_trimmed = readTotals(trmdFqc)
#reads %>%
  #as.data.frame() %>% 
  #dplyr::rename("Raw" = "Total_Sequences") %>% 
  #cbind(dplyr::rename(as.data.frame(reads_trimmed),"Trimmed" = "Total_Sequences", "New" = "Filename")) %>% 
  #dplyr::select(-New) %>% 
  #mutate(names = gsub("\\_.*", "", Filename)) %>% 
  #left_join(md) %>% 
  #dplyr::select(names, Group, Condition_UPF3B, Condition_UPF3A, "Label"= Please_Label, everything()) %>% 
  #stargazer(title="Sample information including number of raw reads obtained for each sample through Illumina paired-end sequencing and the #number of reads retained after adapter and quality trimming",  summary=FALSE, rownames = FALSE, out = #"output/QC/Thesis_tables/data_summary.tex")
```

## Library sizes

Library sizes for unprocessed data ranged between 86,178,215 and 129,449,694 reads.

```{r fig.height = 7, fig.width =7, fig.cap="*Total numner 0f reads from each sample (a) before and (b) after quality trimming with trimgalore.*"}
r1 <- grepl("_1", fqName(rawFqc))
r2 = grepl("_2", fqName(rawFqc))
trmd_r1 <- grepl("_1", fqName(trmdFqc))
raw_reads = plotReadTotals(rawFqc[r1], barCols = twoCols)
trmd_reads = plotReadTotals(trmdFqc[trmd_r1], barCols = twoCols)
grid.arrange(raw_reads, trmd_reads, nrow=2)

```



## GC content 

In poly(A) selected RNA-seq library preparation methods, the nonuniform coverage of transcripts is a prevalent issue. As poly(A) tail only occurs at the 3' end of the mRNA, this can usually result in an over-representation of the 3' end. Bias at the 5′ end of RNA can also happen because of various factors, such as the fragmentation method (the 5′ end of RNA is more stable), reverse transcription from RNA to cDNA and strand-oriented library construction protocol.

GC content allows the exploration of the sequencing coverage and can indicate issues in overrepresentation. It has been observed that either high or low GC content will result in lower depth coverage. 

```{r}
gcPlots <- list(
   r1 = plotGcContent(
       x = rawFqc[r1],
       plotType = "line",
       gcType = "Transcriptome",
       species = "Mmusculus"
   ),
   r2 = plotGcContent(
       x = rawFqc[!r1],
       plotType = "line",
       gcType = "Transcriptome",
       species = "Mmusculus"
   )
)
lg <- get_legend(gcPlots$r2 + theme(legend.position = "bottom"))
plot_grid(
   plot_grid(
       r1 = gcPlots$r1 +
           ggtitle("R1: GC Distribution", subtitle = c()) +
           theme(legend.position = "none"),
       r2 = gcPlots$r2 +
           ggtitle("R2: GC Distribution", subtitle = c()) +
           theme(legend.position = "none")
   ),
   lg = lg,
   nrow = 2,
   rel_heights = c(5,2)
)

```

## Trimmed data 

```{r}
trimStats = readTotals(rawFqc) %>%
    dplyr::rename(Raw = Total_Sequences, Name = Filename) %>%
    cbind(readTotals(trmdFqc)) %>%
    dplyr::rename(Trimmed = Total_Sequences) %>% 
  dplyr::select(-Filename) %>%
    dplyr::filter(grepl("_1", Name)) %>%
    mutate(
        Discarded = 1 - Trimmed/Raw,
        Retained = Trimmed / Raw
    )
```


```{r}
reads %>%
  as.data.frame() %>% 
  dplyr::rename("Raw" = "Total_Sequences") %>% 
  cbind(dplyr::rename(as.data.frame(reads_trimmed),"Trimmed" = "Total_Sequences", "New" = "Filename")) %>% 
  cbind(dplyr::select(as.data.frame(trimStats), Discarded, Retained)) %>% 
  dplyr::select(-New) %>% 
  mutate(names = gsub("\\_.*", "", Filename)) %>% 
  left_join(md) %>% 
  dplyr::select(names, Group, Condition_UPF3B, Condition_UPF3A, "Label"= Please_Label, everything(), -files) %>% 
  pander(split.tables = Inf)
```


After adapter trimming, < 1% of reads were discarded. 


# Aligned data - Salmon quantifications

Counts were generated using Salmon. Briefly, an index was generated using the GRCm39 build of the mouse transcript with decoys.txt. Selective alignment mode was used. The reads were also aligned to the human genome to check for genotyping and to ensure no mislabeling occurred during any part of the bench work. Counts were imported as transcript-level and gene-level using `tximport` and `tximeta` respectively. 

Annotation data was loaded as an `EnsDb` object, using Ensembl resealse 107. Transcript level gene lengths and GC content was converted to gene level values using:

* GC Content: The total GC content divided by the total length of transcripts 
* Gene Length: The mean transcript length 


```{r transcrpt}
indexDir = file.path("/home/neuro/Documents/NMD_analysis/Analysis/Genome/for_salmon/April_2023/decoy_awaresalmon_index/")
fasta = file.path("/home/neuro/Documents/NMD_analysis/Analysis/Genome/for_salmon/Mus_musculus.GRCm39.cdna.all.fa")
gtf = file.path("/home/neuro/Documents/NMD_analysis/Analysis/Genome/Mus_musculus.GRCm39.107.gtf.gz")
tmp <- tempdir()
jsonFile <- file.path(tmp, paste0(basename(indexDir), ".json"))

makeLinkedTxome(indexDir=indexDir,
                source="Ensembl", organism="Mus musculus",
                release="107", genome="GRCm39.107",
                fasta=fasta, gtf=gtf,
                jsonFile=jsonFile)
md$Group <- factor(md$Group, 
                            levels=c("Control","UPF3B_KD", "UPF3A_KD", "UPF3A_OE", 
                                     "UPF3A_KD_UPF3B_KD", "UPF3A_OE_UPF3B_KD"))

se <- tximeta(md)
keep = (rowSums(se@assays@data$abundance >= 5) >= 3)

```

```{r}
txi_genes =  tximport(all_files, type="salmon", txOut=FALSE,
                      countsFromAbundance="scaledTPM", tx2gene = tx2gene, ignoreTxVersion = TRUE, ignoreAfterBar = TRUE)
keep.genes = (rowSums(txi_genes$abundance >= .5 ) >= 3)
```

## Genotype checking 


```{r, fig.height = 6, fig.width=10, fig.cap="All samples demonstrated the expected expression patterns, no mislabeling was detected in the dataset."}
group.labs <- c("Control", "UPF3A KD", "UPF3A OE", "UPF3B KD", "Double KD", "UPF3A OE, UPF3B KD")
names(group.labs) <- unique(md$Group)


upf3a = txi_genes$abundance["ENSMUSG00000038398",] %>% melt() %>%
   rownames_to_column("name") %>% cbind(md) %>%
   ggplot(aes(x=as.character(names), y= value)) +
   geom_bar(stat="identity", width = 4, color = "black", alpha = 0.9) +
   facet_grid(~Group, labeller = labeller(Group = group.labs) ) + ylab("tpm") + theme_bw() + xlab("") +
   scale_fill_manual(labels = c("UPF3A_KD_UPF3B_KD" = "Double KD", "UPF3A_KD" = "UPF3A KD",
                                 "UPF3B_KD" = "UPF3B KD", "UPF3A_OE" = "UPF3A OE",
                                 "UPF3A_OE_UPF3B_KD" = "UPF3A OE UPF3B KD")) +
   theme(legend.position = "none", axis.text.x=element_blank()) + ggtitle("UPF3A expression")
upf3b = txi_genes$abundance["ENSMUSG00000036572",] %>% melt() %>%
   rownames_to_column("sample") %>% cbind(md) %>%
    ggplot(aes(x=as.character(names), y= value)) +
   geom_bar(stat="identity", width = 4, color = "black", alpha = 0.9) +
   facet_grid(~Group, labeller = labeller(Group = group.labs) ) + ylab("tpm") + theme_bw() + xlab("") +
   scale_fill_manual(labels = c("UPF3A_KD_UPF3B_KD" = "Double KD", "UPF3A_KD" = "UPF3A KD",
                                 "UPF3B_KD" = "UPF3B KD", "UPF3A_OE" = "UPF3A OE",
                                 "UPF3A_OE_UPF3B_KD" = "UPF3A OE UPF3B KD")) +
   theme(legend.position = "none", axis.text.x=element_blank()) + ggtitle("UPF3B expression")

grid.arrange(upf3a, upf3b)

```

## Filtering 

```{r fig.height=6, fig.width=6}
exp_den_trans = log10(se@assays@data$abundance + 0.01) %>%
    melt() %>%
    dplyr::filter(is.finite(value)) %>% 
    ggplot(aes(x=value, color = as.character(Var2))) +
    geom_density() +
    ggtitle("Before filtering") +
    labs(x = "logTPM", y = "Proportion of Transcripts") + 
    theme_bw() +
    theme(legend.position = "none")

exp_den_filt_trans = log10(se@assays@data$abundance[keep,] + 0.01) %>%
    melt() %>%
    dplyr::filter(is.finite(value)) %>% 
    ggplot(aes(x=value, color = as.character(Var2))) +
    geom_density() +
    ggtitle("Before filtering") +
    labs(x = "logTPM", y = "Proportion of Transcripts") + 
    theme_bw() +
    theme(legend.position = "none")

```

```{r}
exp_den = txi_genes$counts %>% cpm(log=TRUE) %>%
    melt() %>%
    dplyr::filter(is.finite(value)) %>% 
    ggplot(aes(x=value, color = as.character(Var2))) +
    geom_density() +
    ggtitle("Before filtering") +
    labs(x = "logCPM", y = "Proportion of Genes") + 
    theme_bw() +
    theme(legend.position = "none")

exp_den_filt = txi_genes$counts %>%
    cpm(log=TRUE) %>%
    magrittr::extract(keep.genes,) %>% 
    melt() %>%
    dplyr::filter(is.finite(value)) %>% 
    ggplot(aes(x=value, color = as.character(Var2))) +
    geom_density() +
    ggtitle("After filtering") +
    labs(x = "logCPM", y = "Proportion of Genes") + theme_bw() +
    theme(legend.position = "none")

grid.arrange(exp_den,exp_den_filt, exp_den_trans, exp_den_filt_trans, 
             ncol=2, nrow=2)
```


```{r transcript-level}
dge = se@assays@data$counts[keep,] %>% 
  as.data.frame %>% 
  rownames_to_column("tx_id") %>% 
  set_colnames(basename(colnames(.))) %>% 
  dplyr::filter(tx_id %in% trans2Gene$tx_id) %>% 
  pivot_longer(cols = all_of(md$names), names_to = "Mm", values_to = "count") %>% 
    left_join(trans2Gene) %>%
    group_by(Mm, gene_id) %>%
    summarise(count = sum(count)) %>% 
  pivot_wider(names_from = "Mm", values_from = "count") %>% 
    dplyr::filter(grepl("ENSMUS", gene_id)) %>%
    as.data.frame() %>% 
    column_to_rownames("gene_id") %>% 
    DGEList() 
dge$samples %<>%
    mutate(names = rownames(.)) %>%
  dplyr::select(-group) %>% 
    left_join(md, by = "names") %>%
    set_rownames(.$names)
dge$genes <- genesGR[rownames(dge)] %>% 
  mcols()
```

## Library size 
```{r}
dge$samples %>% 
  mutate(lib.size = lib.size / 1e6) %>% 
  ggplot(aes(names, lib.size, fill = Group)) +
  geom_col() +
  facet_wrap(~Group, scales = "free_x") +
  scale_y_continuous(expand = expansion(c(0, .05))) +
  scale_fill_manual(values = group_cols) +
  labs(
    x = "Sample", y = "Library Size (millions)",
    fill = "Group"
  ) + theme(legend.position = "none")
```

## Counts assingment rate 

```{r}
trmdFqc %>% 
  getModule("Basic") %>% 
  mutate(names = gsub("\\_.*", "", Filename)) %>% 
  left_join(dge$samples, by = "names") %>% 
  mutate(`% Assigned To Genes` = lib.size / Total_Sequences) %>% 
  ggplot(aes(names, `% Assigned To Genes`, fill = Group)) +
  geom_col() +
  facet_wrap(~Group, scales = "free_x") +
  scale_fill_manual(values = group_cols) +
  scale_y_continuous(labels = percent, expand = expansion(c(0, 0.05))) +
  labs(fill = "Group")
```

## Total detected genes 

```{r}
dge$counts %>%
  as_tibble() %>%
  mutate(
    across(everything(), as.logical)
  ) %>%
  summarise(
    across(everything(), sum)
  ) %>%
  pivot_longer(
    everything(), names_to = "names", values_to = "Detected"
  ) %>%
  left_join(md)%>%
  ggplot(aes(Group, Detected, colour = Group)) +
  geom_point() +
  geom_segment(
    aes(xend = Group, y = 0, yend = Detected),
    data = . %>% 
      group_by(Group) %>%
      summarise(Detected = min(Detected)),
    colour = "black", size = 1/4) +
  scale_y_continuous(labels = comma, expand = expansion(c(0, 0.05))) +
  scale_colour_manual(values = group_cols) +
  labs(
    x = "Group", 
    y = "Genes Detected",
    colour = "Group"
  )

```

## Transcript level exploration of data


```{r}
## Filtering threshold
before =  se@assays@data$abundance %>% as.data.frame() %>%
    mutate(rowMeans = rowMeans(.)) %>% 
  rownames_to_column("tx_id") %>%
  left_join(tx2gene, by = "tx_id") %>%
  group_by(gene_id) %>% mutate(ntx = n()) %>% 
dplyr::select(gene_id, tx_id, tx_biotype, rowMeans, ntx) 

after = se@assays@data$abundance[keep,] %>% as.data.frame() %>%
    mutate(rowMeans = rowMeans(.)) %>% 
  rownames_to_column("tx_id") %>%
  left_join(tx2gene, by = "tx_id") %>%
  group_by(gene_id) %>% mutate(ntx = n()) %>% 
dplyr::select(gene_id, tx_id, tx_biotype, rowMeans, ntx) 


df <- gdata::combine(before, after) %>% dplyr::rename(Filter=source)

dplyr::group_by(df,Filter) %>% dplyr::summarise(median(rowMeans),sd(rowMeans))


ntxInfo  <- df %>% dplyr::group_by(Filter) %>%
    dplyr::distinct(gene_id,.keep_all = TRUE)  
biotypeCount <- df %>% dplyr::group_by(Filter) %>%
    dplyr::count(tx_biotype) %>%
    mutate(sum=sum(n),frac = n / sum(n)) 

```


```{r fig.height=6, fig.width=6, fig.cap= "*Log10 of the mean TPMs (transcript per million) over all samples before and after filtering out low-expressed transcripts and genes*"}
distribution = ggplot(df, aes(x = log10(rowMeans), col = Filter)) +
    geom_histogram(aes(y = ..density.., fill = Filter), alpha = 0.4, bins = 100) +
    geom_density(size = 1.1) +
    scale_x_continuous(name = "log10(mean TPM)", limits = c(-8,10)) +
    theme_bw() +
    scale_color_manual(values = c("#FCC98A", "#DF1861"))+
    scale_fill_manual(values = c("#FCC98A", "#DF1861"))
distribution
#ggsave(plot = dist, file = "/home/neuro/Documents/NMD_analysis/Analysis/NMD-analysis/output/Transcript/Thesis_figures/transcript-distribution.svg", 
 #      height = 4.29, width = 8.02, units = "in")
```



```{r fig.height=6, fig.width=6, fig.cap="*Violin plots showing the distribution of the number of transcripts per gene (in logarithmic scale). Violin width is scaled by the total number of observations while jittered points represent actual observations.*"}
biocount = ggplot(subset(biotypeCount, frac >= 0.001), aes(x = as.factor(tx_biotype), y = frac, fill = Filter)) +
    geom_bar( stat = "identity") +
    coord_flip() +
    facet_wrap(.~Filter) +
    theme_bw() +
    theme(axis.text.x = element_text(angle = -90)) +
    labs(x = "Transcript biotype (Ensembl v75)", y = "Frequency") +
        scale_color_manual(values = c("#FCC98A", "#DF1861"))+
    scale_fill_manual(values = c("#FCC98A", "#DF1861"))+  
theme(panel.spacing = unit(1, "lines")) +
theme(strip.background = element_rect(fill = "black"))+
theme(strip.text = element_text(color = "white", face = "bold"))
#ggsave(plot = biocount, file ="/home/neuro/Documents/NMD_analysis/Analysis/NMD-analysis/output/Transcript/Thesis_figures/transcript-composition.svg", 
 #      height = 4.05, width = 7.74, units = "in")
```


```{r  fig.height=6, fig.width=6, fig.cap="*Violin plots showing the distribution of the number of transcripts per gene (in logarithmic scale). Violin width is scaled by the total number of observations while jittered points represent actual observations.*"}
gene_info =ggplot(ntxInfo, aes(x = Filter, y = log10(ntx))) +
    geom_violin(scale = "count", aes(fill = Filter)) +
    geom_jitter(width = 0.1, size = 0.3, alpha = 0.01) +
    theme_bw() +
    scale_y_continuous(name = "log10(# transcripts per gene)") + 
    labs(x = "") +
        scale_color_manual(values = c("deepskyblue3", "darkblue"))+
    scale_fill_manual(values = c("deepskyblue3", "darkblue"))   
gene_info
```


## PCA 


```{r transcript-pca}
pca = log(assays(se)$abundance[keep,] + 0.05) %>%
  t() %>% 
  prcomp(scale = TRUE)

showLabel = nrow(md) <= 18
trans_pca =  pca %>% 
  tidy() %>% 
  dplyr::rename(names = row) %>% 
  left_join(dge$samples, by = "names") %>%
   dplyr::filter(PC %in% 1:3) %>% 
  pivot_wider(names_from = "PC", names_prefix = "PC", values_from = "value") %>% 
  ggplot(
    aes(PC1, PC2, colour = Group, size = lib.size/1e6)
  ) +
  geom_point() +
  geom_text_repel(aes(label = names), show.legend = FALSE) +
  scale_colour_manual(values = group_cols, 
                      labels = c("UPF3A_KD_UPF3B_KD" = "UPF3 dKD", "UPF3A_KD" = "UPF3A KD", 
                                                                   "UPF3B_KD" = "UPF3B KD", "UPF3A_OE" = "UPF3A OE", 
                                                                   "UPF3A_OE_UPF3B_KD" = "UPF3A OE UPF3B KD")) +
  #scale_size_continuous(limits = c(5, 15), breaks = seq(5, 15, by = 5)) +
  labs(
    x = glue("PC1 ({percent(pca$sdev[[1]]^2 / sum(pca$sdev^2), 0.1)})"),
    y = glue("PC2 ({percent(pca$sdev[[2]]^2 / sum(pca$sdev^2), 0.1)})"),
    colour = "Group",
    size = "Library Size\n(millions)"
  ) + theme_bw() + ggtitle("Transcript Level")
```


```{r fig.height = 7, fig.width=10, fig.cap="*Principal component analysis of gene (left) and transcript (right) level data. PCA was performed on log2 transformed TPMs after filtering for each respective datatype (gene and transcript level). Gene level PCA shows that all samples are clustering closely based on condition, with no impact of library size in PC1 and PC2. Transcript level PCA shows that samples are clustering close to their conditions based on PC1, however one of the samples of the UPF3A OE in UPF3B KD cell line (sample 223), seems to deviate from its condition group and the rest of the data, so needs to be further investigated*"}

pca_genes = log2(txi_genes$abundance[keep.genes,]  + 10^-6 ) %>%
    t() %>% 
    prcomp(scale = TRUE)

pcaVars <- percent_format(0.1)(summary(pca)$importance["Proportion of Variance",]) 

gene_pca = pca_genes %>% 
  tidy() %>% 
  dplyr::rename(names = row) %>% 
  left_join(dge$samples, by = "names") %>%
   dplyr::filter(PC %in% 1:3) %>% 
  pivot_wider(names_from = "PC", names_prefix = "PC", values_from = "value") %>% 
  ggplot(
    aes(PC1, PC2, colour = Group, size = lib.size/1e6)
  ) +
  geom_point() +
  geom_text_repel(aes(label = names), show.legend = FALSE) +
  scale_colour_manual(values = group_cols, 
                         labels = c("UPF3A_KD_UPF3B_KD" = "UPF3 dKD", "UPF3A_KD" = "UPF3A KD", 
                                                                   "UPF3B_KD" = "UPF3B KD", "UPF3A_OE" = "UPF3A OE", 
                                                                   "UPF3A_OE_UPF3B_KD" = "UPF3A OE UPF3B KD")) +
  #scale_size_continuous(limits = c(5, 15), breaks = seq(5, 15, by = 5)) +
  labs(
    x = glue("PC1 ({percent(pca_genes$sdev[[1]]^2 / sum(pca_genes$sdev^2), 0.1)})"),
    y = glue("PC2 ({percent(pca_genes$sdev[[2]]^2 / sum(pca_genes$sdev^2), 0.1)})"),
    colour = "Group",
    size = "Library Size\n(millions)"
  ) + theme_bw()  + ggtitle("Gene Level")

ggarrange(gene_pca, trans_pca , ncol=2,  common.legend = TRUE, legend="right")


```

To check if the same variation is observed at PC3 in gene level: 

```{r fig.height = 4, fig.width=6, fig.cap= "*Principal component analysis of gene level data showing PC2 and PC3. Sample 223 seems to cluster away from its condition group in PC3*"}
pca_genes %>% 
  tidy() %>% 
  dplyr::rename(names = row) %>% 
  left_join(dge$samples, by = "names") %>%
   dplyr::filter(PC %in% 1:3) %>% 
  pivot_wider(names_from = "PC", names_prefix = "PC", values_from = "value") %>% 
  ggplot(
    aes(PC2, PC3, colour = Group, size = lib.size/1e6)
  ) +
  geom_point() +
  geom_text_repel(aes(label = names), show.legend = FALSE) +
  scale_colour_manual(values = group_cols) +
  #scale_size_continuous(limits = c(5, 15), breaks = seq(5, 15, by = 5)) +
  labs(
    x = glue("PC2 ({percent(pca_genes$sdev[[2]]^2 / sum(pca_genes$sdev^2), 0.1)})"),
    y = glue("PC3 ({percent(pca_genes$sdev[[3]]^2 / sum(pca_genes$sdev^2), 0.1)})"),
    colour = "Group",
    size = "Library Size\n(millions)"
  ) + theme_bw() 

```

### Investigating the differences in samples at transcript level 

```{r}
exp_upf3a_oe = log2(se@assays@data$abundance[keep,]+ 10^-6) %>% 
  as.data.frame() %>%
  dplyr::select("223", "217", "229")

#qplot(log2(exp_upf3aoe[,2]), exp_upf3aoe[,3]) +geom_abline(intercept = 0, slope = 1)
grid.arrange((qplot(exp_upf3a_oe[,1], exp_upf3a_oe[,2], xlab="223", ylab= "217") +geom_abline(intercept = 0, slope = 1)), 
             (qplot(exp_upf3a_oe[,1], exp_upf3a_oe[,3], xlab= "223", ylab="229") +geom_abline(intercept = 0, slope = 1)),
             (qplot(exp_upf3a_oe[,2], exp_upf3a_oe[,3], xlab= "217", ylab= "229" ) +geom_abline(intercept = 0, slope = 1)), ncol =3)
```

```{r}
exp_upf3a_oe %>% melt() %>% 
  ggplot(aes(x=variable, y = value)) + geom_boxplot()
```

```{r}
gc <- getModule(rawFqc, "Per_sequence_GC") 

gcDev <- gc %>%
  mutate(names = gsub("\\_.*", "", Filename)) %>% 
    left_join(md, by = "names") %>%
    group_by(names) %>%
    mutate(Freq = Count / sum(Count)) %>%
    left_join(
        getGC(gcTheoretical, "Mmusculus", "Trans")) %>%
    dplyr::rename(actual = Mmusculus) %>%
    mutate(res = Freq - actual) %>%
    summarise(ss = sum(res^2), n = n()) %>%
    ungroup() %>%
    mutate(sd = sqrt(ss / (n - 1)))

rawGC <- gc %>% 
    group_by(Filename) %>% 
    mutate(Freq = Count / sum(Count)) %>%
    dplyr::filter(GC_Content > 70) %>% 
    summarise(Freq = sum(Freq)) %>% 
    arrange(desc(Freq)) %>%
     mutate(names = gsub("\\_.*", "", Filename)) %>% 
    left_join(md, by= "names") 


gcInfo <- se@assays@data$abundance[keep,] %>%
    as.data.frame() %>%
    rownames_to_column("tx_id")  %>%
    as_tibble() %>%
    pivot_longer(
        cols =  starts_with("2"), 
        names_to = "sampleName", 
        values_to = "counts"
    ) %>%
    dplyr::filter(
        counts > 0
    )  %>%
    left_join(
        mcols(grTrans) %>% as.data.frame()
    ) %>% 
   dplyr::select(
        ends_with("id"), sampleName, counts, gc_content, length
    ) %>%
    split(f = .$sampleName) %>%
    lapply(function(x){
        DataFrame(
            gc = Rle(x$gc_content/100, x$counts),
            logLen = Rle(log10(x$length), x$counts)
        )
    }
    ) 


gcSummary <- gcInfo %>%
    vapply(function(x){
        c(mean(x$gc), sd(x$gc), mean(x$logLen), sd(x$logLen))
        }, numeric(4)
        ) %>%
    t() %>%
    set_colnames(
        c("mn_gc", "sd_gc", "mn_logLen", "sd_logLen")
    ) %>%
    as.data.frame() %>%
    rownames_to_column("names") %>%
    as_tibble() %>%
    left_join(dplyr::filter(md)) #%>%
    #dplyr::select(starts_with("sample"), contains("_"))

```

```{r fig.cap="*Correlations between the first three principal components and measured variables at transcript level. Sample conditions were converted to an ordered categorical variable for the purposes of visualisation*"}
pca$x %>%
    as.data.frame() %>%
  rownames_to_column("names") %>%
    left_join(gcSummary) %>%
    as_tibble() %>%
     left_join(
        dplyr::filter(rawGC)
    ) %>% 
  left_join(gcDev) %>%
    dplyr::select(
        PC1, PC2, PC3, 
        Mean_GC = mn_gc, 
        Mean_Length = mn_logLen, 
        Group
    ) %>% 
   mutate(Group = as.numeric(as.factor(Group))) %>%
    cor() %>% 
    corrplot(
        type = "lower", 
        diag = FALSE, 
        addCoef.col = 1)
```


Transcripts were divided in 10 approximately equal sized bins based on increasing length, and 10 approximately equal sized bins based on increasing GC content, with the final GC/Length bins being the combination 100 bins using both sets. The contribution of each gene to PC1 and PC2 was assessed and a t-test performed on each bin.

If any bin makes a contribution to PC1 the mean will be clearly non-zero, whilst if there is no contribution the mean will be near zero. In this way, the impact of gene length and GC content on variance within the dataset can be assessed. 

```{r fig.cap="*Contribution of each GC/Length Bin to PC1 and PC2. Fill colours indicate the t-statistic, with tranparency denoting significance as -log10(p), using Bonferroni-adjusted p-values.*"}
mcols(transGR) %>%
  as.data.frame() %>%
  dplyr::filter(tx_id %in% rownames(pca$rotation)) %>%
  as_tibble() %>%
  mutate(
    bin_length = cut(
      x = tx_len,
      labels = seq_len(10),
      breaks = quantile(tx_len, probs = seq(0, 1, length.out = 11)),
      include.lowest = TRUE
    ),
    bin_gc = cut(
      x = gc_content,
      labels = seq_len(10),
      breaks = quantile(gc_content, probs = seq(0, 1, length.out = 11)),
      include.lowest = TRUE
    ),
    bin = paste(bin_gc, bin_length, sep = "_")
  ) %>%
  dplyr::select(tx_id, contains("bin")) %>%
  mutate(
    PC1 = pca$rotation[tx_id, "PC1"],
    PC2 = pca$rotation[tx_id, "PC2"]
  ) %>%
  pivot_longer(
    cols = c("PC1", "PC2"),
    names_to = "PC",
    values_to = "value"
  ) %>%
  group_by(PC, bin_gc, bin_length, bin) %>%
  summarise(
    Size = n(),
    mean = mean(value),
    sd = sd(value),
    t = t.test(value)$statistic,
    p = t.test(value)$p.value,
    adjP = p.adjust(p, method = "bonf")
  ) %>%
  ggplot(
    aes(bin_length, bin_gc, colour = t, alpha = -log10(adjP), size = Size)
  ) +
  geom_point() +
  facet_wrap(~PC) +
  scale_colour_gradient2() +
  scale_size_continuous(range = c(1, 10)) +
  labs(
    x = "Transcript Length",
    y = "GC Content",
    alpha = expression(paste(-log[10], p[adj]))) +
  theme(
    panel.grid = element_blank(),
    legend.position = "bottom"
    ) 
```

## Correlation of samples

```{r}
pheatmap(cor(log(se@assays@data$abundance[keep,]  + 10^-6 )))

log_trans_cor = cor(log(se@assays@data$abundance[keep,]  + 10^-6 ))

log_trans_cor[log_trans_cor==1] <- NA

trans = rowMeans(log_trans_cor, na.rm = TRUE) %>% scale() %>% as.data.frame() %>%
  set_colnames("transcript")


log_gene_cor = cor(log2(txi_genes$abundance[keep.genes,]  + 10^-6 ))
log_gene_cor[log_gene_cor==1] <- NA
gene = rowMeans(log_gene_cor, na.rm = TRUE) %>% scale() %>% as.data.frame() %>%
  set_colnames("gene")
```


```{r}

cbind(trans,gene) %>% as.data.frame() %>%
  DT::datatable(caption = "For each sample, a mean of its correlation to the other samples was calculated followed by a z-score transformation. In principle, samples that deviate 3 standard deviations away from the mean would be removed.")

```

### Correlation of samples with GC and transcript length

```{r}
cor_with_attributes = se@assays@data$abundance[keep,] %>% as.data.frame() %>%
  rownames_to_column("tx_id") %>% 
  left_join(transGR %>% 
              as.data.frame() %>% 
              dplyr::select(tx_id, gc_content, tx_len)) %>% 
  column_to_rownames("tx_id")

heatmap(cor(log2(cor_with_attributes + 10^-6)))

cor_with_attributes_log = log2(cor_with_attributes + 10^-6)

length = qplot(cor_with_attributes_log$`223`, cor_with_attributes_log$tx_len, xlab = "logTPM 223", ylab = "log transcript length")
gc = qplot(cor_with_attributes_log$`223`, cor_with_attributes_log$gc_content, xlab = "logTPM 223", ylab = "log GC content")
grid.arrange(length, gc)
```




```{r}
library(cqn)
dge = makeDGEList(se[keep,])
dge$genes %<>% 
  left_join(transGR %>% 
              as.data.frame() %>% 
              dplyr::select(tx_id, tx_len))

cqNorm <- with(
  dge,
  cqn(
    counts= counts,
    x = genes$gc_content,
    lengths = genes$tx_len
  )
)


dge$offset <- cqNorm$glm.offset
logCPM <- cqNorm$y + cqNorm$offset

a = cqNorm$func1 %>%
  as.data.frame() %>%
  mutate(x = cqNorm$grid1) %>%
  pivot_longer(
    cols = any_of(colnames(dge)),
    names_to = "names",
    values_to = "QR fit"
  ) %>%
  left_join(dge$samples) %>%
  ggplot(
    aes(x, `QR fit`, colour = Group, group = names, linetype = Condition_UPF3B )
  ) +
  geom_line() +
  labs(x = "GC content", colour = "Replicate", linetype = "Treatment")

b = cqNorm$func2 %>%
  as.data.frame() %>%
  mutate(x = cqNorm$grid2) %>%
  pivot_longer(
    cols = any_of(colnames(dge)),
    names_to = "names",
    values_to = "QR fit"
  ) %>%
  left_join(dge$samples) %>%
  ggplot(
    aes(x, `QR fit`, colour = Group, group = names, linetype = Condition_UPF3B)
  ) +
  geom_line() +
  labs(
    x = expression(paste(log[10], " Transcript Length (kb)")),
    colour = "Replicate", linetype = "Treatment"
  )


plot_grid(
  a + theme(legend.position = "none"), 
  b + theme(legend.position = "none"),
  get_legend(a),
  nrow = 1,
  rel_widths = c(3, 3, 1)
)

```


```{r fig.cap = "PCA on logCPM from counts post normalization with cqn. As a result of the normlization, it seems that whilst the overall variation within the dataset has reduced, sample 223 is no longer cluster with its condition group on PC1 and PC2"}
### PCA post-normalization

pcaPost <- logCPM %>%
  t() %>%
  prcomp() 


pcaPost$x %>% 
  as.data.frame() %>%
  rownames_to_column("names") %>%
  left_join(dge$samples, by = "names") %>%
  ggplot(
    aes(PC1, PC2, colour = Group,  shape = Condition_UPF3B)
  ) +
  geom_point() +
  geom_text_repel(aes(label = names), show.legend = FALSE) +
  scale_colour_manual(values = group_cols) +
  #scale_size_continuous(limits = c(5, 15), breaks = seq(5, 15, by = 5)) +
  labs(
    x = glue("PC1 ({percent(pcaPost$sdev[[1]]^2 / sum(pca$sdev^2), 0.1)})"),
    y = glue("PC2 ({percent(pcaPost$sdev[[2]]^2 / sum(pca$sdev^2), 0.1)})"),
    colour = "Group",
    size = "Library Size\n(millions)"
  ) + theme_bw() + ggtitle("Transcript Level")



```

## Removing sample

```{r}
tx_df_removed = assays(se)$abundance[,-12]
pca= log(tx_df_removed[keep,] + 0.05) %>%  
  t() %>% 
  prcomp(scale = TRUE)
  
pca %>% 
  tidy() %>% 
  dplyr::rename(names = row) %>% 
  left_join(md %>% dplyr::filter(names != "223"), by = "names") %>%
   dplyr::filter(PC %in% 1:3) %>% 
  pivot_wider(names_from = "PC", names_prefix = "PC", values_from = "value") %>% 
  ggplot(
    aes(PC1, PC2, colour = Group)
  ) +
  geom_point() +
  geom_text_repel(aes(label = names), show.legend = FALSE) +
  scale_colour_manual(values = group_cols, 
                      labels = c("UPF3A_KD_UPF3B_KD" = "UPF3 dKD", "UPF3A_KD" = "UPF3A KD", 
                                                                   "UPF3B_KD" = "UPF3B KD", "UPF3A_OE" = "UPF3A OE", 
                                                                   "UPF3A_OE_UPF3B_KD" = "UPF3A OE UPF3B KD")) +
  #scale_size_continuous(limits = c(5, 15), breaks = seq(5, 15, by = 5)) +
  labs(
    x = glue("PC1 ({percent(pca$sdev[[1]]^2 / sum(pca$sdev^2), 0.1)})"),
    y = glue("PC2 ({percent(pca$sdev[[2]]^2 / sum(pca$sdev^2), 0.1)})"),
    colour = "Group",
    size = "Library Size\n(millions)"
  ) + theme_bw() + ggtitle("Transcript Level")

```


<!-- # ```{r} -->
<!-- # ## deseq2 -->
<!-- # ddsTxi <- DESeqDataSet(se, design = ~Group) -->
<!-- # vsd <- vst(ddsTxi) -->
<!-- # DESeq2::plotPCA(vsd, intgroup = "Group") -->
<!-- #  -->
<!-- # ``` -->


<!-- # ```{r} -->
<!-- # library(EnsDb.Hsapiens.v86) -->
<!-- # txdf = transcripts(EnsDb.Hsapiens.v86, return.type="DataFrame") -->
<!-- # tx2gene = as.data.frame(txdf[,c("tx_id","gene_id", "tx_biotype")]) -->
<!-- # salmon.files = ("/home/neuro/Documents/neurogenetics/alignments/Illumina/RNAseq/mouse/LJ_LTK/Salmon/Salmon_phoenix") -->
<!-- # salmon = list.files(salmon.files, pattern = "transcripts$", full.names = TRUE) -->
<!-- # all_files = file.path(salmon, "quant.sf") -->
<!-- # names(all_files) <- paste(c(212:229)) -->
<!-- # txi_genes_human  = tximport(all_files, type="salmon", txOut=FALSE, -->
<!-- #                       countsFromAbundance="scaledTPM", tx2gene = tx2gene, ignoreTxVersion = TRUE, ignoreAfterBar = TRUE) -->
<!-- #  -->
<!-- #  -->
<!-- # txi_genes_human$abundance["ENSG00000169062",] %>% melt() %>% -->
<!-- #    rownames_to_column("name") %>% cbind(md) %>% -->
<!-- #    ggplot(aes(x=as.character(names), y= value)) + -->
<!-- #    geom_bar(stat="identity", color = "black", alpha = 0.9) + -->
<!-- #    facet_grid(~Group, labeller = labeller(Group = group.labs), scales = "free_x") + ylab("tpm") + theme_bw() + xlab("") + -->
<!-- #    scale_fill_manual(labels = c("UPF3A_KD_UPF3B_KD" = "Double KD", "UPF3A_KD" = "UPF3A KD", -->
<!-- #                                  "UPF3B_KD" = "UPF3B KD", "UPF3A_OE" = "UPF3A OE", -->
<!-- #                                  "UPF3A_OE_UPF3B_KD" = "UPF3A OE UPF3B KD")) + -->
<!-- #    theme(legend.position = "none") + ggtitle("UPF3A expression") -->
<!-- # ``` -->

